{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "520450df-370b-4e6a-99c3-4f49df75c670",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "# FASTLMM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "e53ce06e-5163-42fa-9857-6b3ca27c1514",
   "metadata": {
    "kernel": "Bash",
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 82208\n",
      "-rw-r--r-- 1 jovyan users     7222 Apr 17 21:21 FASTLMM_GCTA_commands.txt\n",
      "-rw-r--r-- 1 jovyan users 42413693 Apr 18 15:29 FLMMresults\n",
      "-rw-r--r-- 1 jovyan users     2217 Apr 18 15:31 GCTAresults.log\n",
      "-rw-r--r-- 1 jovyan users  8369621 Apr 18 15:31 GCTAresults.mlma\n",
      "-rw-r--r-- 1 jovyan users     1089 Apr 17 21:21 LICENSE\n",
      "-rw-r--r-- 1 jovyan users     1075 Apr 17 21:23 SYNAPSE_METADATA_MANIFEST.tsv\n",
      "-rw-r--r-- 1 jovyan users    38887 Apr 18 15:33 Untitled.ipynb\n",
      "-rw-r--r-- 1 jovyan users    27276 Apr 18 15:28 \u001b[0m\u001b[01;35mmh1.png\u001b[0m\n",
      "-rw-r--r-- 1 jovyan users    26453 Apr 18 15:29 \u001b[01;35mmh2.png\u001b[0m\n",
      "-rw-r--r-- 1 jovyan users      278 Apr 17 21:23 multipleGRMs.txt\n",
      "-rw-r--r-- 1 jovyan users       24 Apr 17 21:23 multipleGRMs6andnot6.txt\n",
      "-rw-r--r-- 1 jovyan users      266 Apr 17 21:23 multipleGRMsnot6.txt\n",
      "-rw-r--r-- 1 jovyan users     9627 Apr 17 21:23 myphenos.txt\n",
      "-rw-r--r-- 1 jovyan users     9627 Apr 18 15:29 phenos.txt\n",
      "-rw-r--r-- 1 jovyan users      851 Apr 18 15:28 plinkresults.log\n",
      "-rw-r--r-- 1 jovyan users 12550638 Apr 18 15:28 plinkresults.qassoc\n",
      "-rw-r--r-- 1 jovyan users    14775 Apr 18 15:28 \u001b[01;35mqq1.png\u001b[0m\n",
      "-rw-r--r-- 1 jovyan users    14192 Apr 18 15:29 \u001b[01;35mqq2.png\u001b[0m\n",
      "-rw-r--r-- 1 jovyan users     7019 Apr 17 21:21 qqmanHJCupdated.R\n",
      "-rw-r--r-- 1 jovyan users 16868253 Apr 17 21:23 quantfamdata.bed\n",
      "-rw-r--r-- 1 jovyan users  3751046 Apr 17 21:23 quantfamdata.bim\n",
      "-rw-r--r-- 1 jovyan users    14938 Apr 17 21:23 quantfamdata.fam\n",
      "PLINK v1.90b6.21 64-bit (19 Oct 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to plinkresults.log.\n",
      "Options in effect:\n",
      "  --assoc\n",
      "  --bfile quantfamdata\n",
      "  --out plinkresults\n",
      "\n",
      "31376 MB RAM detected; reserving 15688 MB for main workspace.\n",
      "134946 variants loaded from .bim file.\n",
      "498 people (226 males, 272 females) loaded from .fam.\n",
      "498 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 124 founders and 374 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999457.\n",
      "134946 variants and 498 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing QT --assoc report to plinkresults.qassoc ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "ls -l\n",
    "\n",
    "plink --bfile quantfamdata --assoc --out plinkresults "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "891d787b-9d96-4d23-9c9b-591e21506d55",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>CHR</th><th scope=col>SNP</th><th scope=col>BP</th><th scope=col>NMISS</th><th scope=col>BETA</th><th scope=col>SE</th><th scope=col>R2</th><th scope=col>T</th><th scope=col>P</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>rs9326624 </td><td> 558615</td><td>498</td><td> 0.526900</td><td>0.9159</td><td>6.668e-04</td><td> 0.57530</td><td>0.5654</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1</td><td>rs3094315 </td><td> 742429</td><td>498</td><td> 0.306700</td><td>0.5377</td><td>6.555e-04</td><td> 0.57040</td><td>0.5687</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1</td><td>rs12562034</td><td> 758311</td><td>495</td><td> 0.793000</td><td>0.8845</td><td>1.628e-03</td><td> 0.89650</td><td>0.3704</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1</td><td>rs3934834 </td><td> 995669</td><td>498</td><td>-0.007573</td><td>0.6342</td><td>2.875e-07</td><td>-0.01194</td><td>0.9905</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1</td><td>rs3737728 </td><td>1011278</td><td>498</td><td> 0.465700</td><td>0.6392</td><td>1.069e-03</td><td> 0.72860</td><td>0.4666</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1</td><td>rs4970405 </td><td>1038818</td><td>498</td><td>-0.829400</td><td>0.7574</td><td>2.412e-03</td><td>-1.09500</td><td>0.2740</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 9\n",
       "\\begin{tabular}{r|lllllllll}\n",
       "  & CHR & SNP & BP & NMISS & BETA & SE & R2 & T & P\\\\\n",
       "  & <int> & <chr> & <int> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & rs9326624  &  558615 & 498 &  0.526900 & 0.9159 & 6.668e-04 &  0.57530 & 0.5654\\\\\n",
       "\t2 & 1 & rs3094315  &  742429 & 498 &  0.306700 & 0.5377 & 6.555e-04 &  0.57040 & 0.5687\\\\\n",
       "\t3 & 1 & rs12562034 &  758311 & 495 &  0.793000 & 0.8845 & 1.628e-03 &  0.89650 & 0.3704\\\\\n",
       "\t4 & 1 & rs3934834  &  995669 & 498 & -0.007573 & 0.6342 & 2.875e-07 & -0.01194 & 0.9905\\\\\n",
       "\t5 & 1 & rs3737728  & 1011278 & 498 &  0.465700 & 0.6392 & 1.069e-03 &  0.72860 & 0.4666\\\\\n",
       "\t6 & 1 & rs4970405  & 1038818 & 498 & -0.829400 & 0.7574 & 2.412e-03 & -1.09500 & 0.2740\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 9\n",
       "\n",
       "| <!--/--> | CHR &lt;int&gt; | SNP &lt;chr&gt; | BP &lt;int&gt; | NMISS &lt;int&gt; | BETA &lt;dbl&gt; | SE &lt;dbl&gt; | R2 &lt;dbl&gt; | T &lt;dbl&gt; | P &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | rs9326624  |  558615 | 498 |  0.526900 | 0.9159 | 6.668e-04 |  0.57530 | 0.5654 |\n",
       "| 2 | 1 | rs3094315  |  742429 | 498 |  0.306700 | 0.5377 | 6.555e-04 |  0.57040 | 0.5687 |\n",
       "| 3 | 1 | rs12562034 |  758311 | 495 |  0.793000 | 0.8845 | 1.628e-03 |  0.89650 | 0.3704 |\n",
       "| 4 | 1 | rs3934834  |  995669 | 498 | -0.007573 | 0.6342 | 2.875e-07 | -0.01194 | 0.9905 |\n",
       "| 5 | 1 | rs3737728  | 1011278 | 498 |  0.465700 | 0.6392 | 1.069e-03 |  0.72860 | 0.4666 |\n",
       "| 6 | 1 | rs4970405  | 1038818 | 498 | -0.829400 | 0.7574 | 2.412e-03 | -1.09500 | 0.2740 |\n",
       "\n"
      ],
      "text/plain": [
       "  CHR SNP        BP      NMISS BETA      SE     R2        T        P     \n",
       "1 1   rs9326624   558615 498    0.526900 0.9159 6.668e-04  0.57530 0.5654\n",
       "2 1   rs3094315   742429 498    0.306700 0.5377 6.555e-04  0.57040 0.5687\n",
       "3 1   rs12562034  758311 495    0.793000 0.8845 1.628e-03  0.89650 0.3704\n",
       "4 1   rs3934834   995669 498   -0.007573 0.6342 2.875e-07 -0.01194 0.9905\n",
       "5 1   rs3737728  1011278 498    0.465700 0.6392 1.069e-03  0.72860 0.4666\n",
       "6 1   rs4970405  1038818 498   -0.829400 0.7574 2.412e-03 -1.09500 0.2740"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "1.1024932733801"
      ],
      "text/latex": [
       "1.1024932733801"
      ],
      "text/markdown": [
       "1.1024932733801"
      ],
      "text/plain": [
       "[1] 1.102493"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "res1<-read.table(\"plinkresults.qassoc\", header=T)\n",
    "head(res1) \n",
    "source(\"qqmanHJCupdated.R\") \n",
    "\n",
    "png(\"mh1.png\")\n",
    "manhattan(res1, pch=20, suggestiveline=F, genomewideline=F, ymin=2,\n",
    "cex.x.axis=0.65, colors=c(\"black\",\"dodgerblue\"), cex=0.5) \n",
    "dev.off() \n",
    "\n",
    "png(\"qq1.png\")\n",
    "qq(res1$P)\n",
    "dev.off() \n",
    "\n",
    "chi<-(qchisq(1-res1$P,1))\n",
    "lambda=median(chi)/0.456\n",
    "lambda "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "0b09f927-da19-4056-baf5-71b27c436619",
   "metadata": {
    "kernel": "Bash",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "FastLmmC v2.07.20240410 - Factored Spectrally Transformed Linear Mixed Models [Release]\n",
      "  Copyright Microsoft Corporation -- Licensed Only for Non-Commercial use.\n",
      "  Compiled Apr 10 2024 at 22:40:56 by 129dcc04da00 for Linux v3.x kernel\n",
      "  using 0\n",
      "\n",
      " ++    Start Processing CommandLine:\n",
      " --      End Processing CommandLine:\n",
      "\n",
      " ++      Start Loading FastLmm Data:\n",
      "  ++  Start Loading Covariance Data:\n",
      "   ++      Processing PLINK fileset: [quantfamdata]\n",
      "   ReadBinaryFiles4()* elapsed time: 477.803 ms\n",
      "     Number of Individuals Selected:     498\n",
      "               Number of Phenotypes:       1\n",
      "                Number of SNPs Read:  134946\n",
      "                Number of SNPs Used:  134946\n",
      "   --  End Processing PLINK fileset: [quantfamdata]\n",
      "  --    End Loading Covariance Data:\n",
      "  ++        Start Loading Test Data:\n",
      "   ++      Processing PLINK fileset: [quantfamdata]\n",
      "   ReadBinaryFiles4()* elapsed time: 463.565 ms\n",
      "     Number of Individuals Selected:     498\n",
      "               Number of Phenotypes:       1\n",
      "                Number of SNPs Read:  134946\n",
      "                Number of SNPs Used:  134946\n",
      "   --  End Processing PLINK fileset: [quantfamdata]\n",
      "  --          End Loading Test Data:\n",
      "\n",
      " --        End Loading FastLmm Data:\n",
      "\n",
      "              Compute/Load EigenSym:\n",
      "             Compute GWAS using LMM:\n",
      "                  GWAS elapsed time: 41.234 sec\n",
      "                  Write output file: [/home/jovyan/handson-tutorials/contents/fastlmm_gcta/FLMMresults]\n",
      "                 Total elapsed time: 45.222 sec\n"
     ]
    }
   ],
   "source": [
    "fastlmmc -bfile quantfamdata -pheno quantfamdata.fam -mpheno 4 -bfileSim quantfamdata -ML -out FLMMresults"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8f1b1dd4-8321-4ffb-b518-f09c76c47a1c",
   "metadata": {
    "kernel": "R",
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 17</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>SNP</th><th scope=col>Chromosome</th><th scope=col>GeneticDistance</th><th scope=col>Position</th><th scope=col>Pvalue</th><th scope=col>Qvalue</th><th scope=col>N</th><th scope=col>NullLogLike</th><th scope=col>AltLogLike</th><th scope=col>SNPWeight</th><th scope=col>SNPWeightSE</th><th scope=col>OddsRatio</th><th scope=col>WaldStat</th><th scope=col>NullLogDelta</th><th scope=col>NullGeneticVar</th><th scope=col>NullResidualVar</th><th scope=col>NullBias</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>rs9271209 </td><td> 6</td><td>0</td><td> 32687013</td><td>1.259515e-24</td><td>4.323843e-20</td><td>498</td><td>-1728.182</td><td>-1675.713</td><td>3.497918</td><td>0.3213077</td><td>573.51257</td><td>118.51599</td><td>-0.01280211</td><td>31.17498</td><td>30.77841</td><td>125.3145</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>rs9271252 </td><td> 6</td><td>0</td><td> 32688266</td><td>1.259515e-24</td><td>4.323843e-20</td><td>498</td><td>-1728.182</td><td>-1675.713</td><td>3.497918</td><td>0.3213077</td><td>573.51257</td><td>118.51599</td><td>-0.01280211</td><td>31.17498</td><td>30.77841</td><td>125.3145</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>rs9271255 </td><td> 6</td><td>0</td><td> 32688335</td><td>1.259515e-24</td><td>4.323843e-20</td><td>498</td><td>-1728.182</td><td>-1675.713</td><td>3.497918</td><td>0.3213077</td><td>573.51257</td><td>118.51599</td><td>-0.01280211</td><td>31.17498</td><td>30.77841</td><td>125.3145</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>rs28366284</td><td> 6</td><td>0</td><td> 32668305</td><td>1.281651e-24</td><td>4.323843e-20</td><td>498</td><td>-1728.182</td><td>-1675.730</td><td>3.502775</td><td>0.3212447</td><td>671.93014</td><td>118.89199</td><td>-0.01280211</td><td>31.17498</td><td>30.77841</td><td>125.3145</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>rs233722  </td><td>12</td><td>0</td><td>111515857</td><td>3.800076e-22</td><td>1.025610e-17</td><td>498</td><td>-1728.182</td><td>-1681.366</td><td>3.343238</td><td>0.3288364</td><td>116.06353</td><td>103.36532</td><td>-0.01280211</td><td>31.17498</td><td>30.77841</td><td>125.3145</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>rs233716  </td><td>12</td><td>0</td><td>111524326</td><td>1.682225e-12</td><td>3.783491e-08</td><td>498</td><td>-1728.182</td><td>-1703.270</td><td>2.469427</td><td>0.3405066</td><td> 35.65106</td><td> 52.59462</td><td>-0.01280211</td><td>31.17498</td><td>30.77841</td><td>125.3145</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 17\n",
       "\\begin{tabular}{r|lllllllllllllllll}\n",
       "  & SNP & Chromosome & GeneticDistance & Position & Pvalue & Qvalue & N & NullLogLike & AltLogLike & SNPWeight & SNPWeightSE & OddsRatio & WaldStat & NullLogDelta & NullGeneticVar & NullResidualVar & NullBias\\\\\n",
       "  & <chr> & <int> & <dbl> & <int> & <dbl> & <dbl> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & rs9271209  &  6 & 0 &  32687013 & 1.259515e-24 & 4.323843e-20 & 498 & -1728.182 & -1675.713 & 3.497918 & 0.3213077 & 573.51257 & 118.51599 & -0.01280211 & 31.17498 & 30.77841 & 125.3145\\\\\n",
       "\t2 & rs9271252  &  6 & 0 &  32688266 & 1.259515e-24 & 4.323843e-20 & 498 & -1728.182 & -1675.713 & 3.497918 & 0.3213077 & 573.51257 & 118.51599 & -0.01280211 & 31.17498 & 30.77841 & 125.3145\\\\\n",
       "\t3 & rs9271255  &  6 & 0 &  32688335 & 1.259515e-24 & 4.323843e-20 & 498 & -1728.182 & -1675.713 & 3.497918 & 0.3213077 & 573.51257 & 118.51599 & -0.01280211 & 31.17498 & 30.77841 & 125.3145\\\\\n",
       "\t4 & rs28366284 &  6 & 0 &  32668305 & 1.281651e-24 & 4.323843e-20 & 498 & -1728.182 & -1675.730 & 3.502775 & 0.3212447 & 671.93014 & 118.89199 & -0.01280211 & 31.17498 & 30.77841 & 125.3145\\\\\n",
       "\t5 & rs233722   & 12 & 0 & 111515857 & 3.800076e-22 & 1.025610e-17 & 498 & -1728.182 & -1681.366 & 3.343238 & 0.3288364 & 116.06353 & 103.36532 & -0.01280211 & 31.17498 & 30.77841 & 125.3145\\\\\n",
       "\t6 & rs233716   & 12 & 0 & 111524326 & 1.682225e-12 & 3.783491e-08 & 498 & -1728.182 & -1703.270 & 2.469427 & 0.3405066 &  35.65106 &  52.59462 & -0.01280211 & 31.17498 & 30.77841 & 125.3145\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 17\n",
       "\n",
       "| <!--/--> | SNP &lt;chr&gt; | Chromosome &lt;int&gt; | GeneticDistance &lt;dbl&gt; | Position &lt;int&gt; | Pvalue &lt;dbl&gt; | Qvalue &lt;dbl&gt; | N &lt;int&gt; | NullLogLike &lt;dbl&gt; | AltLogLike &lt;dbl&gt; | SNPWeight &lt;dbl&gt; | SNPWeightSE &lt;dbl&gt; | OddsRatio &lt;dbl&gt; | WaldStat &lt;dbl&gt; | NullLogDelta &lt;dbl&gt; | NullGeneticVar &lt;dbl&gt; | NullResidualVar &lt;dbl&gt; | NullBias &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | rs9271209  |  6 | 0 |  32687013 | 1.259515e-24 | 4.323843e-20 | 498 | -1728.182 | -1675.713 | 3.497918 | 0.3213077 | 573.51257 | 118.51599 | -0.01280211 | 31.17498 | 30.77841 | 125.3145 |\n",
       "| 2 | rs9271252  |  6 | 0 |  32688266 | 1.259515e-24 | 4.323843e-20 | 498 | -1728.182 | -1675.713 | 3.497918 | 0.3213077 | 573.51257 | 118.51599 | -0.01280211 | 31.17498 | 30.77841 | 125.3145 |\n",
       "| 3 | rs9271255  |  6 | 0 |  32688335 | 1.259515e-24 | 4.323843e-20 | 498 | -1728.182 | -1675.713 | 3.497918 | 0.3213077 | 573.51257 | 118.51599 | -0.01280211 | 31.17498 | 30.77841 | 125.3145 |\n",
       "| 4 | rs28366284 |  6 | 0 |  32668305 | 1.281651e-24 | 4.323843e-20 | 498 | -1728.182 | -1675.730 | 3.502775 | 0.3212447 | 671.93014 | 118.89199 | -0.01280211 | 31.17498 | 30.77841 | 125.3145 |\n",
       "| 5 | rs233722   | 12 | 0 | 111515857 | 3.800076e-22 | 1.025610e-17 | 498 | -1728.182 | -1681.366 | 3.343238 | 0.3288364 | 116.06353 | 103.36532 | -0.01280211 | 31.17498 | 30.77841 | 125.3145 |\n",
       "| 6 | rs233716   | 12 | 0 | 111524326 | 1.682225e-12 | 3.783491e-08 | 498 | -1728.182 | -1703.270 | 2.469427 | 0.3405066 |  35.65106 |  52.59462 | -0.01280211 | 31.17498 | 30.77841 | 125.3145 |\n",
       "\n"
      ],
      "text/plain": [
       "  SNP        Chromosome GeneticDistance Position  Pvalue       Qvalue       N  \n",
       "1 rs9271209   6         0                32687013 1.259515e-24 4.323843e-20 498\n",
       "2 rs9271252   6         0                32688266 1.259515e-24 4.323843e-20 498\n",
       "3 rs9271255   6         0                32688335 1.259515e-24 4.323843e-20 498\n",
       "4 rs28366284  6         0                32668305 1.281651e-24 4.323843e-20 498\n",
       "5 rs233722   12         0               111515857 3.800076e-22 1.025610e-17 498\n",
       "6 rs233716   12         0               111524326 1.682225e-12 3.783491e-08 498\n",
       "  NullLogLike AltLogLike SNPWeight SNPWeightSE OddsRatio WaldStat  NullLogDelta\n",
       "1 -1728.182   -1675.713  3.497918  0.3213077   573.51257 118.51599 -0.01280211 \n",
       "2 -1728.182   -1675.713  3.497918  0.3213077   573.51257 118.51599 -0.01280211 \n",
       "3 -1728.182   -1675.713  3.497918  0.3213077   573.51257 118.51599 -0.01280211 \n",
       "4 -1728.182   -1675.730  3.502775  0.3212447   671.93014 118.89199 -0.01280211 \n",
       "5 -1728.182   -1681.366  3.343238  0.3288364   116.06353 103.36532 -0.01280211 \n",
       "6 -1728.182   -1703.270  2.469427  0.3405066    35.65106  52.59462 -0.01280211 \n",
       "  NullGeneticVar NullResidualVar NullBias\n",
       "1 31.17498       30.77841        125.3145\n",
       "2 31.17498       30.77841        125.3145\n",
       "3 31.17498       30.77841        125.3145\n",
       "4 31.17498       30.77841        125.3145\n",
       "5 31.17498       30.77841        125.3145\n",
       "6 31.17498       30.77841        125.3145"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "0.992125686611793"
      ],
      "text/latex": [
       "0.992125686611793"
      ],
      "text/markdown": [
       "0.992125686611793"
      ],
      "text/plain": [
       "[1] 0.9921257"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>SNP</th><th scope=col>CHR</th><th scope=col>BP</th><th scope=col>P</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>rs9271209 </td><td> 6</td><td> 32687013</td><td>1.259515e-24</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>rs9271252 </td><td> 6</td><td> 32688266</td><td>1.259515e-24</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>rs9271255 </td><td> 6</td><td> 32688335</td><td>1.259515e-24</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>rs28366284</td><td> 6</td><td> 32668305</td><td>1.281651e-24</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>rs233722  </td><td>12</td><td>111515857</td><td>3.800076e-22</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>rs233716  </td><td>12</td><td>111524326</td><td>1.682225e-12</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & SNP & CHR & BP & P\\\\\n",
       "  & <chr> & <int> & <int> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & rs9271209  &  6 &  32687013 & 1.259515e-24\\\\\n",
       "\t2 & rs9271252  &  6 &  32688266 & 1.259515e-24\\\\\n",
       "\t3 & rs9271255  &  6 &  32688335 & 1.259515e-24\\\\\n",
       "\t4 & rs28366284 &  6 &  32668305 & 1.281651e-24\\\\\n",
       "\t5 & rs233722   & 12 & 111515857 & 3.800076e-22\\\\\n",
       "\t6 & rs233716   & 12 & 111524326 & 1.682225e-12\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 4\n",
       "\n",
       "| <!--/--> | SNP &lt;chr&gt; | CHR &lt;int&gt; | BP &lt;int&gt; | P &lt;dbl&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 1 | rs9271209  |  6 |  32687013 | 1.259515e-24 |\n",
       "| 2 | rs9271252  |  6 |  32688266 | 1.259515e-24 |\n",
       "| 3 | rs9271255  |  6 |  32688335 | 1.259515e-24 |\n",
       "| 4 | rs28366284 |  6 |  32668305 | 1.281651e-24 |\n",
       "| 5 | rs233722   | 12 | 111515857 | 3.800076e-22 |\n",
       "| 6 | rs233716   | 12 | 111524326 | 1.682225e-12 |\n",
       "\n"
      ],
      "text/plain": [
       "  SNP        CHR BP        P           \n",
       "1 rs9271209   6   32687013 1.259515e-24\n",
       "2 rs9271252   6   32688266 1.259515e-24\n",
       "3 rs9271255   6   32688335 1.259515e-24\n",
       "4 rs28366284  6   32668305 1.281651e-24\n",
       "5 rs233722   12  111515857 3.800076e-22\n",
       "6 rs233716   12  111524326 1.682225e-12"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "res2<-read.table(\"FLMMresults\", header=T) \n",
    "head(res2) \n",
    "\n",
    "chi<-(qchisq(1-res2$Pvalue,1))\n",
    "lambda=median(chi)/0.456\n",
    "lambda \n",
    "\n",
    "new<-data.frame(res2$SNP, res2$Chromosome, res2$Position, res2$Pvalue)\n",
    "names(new)<-c(\"SNP\", \"CHR\", \"BP\", \"P\")\n",
    "head(new) \n",
    "\n",
    "png(\"qq2.png\")\n",
    "qq(new$P)\n",
    "dev.off() \n",
    "\n",
    "png(\"mh2.png\")\n",
    "manhattan(new, pch=20, suggestiveline=F, genomewideline=F, ymin=2,\n",
    "cex.x.axis=0.65, colors=c(\"black\",\"dodgerblue\"), cex=0.5)\n",
    "dev.off() "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dfb9acf9-a383-4676-83a2-269e83c280d4",
   "metadata": {
    "kernel": "R"
   },
   "source": [
    "# GCTA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "b8e1dc84-7888-4d0a-9440-ef8aea4ce1a7",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "fam<-read.table(\"quantfamdata.fam\", header=F)\n",
    "pheno=data.frame(fam[,1:2],fam[,6])\n",
    "write.table(pheno,file=\"phenos.txt\",col.names=F,row.names=F,quote=F)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "e9b27c42-cc55-4ea8-a53d-58278a6c0e6f",
   "metadata": {
    "kernel": "Bash",
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 00:37:51 UTC on Fri Apr 19 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--mlma \n",
      "--bfile quantfamdata\n",
      "--pheno phenos.txt\n",
      "--out GCTAresults\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading PLINK FAM file from [quantfamdata.fam].\n",
      "498 individuals to be included from [quantfamdata.fam].\n",
      "Reading PLINK BIM file from [quantfamdata.bim].\n",
      "134946 SNPs to be included from [quantfamdata.bim].\n",
      "Reading PLINK BED file from [quantfamdata.bed] in SNP-major format ...\n",
      "Genotype data for 498 individuals and 134946 SNPs to be included from [quantfamdata.bed].\n",
      "Reading phenotypes from [phenos.txt].\n",
      "Non-missing phenotypes of 498 individuals are included from [phenos.txt].\n",
      "Calculating allele frequencies ...\n",
      "Recoding genotypes (individual major mode) ...\n",
      "\n",
      "Calculating the genetic relationship matrix (GRM) ... \n",
      "498 individuals are in common in these files.\n",
      "\n",
      "Performing MLM association analyses (including the candidate SNP) ...\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "498 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 31.0444 31.1001\n",
      "logL: -1272.01\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-1272.00\t30.59135\t31.46507\t\n",
      "2\t-1272.00\t29.65725\t32.21513\t\n",
      "3\t-1271.99\t29.74705\t32.13462\t\n",
      "4\t-1271.99\t29.73839\t32.14262\t\n",
      "5\t-1271.99\t29.73924\t32.14184\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Running association tests for 134946 SNPs ...\n",
      "\n",
      "Saving the results of the mixed linear model association analyses of 134946 SNPs to [GCTAresults.mlma] ...\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 00:39:22 UTC on Fri Apr 19 2024\n",
      "Overall computational time: 1 minute 31 sec.\n"
     ]
    }
   ],
   "source": [
    "gcta64 --mlma --bfile quantfamdata --pheno phenos.txt --out GCTAresults "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "5e6b4b6c-db67-46dd-b2b3-d2b579b6eb9e",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "source(\"qqmanHJCupdated.R\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "43693225-e7ec-4ebc-a678-e99d75bc11c3",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Chr</th><th scope=col>SNP</th><th scope=col>bp</th><th scope=col>A1</th><th scope=col>A2</th><th scope=col>Freq</th><th scope=col>b</th><th scope=col>se</th><th scope=col>p</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>rs9326624 </td><td> 558615</td><td>G</td><td>A</td><td>0.0913655</td><td> 0.693811</td><td>0.924665</td><td>0.453052</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1</td><td>rs3094315 </td><td> 742429</td><td>G</td><td>A</td><td>0.3162650</td><td> 0.494021</td><td>0.550792</td><td>0.369758</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1</td><td>rs12562034</td><td> 758311</td><td>A</td><td>G</td><td>0.0818182</td><td> 0.827582</td><td>0.882891</td><td>0.348576</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1</td><td>rs3934834 </td><td> 995669</td><td>A</td><td>G</td><td>0.1817270</td><td>-0.121518</td><td>0.640661</td><td>0.849563</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1</td><td>rs3737728 </td><td>1011278</td><td>A</td><td>G</td><td>0.1957830</td><td> 0.513895</td><td>0.644486</td><td>0.425236</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1</td><td>rs4970405 </td><td>1038818</td><td>G</td><td>A</td><td>0.1194780</td><td>-0.697621</td><td>0.764350</td><td>0.361401</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 9\n",
       "\\begin{tabular}{r|lllllllll}\n",
       "  & Chr & SNP & bp & A1 & A2 & Freq & b & se & p\\\\\n",
       "  & <int> & <chr> & <int> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & rs9326624  &  558615 & G & A & 0.0913655 &  0.693811 & 0.924665 & 0.453052\\\\\n",
       "\t2 & 1 & rs3094315  &  742429 & G & A & 0.3162650 &  0.494021 & 0.550792 & 0.369758\\\\\n",
       "\t3 & 1 & rs12562034 &  758311 & A & G & 0.0818182 &  0.827582 & 0.882891 & 0.348576\\\\\n",
       "\t4 & 1 & rs3934834  &  995669 & A & G & 0.1817270 & -0.121518 & 0.640661 & 0.849563\\\\\n",
       "\t5 & 1 & rs3737728  & 1011278 & A & G & 0.1957830 &  0.513895 & 0.644486 & 0.425236\\\\\n",
       "\t6 & 1 & rs4970405  & 1038818 & G & A & 0.1194780 & -0.697621 & 0.764350 & 0.361401\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 9\n",
       "\n",
       "| <!--/--> | Chr &lt;int&gt; | SNP &lt;chr&gt; | bp &lt;int&gt; | A1 &lt;chr&gt; | A2 &lt;chr&gt; | Freq &lt;dbl&gt; | b &lt;dbl&gt; | se &lt;dbl&gt; | p &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | rs9326624  |  558615 | G | A | 0.0913655 |  0.693811 | 0.924665 | 0.453052 |\n",
       "| 2 | 1 | rs3094315  |  742429 | G | A | 0.3162650 |  0.494021 | 0.550792 | 0.369758 |\n",
       "| 3 | 1 | rs12562034 |  758311 | A | G | 0.0818182 |  0.827582 | 0.882891 | 0.348576 |\n",
       "| 4 | 1 | rs3934834  |  995669 | A | G | 0.1817270 | -0.121518 | 0.640661 | 0.849563 |\n",
       "| 5 | 1 | rs3737728  | 1011278 | A | G | 0.1957830 |  0.513895 | 0.644486 | 0.425236 |\n",
       "| 6 | 1 | rs4970405  | 1038818 | G | A | 0.1194780 | -0.697621 | 0.764350 | 0.361401 |\n",
       "\n"
      ],
      "text/plain": [
       "  Chr SNP        bp      A1 A2 Freq      b         se       p       \n",
       "1 1   rs9326624   558615 G  A  0.0913655  0.693811 0.924665 0.453052\n",
       "2 1   rs3094315   742429 G  A  0.3162650  0.494021 0.550792 0.369758\n",
       "3 1   rs12562034  758311 A  G  0.0818182  0.827582 0.882891 0.348576\n",
       "4 1   rs3934834   995669 A  G  0.1817270 -0.121518 0.640661 0.849563\n",
       "5 1   rs3737728  1011278 A  G  0.1957830  0.513895 0.644486 0.425236\n",
       "6 1   rs4970405  1038818 G  A  0.1194780 -0.697621 0.764350 0.361401"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "res3<-read.table(\"GCTAresults.mlma\", header=T)\n",
    "head(res3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "afae027c-85fc-4e3f-a3ca-2533fb0576ac",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "chi<-(qchisq(1-res3$p,1))\n",
    "lambda=median(chi)/0.456\n",
    "lambda"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "31ccc85e-9611-4f58-afb4-b27ad7b2c0d5",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "new3<-data.frame(res3$SNP, res3$Chr, res3$bp, res3$p)\n",
    "names(new3)<-c(\"SNP\", \"CHR\", \"BP\", \"P\")\n",
    "head(new3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "003f4982-97f5-4458-88f4-116e0950b94a",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "png(\"qq3.png\")\n",
    "qq(new3$P)\n",
    "dev.off() \n",
    "\n",
    "png(\"mh3.png\")\n",
    "manhattan(new3, pch=20, suggestiveline=F, genomewideline=F, ymin=2,\n",
    "cex.x.axis=0.65, colors=c(\"black\",\"dodgerblue\"), cex=0.5)\n",
    "dev.off() \n",
    "\n",
    "res2<-read.table(\"FLMMresults\", header=T)\n",
    "new2<-data.frame(res2$SNP, res2$Chromosome, res2$Position, res2$Pvalue)\n",
    "names(new2)<-c(\"SNP\", \"CHR\", \"BP\", \"P\")\n",
    "merged=merge(new3,new2, by=\"SNP\", sort=F)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8326d764-b291-4bf0-a818-24813f685bef",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "head(res2)\n",
    "head(new2)\n",
    "head(new3)\n",
    "head(merged)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "669ff7bb-e517-48fb-8a15-d23aeda29180",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "png(\"compareGCTAFLMM.png\")\n",
    "plot(-log10(merged$P.x),-log10(merged$P.y))\n",
    "abline(0,1)\n",
    "dev.off() "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "02286ef4-afca-4f14-b654-6f84faa83fe9",
   "metadata": {
    "kernel": "Bash",
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:36 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile quantfamdata\n",
      "--autosome\n",
      "--make-grm-bin\n",
      "--out GCTAgrm\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading PLINK FAM file from [quantfamdata.fam].\n",
      "498 individuals to be included from [quantfamdata.fam].\n",
      "Reading PLINK BIM file from [quantfamdata.bim].\n",
      "134946 SNPs to be included from [quantfamdata.bim].\n",
      "134946 SNPs from chromosome 1 to chromosome 22 are included in the analysis.\n",
      "Reading PLINK BED file from [quantfamdata.bed] in SNP-major format ...\n",
      "Genotype data for 498 individuals and 134946 SNPs to be included from [quantfamdata.bed].\n",
      "Calculating allele frequencies ...\n",
      "Recoding genotypes (individual major mode) ...\n",
      "\n",
      "Calculating the genetic relationship matrix (GRM) ... (Note: default speed-optimized mode, may use huge RAM)\n",
      "\n",
      "Summary of the GRM:\n",
      "Mean of diagonals = 1.00289\n",
      "Variance of diagonals = 0.000422078\n",
      "Mean of off-diagonals = -0.00201898\n",
      "Variance of off-diagonals = 0.0004337\n",
      "GRM of 498 individuals has been saved in the file [GCTAgrm.grm.bin] (in binary format).\n",
      "Number of SNPs to calculate the genetic relationship between each pair of individuals has been saved in the file [GCTAgrm.grm.N.bin] (in binary format).\n",
      "IDs for the GRM file [GCTAgrm.grm.bin] have been saved in the file [GCTAgrm.grm.id].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:41 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 5.57 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:41 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--reml\n",
      "--grm-bin GCTAgrm\n",
      "--pheno phenos.txt\n",
      "--out GCTAherit\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading IDs of the GRM from [GCTAgrm.grm.id].\n",
      "498 IDs are read from [GCTAgrm.grm.id].\n",
      "Reading the GRM from [GCTAgrm.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrm.grm.bin].\n",
      "Reading phenotypes from [phenos.txt].\n",
      "Non-missing phenotypes of 498 individuals are included from [phenos.txt].\n",
      "\n",
      "498 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "498 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 31.0442    31.1\n",
      "logL: -1272.01\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-1272.00\t30.59119\t31.46479\t\n",
      "2\t-1272.00\t29.65710\t32.21464\t\n",
      "3\t-1271.99\t29.74689\t32.13413\t\n",
      "4\t-1271.99\t29.73823\t32.14213\t\n",
      "5\t-1271.99\t29.73908\t32.14135\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 62.30497\n",
      "logL: -1278.41752\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(e)\t\n",
      "1\t-1278.42\t62.30497\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G)\t29.739081\t9.571996\n",
      "V(e)\t32.141349\t8.886473\n",
      "Vp\t61.880430\t3.993645\n",
      "V(G)/Vp\t0.480589\t0.145944\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "9.162310e+01\t-7.732165e+01\t\n",
      "-7.732165e+01\t7.896940e+01\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [GCTAherit.hsq].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:41 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.07 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:42 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile quantfamdata\n",
      "--chr 1\n",
      "--make-grm-bin\n",
      "--out GCTAgrmchr1\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading PLINK FAM file from [quantfamdata.fam].\n",
      "498 individuals to be included from [quantfamdata.fam].\n",
      "Reading PLINK BIM file from [quantfamdata.bim].\n",
      "134946 SNPs to be included from [quantfamdata.bim].\n",
      "9003 SNPs on chromosome 1 are included in the analysis.\n",
      "Reading PLINK BED file from [quantfamdata.bed] in SNP-major format ...\n",
      "Genotype data for 498 individuals and 9003 SNPs to be included from [quantfamdata.bed].\n",
      "Calculating allele frequencies ...\n",
      "Recoding genotypes (individual major mode) ...\n",
      "\n",
      "Calculating the genetic relationship matrix (GRM) ... (Note: default speed-optimized mode, may use huge RAM)\n",
      "\n",
      "Summary of the GRM:\n",
      "Mean of diagonals = 1.00293\n",
      "Variance of diagonals = 0.0015773\n",
      "Mean of off-diagonals = -0.0020191\n",
      "Variance of off-diagonals = 0.000981762\n",
      "GRM of 498 individuals has been saved in the file [GCTAgrmchr1.grm.bin] (in binary format).\n",
      "Number of SNPs to calculate the genetic relationship between each pair of individuals has been saved in the file [GCTAgrmchr1.grm.N.bin] (in binary format).\n",
      "IDs for the GRM file [GCTAgrmchr1.grm.bin] have been saved in the file [GCTAgrmchr1.grm.id].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:42 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.77 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:42 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--reml\n",
      "--grm-bin GCTAgrmchr1\n",
      "--pheno phenos.txt\n",
      "--out GCTAheritchr1\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading IDs of the GRM from [GCTAgrmchr1.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr1.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr1.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr1.grm.bin].\n",
      "Reading phenotypes from [phenos.txt].\n",
      "Non-missing phenotypes of 498 individuals are included from [phenos.txt].\n",
      "\n",
      "498 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "498 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 31.1649 32.7147\n",
      "logL: -1280.1\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-1279.59\t25.51133\t37.48693\t\n",
      "2\t-1278.14\t13.16936\t48.55725\t\n",
      "3\t-1276.74\t13.08769\t48.83872\t\n",
      "4\t-1276.74\t13.02830\t49.03582\t\n",
      "5\t-1276.73\t12.89243\t49.47117\t\n",
      "6\t-1276.73\t12.87787\t49.48643\t\n",
      "7\t-1276.73\t12.87689\t49.48734\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 62.30497\n",
      "logL: -1278.41752\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(e)\t\n",
      "1\t-1278.42\t62.30497\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G)\t12.876888\t7.277044\n",
      "V(e)\t49.487343\t7.395969\n",
      "Vp\t62.364231\t3.992076\n",
      "V(G)/Vp\t0.206479\t0.114179\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "5.295537e+01\t-4.585953e+01\t\n",
      "-4.585953e+01\t5.470036e+01\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [GCTAheritchr1.hsq].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:43 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.08 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:43 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile quantfamdata\n",
      "--chr 2\n",
      "--make-grm-bin\n",
      "--out GCTAgrmchr2\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading PLINK FAM file from [quantfamdata.fam].\n",
      "498 individuals to be included from [quantfamdata.fam].\n",
      "Reading PLINK BIM file from [quantfamdata.bim].\n",
      "134946 SNPs to be included from [quantfamdata.bim].\n",
      "9721 SNPs on chromosome 2 are included in the analysis.\n",
      "Reading PLINK BED file from [quantfamdata.bed] in SNP-major format ...\n",
      "Genotype data for 498 individuals and 9721 SNPs to be included from [quantfamdata.bed].\n",
      "Calculating allele frequencies ...\n",
      "Recoding genotypes (individual major mode) ...\n",
      "\n",
      "Calculating the genetic relationship matrix (GRM) ... (Note: default speed-optimized mode, may use huge RAM)\n",
      "\n",
      "Summary of the GRM:\n",
      "Mean of diagonals = 1.00164\n",
      "Variance of diagonals = 0.00203532\n",
      "Mean of off-diagonals = -0.00201645\n",
      "Variance of off-diagonals = 0.00104589\n",
      "GRM of 498 individuals has been saved in the file [GCTAgrmchr2.grm.bin] (in binary format).\n",
      "Number of SNPs to calculate the genetic relationship between each pair of individuals has been saved in the file [GCTAgrmchr2.grm.N.bin] (in binary format).\n",
      "IDs for the GRM file [GCTAgrmchr2.grm.bin] have been saved in the file [GCTAgrmchr2.grm.id].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:43 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.76 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:44 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--reml\n",
      "--grm-bin GCTAgrmchr2\n",
      "--pheno phenos.txt\n",
      "--out GCTAheritchr2\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading IDs of the GRM from [GCTAgrmchr2.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr2.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr2.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr2.grm.bin].\n",
      "Reading phenotypes from [phenos.txt].\n",
      "Non-missing phenotypes of 498 individuals are included from [phenos.txt].\n",
      "\n",
      "498 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "498 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values:   31.1 33.381\n",
      "logL: -1284.45\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-1283.43\t22.49448\t40.29567\t\n",
      "2\t-1280.34\t16.97034\t45.18315\t\n",
      "3\t-1278.89\t13.46291\t48.51428\t\n",
      "4\t-1278.24\t11.22354\t50.75247\t\n",
      "5\t-1277.95\t6.64600\t55.49657\t\n",
      "6\t-1277.71\t6.94925\t55.33098\t\n",
      "7\t-1277.71\t6.94493\t55.33588\t\n",
      "8\t-1277.71\t6.94504\t55.33577\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 62.30497\n",
      "logL: -1278.41752\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(e)\t\n",
      "1\t-1278.42\t62.30497\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G)\t6.945039\t6.391182\n",
      "V(e)\t55.335774\t7.029114\n",
      "Vp\t62.280813\t3.961202\n",
      "V(G)/Vp\t0.111512\t0.101862\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "4.084720e+01\t-3.728226e+01\t\n",
      "-3.728226e+01\t4.940844e+01\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [GCTAheritchr2.hsq].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:44 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.09 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:44 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile quantfamdata\n",
      "--chr 6\n",
      "--make-grm-bin\n",
      "--out GCTAgrmchr6\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading PLINK FAM file from [quantfamdata.fam].\n",
      "498 individuals to be included from [quantfamdata.fam].\n",
      "Reading PLINK BIM file from [quantfamdata.bim].\n",
      "134946 SNPs to be included from [quantfamdata.bim].\n",
      "19811 SNPs on chromosome 6 are included in the analysis.\n",
      "Reading PLINK BED file from [quantfamdata.bed] in SNP-major format ...\n",
      "Genotype data for 498 individuals and 19811 SNPs to be included from [quantfamdata.bed].\n",
      "Calculating allele frequencies ...\n",
      "Recoding genotypes (individual major mode) ...\n",
      "\n",
      "Calculating the genetic relationship matrix (GRM) ... (Note: default speed-optimized mode, may use huge RAM)\n",
      "\n",
      "Summary of the GRM:\n",
      "Mean of diagonals = 1.00362\n",
      "Variance of diagonals = 0.00196349\n",
      "Mean of off-diagonals = -0.00202046\n",
      "Variance of off-diagonals = 0.00121764\n",
      "GRM of 498 individuals has been saved in the file [GCTAgrmchr6.grm.bin] (in binary format).\n",
      "Number of SNPs to calculate the genetic relationship between each pair of individuals has been saved in the file [GCTAgrmchr6.grm.N.bin] (in binary format).\n",
      "IDs for the GRM file [GCTAgrmchr6.grm.bin] have been saved in the file [GCTAgrmchr6.grm.id].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:45 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 1.10 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:45 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--reml\n",
      "--grm-bin GCTAgrmchr6\n",
      "--pheno phenos.txt\n",
      "--out GCTAheritchr6\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading IDs of the GRM from [GCTAgrmchr6.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr6.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr6.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr6.grm.bin].\n",
      "Reading phenotypes from [phenos.txt].\n",
      "Non-missing phenotypes of 498 individuals are included from [phenos.txt].\n",
      "\n",
      "498 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "498 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 31.0069  31.947\n",
      "logL: -1271.19\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-1271.04\t28.12821\t34.35332\t\n",
      "2\t-1270.59\t22.24402\t39.49030\t\n",
      "3\t-1270.25\t22.86422\t39.12826\t\n",
      "4\t-1270.25\t22.81962\t39.16883\t\n",
      "5\t-1270.25\t22.82373\t39.16530\t\n",
      "6\t-1270.25\t22.82336\t39.16562\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 62.30497\n",
      "logL: -1278.41752\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(e)\t\n",
      "1\t-1278.42\t62.30497\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G)\t22.823361\t6.974968\n",
      "V(e)\t39.165618\t6.391105\n",
      "Vp\t61.988979\t4.041888\n",
      "V(G)/Vp\t0.368184\t0.104517\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "4.865018e+01\t-3.657978e+01\t\n",
      "-3.657978e+01\t4.084623e+01\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [GCTAheritchr6.hsq].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:45 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.08 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:45 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile quantfamdata\n",
      "--chr 12\n",
      "--make-grm-bin\n",
      "--out GCTAgrmchr12\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading PLINK FAM file from [quantfamdata.fam].\n",
      "498 individuals to be included from [quantfamdata.fam].\n",
      "Reading PLINK BIM file from [quantfamdata.bim].\n",
      "134946 SNPs to be included from [quantfamdata.bim].\n",
      "18385 SNPs on chromosome 12 are included in the analysis.\n",
      "Reading PLINK BED file from [quantfamdata.bed] in SNP-major format ...\n",
      "Genotype data for 498 individuals and 18385 SNPs to be included from [quantfamdata.bed].\n",
      "Calculating allele frequencies ...\n",
      "Recoding genotypes (individual major mode) ...\n",
      "\n",
      "Calculating the genetic relationship matrix (GRM) ... (Note: default speed-optimized mode, may use huge RAM)\n",
      "\n",
      "Summary of the GRM:\n",
      "Mean of diagonals = 1.00351\n",
      "Variance of diagonals = 0.00188967\n",
      "Mean of off-diagonals = -0.00202017\n",
      "Variance of off-diagonals = 0.00122423\n",
      "GRM of 498 individuals has been saved in the file [GCTAgrmchr12.grm.bin] (in binary format).\n",
      "Number of SNPs to calculate the genetic relationship between each pair of individuals has been saved in the file [GCTAgrmchr12.grm.N.bin] (in binary format).\n",
      "IDs for the GRM file [GCTAgrmchr12.grm.bin] have been saved in the file [GCTAgrmchr12.grm.id].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:46 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 1.07 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:46 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--reml\n",
      "--grm-bin GCTAgrmchr12\n",
      "--pheno phenos.txt\n",
      "--out GCTAheritchr12\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading IDs of the GRM from [GCTAgrmchr12.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr12.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr12.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr12.grm.bin].\n",
      "Reading phenotypes from [phenos.txt].\n",
      "Non-missing phenotypes of 498 individuals are included from [phenos.txt].\n",
      "\n",
      "498 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "498 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 31.0486 32.5394\n",
      "logL: -1276.3\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-1275.89\t26.47687\t36.32801\t\n",
      "2\t-1274.80\t17.05759\t44.64974\t\n",
      "3\t-1273.93\t17.31113\t44.56458\t\n",
      "4\t-1273.92\t17.85560\t44.38722\t\n",
      "5\t-1273.91\t17.83593\t44.40740\t\n",
      "6\t-1273.91\t17.83713\t44.40634\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 62.30497\n",
      "logL: -1278.41752\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(e)\t\n",
      "1\t-1278.42\t62.30497\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G)\t17.837131\t6.876731\n",
      "V(e)\t44.406341\t6.655022\n",
      "Vp\t62.243472\t4.019753\n",
      "V(G)/Vp\t0.286570\t0.105506\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "4.728942e+01\t-3.771016e+01\t\n",
      "-3.771016e+01\t4.428931e+01\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [GCTAheritchr12.hsq].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:46 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.08 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:46 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile quantfamdata\n",
      "--chr 3\n",
      "--make-grm-bin\n",
      "--out GCTAgrmchr3\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading PLINK FAM file from [quantfamdata.fam].\n",
      "498 individuals to be included from [quantfamdata.fam].\n",
      "Reading PLINK BIM file from [quantfamdata.bim].\n",
      "134946 SNPs to be included from [quantfamdata.bim].\n",
      "8040 SNPs on chromosome 3 are included in the analysis.\n",
      "Reading PLINK BED file from [quantfamdata.bed] in SNP-major format ...\n",
      "Genotype data for 498 individuals and 8040 SNPs to be included from [quantfamdata.bed].\n",
      "Calculating allele frequencies ...\n",
      "Recoding genotypes (individual major mode) ...\n",
      "\n",
      "Calculating the genetic relationship matrix (GRM) ... (Note: default speed-optimized mode, may use huge RAM)\n",
      "\n",
      "Summary of the GRM:\n",
      "Mean of diagonals = 1.00364\n",
      "Variance of diagonals = 0.00208634\n",
      "Mean of off-diagonals = -0.00202046\n",
      "Variance of off-diagonals = 0.00115896\n",
      "GRM of 498 individuals has been saved in the file [GCTAgrmchr3.grm.bin] (in binary format).\n",
      "Number of SNPs to calculate the genetic relationship between each pair of individuals has been saved in the file [GCTAgrmchr3.grm.N.bin] (in binary format).\n",
      "IDs for the GRM file [GCTAgrmchr3.grm.bin] have been saved in the file [GCTAgrmchr3.grm.id].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:47 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.73 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:47 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile quantfamdata\n",
      "--chr 4\n",
      "--make-grm-bin\n",
      "--out GCTAgrmchr4\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading PLINK FAM file from [quantfamdata.fam].\n",
      "498 individuals to be included from [quantfamdata.fam].\n",
      "Reading PLINK BIM file from [quantfamdata.bim].\n",
      "134946 SNPs to be included from [quantfamdata.bim].\n",
      "7734 SNPs on chromosome 4 are included in the analysis.\n",
      "Reading PLINK BED file from [quantfamdata.bed] in SNP-major format ...\n",
      "Genotype data for 498 individuals and 7734 SNPs to be included from [quantfamdata.bed].\n",
      "Calculating allele frequencies ...\n",
      "Recoding genotypes (individual major mode) ...\n",
      "\n",
      "Calculating the genetic relationship matrix (GRM) ... (Note: default speed-optimized mode, may use huge RAM)\n",
      "\n",
      "Summary of the GRM:\n",
      "Mean of diagonals = 1.00124\n",
      "Variance of diagonals = 0.00252032\n",
      "Mean of off-diagonals = -0.00201568\n",
      "Variance of off-diagonals = 0.00115717\n",
      "GRM of 498 individuals has been saved in the file [GCTAgrmchr4.grm.bin] (in binary format).\n",
      "Number of SNPs to calculate the genetic relationship between each pair of individuals has been saved in the file [GCTAgrmchr4.grm.N.bin] (in binary format).\n",
      "IDs for the GRM file [GCTAgrmchr4.grm.bin] have been saved in the file [GCTAgrmchr4.grm.id].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:48 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.72 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:48 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile quantfamdata\n",
      "--chr 5\n",
      "--make-grm-bin\n",
      "--out GCTAgrmchr5\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading PLINK FAM file from [quantfamdata.fam].\n",
      "498 individuals to be included from [quantfamdata.fam].\n",
      "Reading PLINK BIM file from [quantfamdata.bim].\n",
      "134946 SNPs to be included from [quantfamdata.bim].\n",
      "7263 SNPs on chromosome 5 are included in the analysis.\n",
      "Reading PLINK BED file from [quantfamdata.bed] in SNP-major format ...\n",
      "Genotype data for 498 individuals and 7263 SNPs to be included from [quantfamdata.bed].\n",
      "Calculating allele frequencies ...\n",
      "Recoding genotypes (individual major mode) ...\n",
      "\n",
      "Calculating the genetic relationship matrix (GRM) ... (Note: default speed-optimized mode, may use huge RAM)\n",
      "\n",
      "Summary of the GRM:\n",
      "Mean of diagonals = 1.00078\n",
      "Variance of diagonals = 0.00155625\n",
      "Mean of off-diagonals = -0.00201473\n",
      "Variance of off-diagonals = 0.00115265\n",
      "GRM of 498 individuals has been saved in the file [GCTAgrmchr5.grm.bin] (in binary format).\n",
      "Number of SNPs to calculate the genetic relationship between each pair of individuals has been saved in the file [GCTAgrmchr5.grm.N.bin] (in binary format).\n",
      "IDs for the GRM file [GCTAgrmchr5.grm.bin] have been saved in the file [GCTAgrmchr5.grm.id].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:49 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.70 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:49 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile quantfamdata\n",
      "--chr 7\n",
      "--make-grm-bin\n",
      "--out GCTAgrmchr7\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading PLINK FAM file from [quantfamdata.fam].\n",
      "498 individuals to be included from [quantfamdata.fam].\n",
      "Reading PLINK BIM file from [quantfamdata.bim].\n",
      "134946 SNPs to be included from [quantfamdata.bim].\n",
      "6255 SNPs on chromosome 7 are included in the analysis.\n",
      "Reading PLINK BED file from [quantfamdata.bed] in SNP-major format ...\n",
      "Genotype data for 498 individuals and 6255 SNPs to be included from [quantfamdata.bed].\n",
      "Calculating allele frequencies ...\n",
      "Recoding genotypes (individual major mode) ...\n",
      "\n",
      "Calculating the genetic relationship matrix (GRM) ... (Note: default speed-optimized mode, may use huge RAM)\n",
      "\n",
      "Summary of the GRM:\n",
      "Mean of diagonals = 1.00317\n",
      "Variance of diagonals = 0.00207579\n",
      "Mean of off-diagonals = -0.00201959\n",
      "Variance of off-diagonals = 0.00119177\n",
      "GRM of 498 individuals has been saved in the file [GCTAgrmchr7.grm.bin] (in binary format).\n",
      "Number of SNPs to calculate the genetic relationship between each pair of individuals has been saved in the file [GCTAgrmchr7.grm.N.bin] (in binary format).\n",
      "IDs for the GRM file [GCTAgrmchr7.grm.bin] have been saved in the file [GCTAgrmchr7.grm.id].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:49 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.67 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:50 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile quantfamdata\n",
      "--chr 8\n",
      "--make-grm-bin\n",
      "--out GCTAgrmchr8\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading PLINK FAM file from [quantfamdata.fam].\n",
      "498 individuals to be included from [quantfamdata.fam].\n",
      "Reading PLINK BIM file from [quantfamdata.bim].\n",
      "134946 SNPs to be included from [quantfamdata.bim].\n",
      "5849 SNPs on chromosome 8 are included in the analysis.\n",
      "Reading PLINK BED file from [quantfamdata.bed] in SNP-major format ...\n",
      "Genotype data for 498 individuals and 5849 SNPs to be included from [quantfamdata.bed].\n",
      "Calculating allele frequencies ...\n",
      "Recoding genotypes (individual major mode) ...\n",
      "\n",
      "Calculating the genetic relationship matrix (GRM) ... (Note: default speed-optimized mode, may use huge RAM)\n",
      "\n",
      "Summary of the GRM:\n",
      "Mean of diagonals = 1.00401\n",
      "Variance of diagonals = 0.00353961\n",
      "Mean of off-diagonals = -0.00202128\n",
      "Variance of off-diagonals = 0.00162606\n",
      "GRM of 498 individuals has been saved in the file [GCTAgrmchr8.grm.bin] (in binary format).\n",
      "Number of SNPs to calculate the genetic relationship between each pair of individuals has been saved in the file [GCTAgrmchr8.grm.N.bin] (in binary format).\n",
      "IDs for the GRM file [GCTAgrmchr8.grm.bin] have been saved in the file [GCTAgrmchr8.grm.id].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:50 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.65 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:50 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile quantfamdata\n",
      "--chr 9\n",
      "--make-grm-bin\n",
      "--out GCTAgrmchr9\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading PLINK FAM file from [quantfamdata.fam].\n",
      "498 individuals to be included from [quantfamdata.fam].\n",
      "Reading PLINK BIM file from [quantfamdata.bim].\n",
      "134946 SNPs to be included from [quantfamdata.bim].\n",
      "4531 SNPs on chromosome 9 are included in the analysis.\n",
      "Reading PLINK BED file from [quantfamdata.bed] in SNP-major format ...\n",
      "Genotype data for 498 individuals and 4531 SNPs to be included from [quantfamdata.bed].\n",
      "Calculating allele frequencies ...\n",
      "Recoding genotypes (individual major mode) ...\n",
      "\n",
      "Calculating the genetic relationship matrix (GRM) ... (Note: default speed-optimized mode, may use huge RAM)\n",
      "\n",
      "Summary of the GRM:\n",
      "Mean of diagonals = 1.00239\n",
      "Variance of diagonals = 0.00213473\n",
      "Mean of off-diagonals = -0.00201799\n",
      "Variance of off-diagonals = 0.0013214\n",
      "GRM of 498 individuals has been saved in the file [GCTAgrmchr9.grm.bin] (in binary format).\n",
      "Number of SNPs to calculate the genetic relationship between each pair of individuals has been saved in the file [GCTAgrmchr9.grm.N.bin] (in binary format).\n",
      "IDs for the GRM file [GCTAgrmchr9.grm.bin] have been saved in the file [GCTAgrmchr9.grm.id].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:51 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.63 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:51 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile quantfamdata\n",
      "--chr 10\n",
      "--make-grm-bin\n",
      "--out GCTAgrmchr10\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading PLINK FAM file from [quantfamdata.fam].\n",
      "498 individuals to be included from [quantfamdata.fam].\n",
      "Reading PLINK BIM file from [quantfamdata.bim].\n",
      "134946 SNPs to be included from [quantfamdata.bim].\n",
      "5331 SNPs on chromosome 10 are included in the analysis.\n",
      "Reading PLINK BED file from [quantfamdata.bed] in SNP-major format ...\n",
      "Genotype data for 498 individuals and 5331 SNPs to be included from [quantfamdata.bed].\n",
      "Calculating allele frequencies ...\n",
      "Recoding genotypes (individual major mode) ...\n",
      "\n",
      "Calculating the genetic relationship matrix (GRM) ... (Note: default speed-optimized mode, may use huge RAM)\n",
      "\n",
      "Summary of the GRM:\n",
      "Mean of diagonals = 1.00488\n",
      "Variance of diagonals = 0.00260223\n",
      "Mean of off-diagonals = -0.00202298\n",
      "Variance of off-diagonals = 0.00137574\n",
      "GRM of 498 individuals has been saved in the file [GCTAgrmchr10.grm.bin] (in binary format).\n",
      "Number of SNPs to calculate the genetic relationship between each pair of individuals has been saved in the file [GCTAgrmchr10.grm.N.bin] (in binary format).\n",
      "IDs for the GRM file [GCTAgrmchr10.grm.bin] have been saved in the file [GCTAgrmchr10.grm.id].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:52 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.64 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:52 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile quantfamdata\n",
      "--chr 11\n",
      "--make-grm-bin\n",
      "--out GCTAgrmchr11\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading PLINK FAM file from [quantfamdata.fam].\n",
      "498 individuals to be included from [quantfamdata.fam].\n",
      "Reading PLINK BIM file from [quantfamdata.bim].\n",
      "134946 SNPs to be included from [quantfamdata.bim].\n",
      "5401 SNPs on chromosome 11 are included in the analysis.\n",
      "Reading PLINK BED file from [quantfamdata.bed] in SNP-major format ...\n",
      "Genotype data for 498 individuals and 5401 SNPs to be included from [quantfamdata.bed].\n",
      "Calculating allele frequencies ...\n",
      "Recoding genotypes (individual major mode) ...\n",
      "\n",
      "Calculating the genetic relationship matrix (GRM) ... (Note: default speed-optimized mode, may use huge RAM)\n",
      "\n",
      "Summary of the GRM:\n",
      "Mean of diagonals = 1.00379\n",
      "Variance of diagonals = 0.00331638\n",
      "Mean of off-diagonals = -0.00202081\n",
      "Variance of off-diagonals = 0.00152512\n",
      "GRM of 498 individuals has been saved in the file [GCTAgrmchr11.grm.bin] (in binary format).\n",
      "Number of SNPs to calculate the genetic relationship between each pair of individuals has been saved in the file [GCTAgrmchr11.grm.N.bin] (in binary format).\n",
      "IDs for the GRM file [GCTAgrmchr11.grm.bin] have been saved in the file [GCTAgrmchr11.grm.id].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:52 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.66 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:52 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile quantfamdata\n",
      "--chr 13\n",
      "--make-grm-bin\n",
      "--out GCTAgrmchr13\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading PLINK FAM file from [quantfamdata.fam].\n",
      "498 individuals to be included from [quantfamdata.fam].\n",
      "Reading PLINK BIM file from [quantfamdata.bim].\n",
      "134946 SNPs to be included from [quantfamdata.bim].\n",
      "3984 SNPs on chromosome 13 are included in the analysis.\n",
      "Reading PLINK BED file from [quantfamdata.bed] in SNP-major format ...\n",
      "Genotype data for 498 individuals and 3984 SNPs to be included from [quantfamdata.bed].\n",
      "Calculating allele frequencies ...\n",
      "Recoding genotypes (individual major mode) ...\n",
      "\n",
      "Calculating the genetic relationship matrix (GRM) ... (Note: default speed-optimized mode, may use huge RAM)\n",
      "\n",
      "Summary of the GRM:\n",
      "Mean of diagonals = 1.00639\n",
      "Variance of diagonals = 0.00387782\n",
      "Mean of off-diagonals = -0.0020261\n",
      "Variance of off-diagonals = 0.00165767\n",
      "GRM of 498 individuals has been saved in the file [GCTAgrmchr13.grm.bin] (in binary format).\n",
      "Number of SNPs to calculate the genetic relationship between each pair of individuals has been saved in the file [GCTAgrmchr13.grm.N.bin] (in binary format).\n",
      "IDs for the GRM file [GCTAgrmchr13.grm.bin] have been saved in the file [GCTAgrmchr13.grm.id].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:53 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.61 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:53 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile quantfamdata\n",
      "--chr 14\n",
      "--make-grm-bin\n",
      "--out GCTAgrmchr14\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading PLINK FAM file from [quantfamdata.fam].\n",
      "498 individuals to be included from [quantfamdata.fam].\n",
      "Reading PLINK BIM file from [quantfamdata.bim].\n",
      "134946 SNPs to be included from [quantfamdata.bim].\n",
      "3613 SNPs on chromosome 14 are included in the analysis.\n",
      "Reading PLINK BED file from [quantfamdata.bed] in SNP-major format ...\n",
      "Genotype data for 498 individuals and 3613 SNPs to be included from [quantfamdata.bed].\n",
      "Calculating allele frequencies ...\n",
      "Recoding genotypes (individual major mode) ...\n",
      "\n",
      "Calculating the genetic relationship matrix (GRM) ... (Note: default speed-optimized mode, may use huge RAM)\n",
      "\n",
      "Summary of the GRM:\n",
      "Mean of diagonals = 1.00059\n",
      "Variance of diagonals = 0.00343677\n",
      "Mean of off-diagonals = -0.00201443\n",
      "Variance of off-diagonals = 0.00176074\n",
      "GRM of 498 individuals has been saved in the file [GCTAgrmchr14.grm.bin] (in binary format).\n",
      "Number of SNPs to calculate the genetic relationship between each pair of individuals has been saved in the file [GCTAgrmchr14.grm.N.bin] (in binary format).\n",
      "IDs for the GRM file [GCTAgrmchr14.grm.bin] have been saved in the file [GCTAgrmchr14.grm.id].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:54 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.59 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:54 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile quantfamdata\n",
      "--chr 15\n",
      "--make-grm-bin\n",
      "--out GCTAgrmchr15\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading PLINK FAM file from [quantfamdata.fam].\n",
      "498 individuals to be included from [quantfamdata.fam].\n",
      "Reading PLINK BIM file from [quantfamdata.bim].\n",
      "134946 SNPs to be included from [quantfamdata.bim].\n",
      "3215 SNPs on chromosome 15 are included in the analysis.\n",
      "Reading PLINK BED file from [quantfamdata.bed] in SNP-major format ...\n",
      "Genotype data for 498 individuals and 3215 SNPs to be included from [quantfamdata.bed].\n",
      "Calculating allele frequencies ...\n",
      "Recoding genotypes (individual major mode) ...\n",
      "\n",
      "Calculating the genetic relationship matrix (GRM) ... (Note: default speed-optimized mode, may use huge RAM)\n",
      "\n",
      "Summary of the GRM:\n",
      "Mean of diagonals = 1.00121\n",
      "Variance of diagonals = 0.00434469\n",
      "Mean of off-diagonals = -0.00201567\n",
      "Variance of off-diagonals = 0.00202123\n",
      "GRM of 498 individuals has been saved in the file [GCTAgrmchr15.grm.bin] (in binary format).\n",
      "Number of SNPs to calculate the genetic relationship between each pair of individuals has been saved in the file [GCTAgrmchr15.grm.N.bin] (in binary format).\n",
      "IDs for the GRM file [GCTAgrmchr15.grm.bin] have been saved in the file [GCTAgrmchr15.grm.id].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:54 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.61 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:54 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile quantfamdata\n",
      "--chr 16\n",
      "--make-grm-bin\n",
      "--out GCTAgrmchr16\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading PLINK FAM file from [quantfamdata.fam].\n",
      "498 individuals to be included from [quantfamdata.fam].\n",
      "Reading PLINK BIM file from [quantfamdata.bim].\n",
      "134946 SNPs to be included from [quantfamdata.bim].\n",
      "3050 SNPs on chromosome 16 are included in the analysis.\n",
      "Reading PLINK BED file from [quantfamdata.bed] in SNP-major format ...\n",
      "Genotype data for 498 individuals and 3050 SNPs to be included from [quantfamdata.bed].\n",
      "Calculating allele frequencies ...\n",
      "Recoding genotypes (individual major mode) ...\n",
      "\n",
      "Calculating the genetic relationship matrix (GRM) ... (Note: default speed-optimized mode, may use huge RAM)\n",
      "\n",
      "Summary of the GRM:\n",
      "Mean of diagonals = 1.00176\n",
      "Variance of diagonals = 0.00472954\n",
      "Mean of off-diagonals = -0.00201675\n",
      "Variance of off-diagonals = 0.0019014\n",
      "GRM of 498 individuals has been saved in the file [GCTAgrmchr16.grm.bin] (in binary format).\n",
      "Number of SNPs to calculate the genetic relationship between each pair of individuals has been saved in the file [GCTAgrmchr16.grm.N.bin] (in binary format).\n",
      "IDs for the GRM file [GCTAgrmchr16.grm.bin] have been saved in the file [GCTAgrmchr16.grm.id].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:55 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.59 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:55 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile quantfamdata\n",
      "--chr 17\n",
      "--make-grm-bin\n",
      "--out GCTAgrmchr17\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading PLINK FAM file from [quantfamdata.fam].\n",
      "498 individuals to be included from [quantfamdata.fam].\n",
      "Reading PLINK BIM file from [quantfamdata.bim].\n",
      "134946 SNPs to be included from [quantfamdata.bim].\n",
      "3114 SNPs on chromosome 17 are included in the analysis.\n",
      "Reading PLINK BED file from [quantfamdata.bed] in SNP-major format ...\n",
      "Genotype data for 498 individuals and 3114 SNPs to be included from [quantfamdata.bed].\n",
      "Calculating allele frequencies ...\n",
      "Recoding genotypes (individual major mode) ...\n",
      "\n",
      "Calculating the genetic relationship matrix (GRM) ... (Note: default speed-optimized mode, may use huge RAM)\n",
      "\n",
      "Summary of the GRM:\n",
      "Mean of diagonals = 1.00504\n",
      "Variance of diagonals = 0.00443132\n",
      "Mean of off-diagonals = -0.00202336\n",
      "Variance of off-diagonals = 0.0020859\n",
      "GRM of 498 individuals has been saved in the file [GCTAgrmchr17.grm.bin] (in binary format).\n",
      "Number of SNPs to calculate the genetic relationship between each pair of individuals has been saved in the file [GCTAgrmchr17.grm.N.bin] (in binary format).\n",
      "IDs for the GRM file [GCTAgrmchr17.grm.bin] have been saved in the file [GCTAgrmchr17.grm.id].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:55 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.59 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:56 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile quantfamdata\n",
      "--chr 18\n",
      "--make-grm-bin\n",
      "--out GCTAgrmchr18\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading PLINK FAM file from [quantfamdata.fam].\n",
      "498 individuals to be included from [quantfamdata.fam].\n",
      "Reading PLINK BIM file from [quantfamdata.bim].\n",
      "134946 SNPs to be included from [quantfamdata.bim].\n",
      "3100 SNPs on chromosome 18 are included in the analysis.\n",
      "Reading PLINK BED file from [quantfamdata.bed] in SNP-major format ...\n",
      "Genotype data for 498 individuals and 3100 SNPs to be included from [quantfamdata.bed].\n",
      "Calculating allele frequencies ...\n",
      "Recoding genotypes (individual major mode) ...\n",
      "\n",
      "Calculating the genetic relationship matrix (GRM) ... (Note: default speed-optimized mode, may use huge RAM)\n",
      "\n",
      "Summary of the GRM:\n",
      "Mean of diagonals = 0.999809\n",
      "Variance of diagonals = 0.00420447\n",
      "Mean of off-diagonals = -0.00201278\n",
      "Variance of off-diagonals = 0.00181342\n",
      "GRM of 498 individuals has been saved in the file [GCTAgrmchr18.grm.bin] (in binary format).\n",
      "Number of SNPs to calculate the genetic relationship between each pair of individuals has been saved in the file [GCTAgrmchr18.grm.N.bin] (in binary format).\n",
      "IDs for the GRM file [GCTAgrmchr18.grm.bin] have been saved in the file [GCTAgrmchr18.grm.id].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:56 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.59 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:56 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile quantfamdata\n",
      "--chr 19\n",
      "--make-grm-bin\n",
      "--out GCTAgrmchr19\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading PLINK FAM file from [quantfamdata.fam].\n",
      "498 individuals to be included from [quantfamdata.fam].\n",
      "Reading PLINK BIM file from [quantfamdata.bim].\n",
      "134946 SNPs to be included from [quantfamdata.bim].\n",
      "2288 SNPs on chromosome 19 are included in the analysis.\n",
      "Reading PLINK BED file from [quantfamdata.bed] in SNP-major format ...\n",
      "Genotype data for 498 individuals and 2288 SNPs to be included from [quantfamdata.bed].\n",
      "Calculating allele frequencies ...\n",
      "Recoding genotypes (individual major mode) ...\n",
      "\n",
      "Calculating the genetic relationship matrix (GRM) ... (Note: default speed-optimized mode, may use huge RAM)\n",
      "\n",
      "Summary of the GRM:\n",
      "Mean of diagonals = 1.00217\n",
      "Variance of diagonals = 0.00398606\n",
      "Mean of off-diagonals = -0.0020176\n",
      "Variance of off-diagonals = 0.00212117\n",
      "GRM of 498 individuals has been saved in the file [GCTAgrmchr19.grm.bin] (in binary format).\n",
      "Number of SNPs to calculate the genetic relationship between each pair of individuals has been saved in the file [GCTAgrmchr19.grm.N.bin] (in binary format).\n",
      "IDs for the GRM file [GCTAgrmchr19.grm.bin] have been saved in the file [GCTAgrmchr19.grm.id].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:57 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.57 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:57 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile quantfamdata\n",
      "--chr 20\n",
      "--make-grm-bin\n",
      "--out GCTAgrmchr20\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading PLINK FAM file from [quantfamdata.fam].\n",
      "498 individuals to be included from [quantfamdata.fam].\n",
      "Reading PLINK BIM file from [quantfamdata.bim].\n",
      "134946 SNPs to be included from [quantfamdata.bim].\n",
      "2472 SNPs on chromosome 20 are included in the analysis.\n",
      "Reading PLINK BED file from [quantfamdata.bed] in SNP-major format ...\n",
      "Genotype data for 498 individuals and 2472 SNPs to be included from [quantfamdata.bed].\n",
      "Calculating allele frequencies ...\n",
      "Recoding genotypes (individual major mode) ...\n",
      "\n",
      "Calculating the genetic relationship matrix (GRM) ... (Note: default speed-optimized mode, may use huge RAM)\n",
      "\n",
      "Summary of the GRM:\n",
      "Mean of diagonals = 1.00077\n",
      "Variance of diagonals = 0.00422422\n",
      "Mean of off-diagonals = -0.00201468\n",
      "Variance of off-diagonals = 0.00198072\n",
      "GRM of 498 individuals has been saved in the file [GCTAgrmchr20.grm.bin] (in binary format).\n",
      "Number of SNPs to calculate the genetic relationship between each pair of individuals has been saved in the file [GCTAgrmchr20.grm.N.bin] (in binary format).\n",
      "IDs for the GRM file [GCTAgrmchr20.grm.bin] have been saved in the file [GCTAgrmchr20.grm.id].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:57 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.58 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:57 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile quantfamdata\n",
      "--chr 21\n",
      "--make-grm-bin\n",
      "--out GCTAgrmchr21\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading PLINK FAM file from [quantfamdata.fam].\n",
      "498 individuals to be included from [quantfamdata.fam].\n",
      "Reading PLINK BIM file from [quantfamdata.bim].\n",
      "134946 SNPs to be included from [quantfamdata.bim].\n",
      "1402 SNPs on chromosome 21 are included in the analysis.\n",
      "Reading PLINK BED file from [quantfamdata.bed] in SNP-major format ...\n",
      "Genotype data for 498 individuals and 1402 SNPs to be included from [quantfamdata.bed].\n",
      "Calculating allele frequencies ...\n",
      "Recoding genotypes (individual major mode) ...\n",
      "\n",
      "Calculating the genetic relationship matrix (GRM) ... (Note: default speed-optimized mode, may use huge RAM)\n",
      "\n",
      "Summary of the GRM:\n",
      "Mean of diagonals = 1.00425\n",
      "Variance of diagonals = 0.00596999\n",
      "Mean of off-diagonals = -0.00202175\n",
      "Variance of off-diagonals = 0.00282492\n",
      "GRM of 498 individuals has been saved in the file [GCTAgrmchr21.grm.bin] (in binary format).\n",
      "Number of SNPs to calculate the genetic relationship between each pair of individuals has been saved in the file [GCTAgrmchr21.grm.N.bin] (in binary format).\n",
      "IDs for the GRM file [GCTAgrmchr21.grm.bin] have been saved in the file [GCTAgrmchr21.grm.id].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:58 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.55 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:58 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--bfile quantfamdata\n",
      "--chr 22\n",
      "--make-grm-bin\n",
      "--out GCTAgrmchr22\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading PLINK FAM file from [quantfamdata.fam].\n",
      "498 individuals to be included from [quantfamdata.fam].\n",
      "Reading PLINK BIM file from [quantfamdata.bim].\n",
      "134946 SNPs to be included from [quantfamdata.bim].\n",
      "1384 SNPs on chromosome 22 are included in the analysis.\n",
      "Reading PLINK BED file from [quantfamdata.bed] in SNP-major format ...\n",
      "Genotype data for 498 individuals and 1384 SNPs to be included from [quantfamdata.bed].\n",
      "Calculating allele frequencies ...\n",
      "Recoding genotypes (individual major mode) ...\n",
      "\n",
      "Calculating the genetic relationship matrix (GRM) ... (Note: default speed-optimized mode, may use huge RAM)\n",
      "\n",
      "Summary of the GRM:\n",
      "Mean of diagonals = 1.0008\n",
      "Variance of diagonals = 0.00553361\n",
      "Mean of off-diagonals = -0.00201503\n",
      "Variance of off-diagonals = 0.00287841\n",
      "GRM of 498 individuals has been saved in the file [GCTAgrmchr22.grm.bin] (in binary format).\n",
      "Number of SNPs to calculate the genetic relationship between each pair of individuals has been saved in the file [GCTAgrmchr22.grm.N.bin] (in binary format).\n",
      "IDs for the GRM file [GCTAgrmchr22.grm.bin] have been saved in the file [GCTAgrmchr22.grm.id].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:36:59 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.54 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:36:59 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--reml\n",
      "--mgrm-bin multipleGRMs.txt\n",
      "--pheno phenos.txt\n",
      "--out GCTAherit22GRMs\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading phenotypes from [phenos.txt].\n",
      "Non-missing phenotypes of 498 individuals are included from [phenos.txt].\n",
      "\n",
      "There are 22 GRM file names specified in the file [multipleGRMs.txt].\n",
      "Reading the GRM from the 1th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr1.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr1.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr1.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr1.grm.bin].\n",
      "Reading the GRM from the 2th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr2.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr2.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr2.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr2.grm.bin].\n",
      "Reading the GRM from the 3th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr3.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr3.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr3.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr3.grm.bin].\n",
      "Reading the GRM from the 4th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr4.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr4.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr4.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr4.grm.bin].\n",
      "Reading the GRM from the 5th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr5.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr5.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr5.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr5.grm.bin].\n",
      "Reading the GRM from the 6th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr6.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr6.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr6.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr6.grm.bin].\n",
      "Reading the GRM from the 7th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr7.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr7.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr7.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr7.grm.bin].\n",
      "Reading the GRM from the 8th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr8.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr8.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr8.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr8.grm.bin].\n",
      "Reading the GRM from the 9th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr9.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr9.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr9.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr9.grm.bin].\n",
      "Reading the GRM from the 10th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr10.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr10.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr10.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr10.grm.bin].\n",
      "Reading the GRM from the 11th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr11.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr11.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr11.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr11.grm.bin].\n",
      "Reading the GRM from the 12th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr12.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr12.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr12.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr12.grm.bin].\n",
      "Reading the GRM from the 13th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr13.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr13.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr13.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr13.grm.bin].\n",
      "Reading the GRM from the 14th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr14.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr14.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr14.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr14.grm.bin].\n",
      "Reading the GRM from the 15th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr15.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr15.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr15.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr15.grm.bin].\n",
      "Reading the GRM from the 16th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr16.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr16.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr16.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr16.grm.bin].\n",
      "Reading the GRM from the 17th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr17.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr17.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr17.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr17.grm.bin].\n",
      "Reading the GRM from the 18th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr18.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr18.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr18.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr18.grm.bin].\n",
      "Reading the GRM from the 19th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr19.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr19.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr19.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr19.grm.bin].\n",
      "Reading the GRM from the 20th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr20.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr20.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr20.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr20.grm.bin].\n",
      "Reading the GRM from the 21th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr21.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr21.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr21.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr21.grm.bin].\n",
      "Reading the GRM from the 22th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr22.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr22.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr22.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr22.grm.bin].\n",
      "498 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "498 observations, 1 fixed effect(s), and 23 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 2.71953 2.71277   2.721 2.71629 2.71682 2.73715 2.71897 2.71797 2.71361 2.70863 2.72112 2.72162 2.71868 2.71389 2.72692 2.71492 2.71224 2.71791 2.71482 2.71747 2.72969  2.7224 2.73949\n",
      "logL: -1281.27\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G1)\tV(G2)\tV(G3)\tV(G4)\tV(G5)\tV(G6)\tV(G7)\tV(G8)\tV(G9)\tV(G10)\tV(G11)\tV(G12)\tV(G13)\tV(G14)\tV(G15)\tV(G16)\tV(G17)\tV(G18)\tV(G19)\tV(G20)\tV(G21)\tV(G22)\tV(e)\t\n",
      "1\t-1281.14\t2.84059\t0.00006\t3.75356\t1.08364\t1.89877\t6.23121\t2.11322\t2.02344\t0.00006\t0.00006\t2.88306\t3.86816\t2.06500\t0.21725\t4.03387\t2.41132\t0.30125\t2.47529\t1.50639\t2.52558\t3.51700\t3.14924\t10.69192\t(3 component(s) constrained)\n",
      "2\t-1271.78\t2.37459\t0.00006\t3.75599\t0.00006\t0.14495\t14.06521\t0.59620\t1.09872\t0.00006\t0.00006\t1.93045\t5.26929\t1.39634\t0.00006\t5.07602\t2.29080\t0.00006\t1.85024\t0.00006\t1.94652\t3.21564\t2.08259\t29.43937\t(7 component(s) constrained)\n",
      "3\t-1272.31\t0.24220\t0.00006\t3.40311\t0.00006\t0.18785\t15.49376\t0.58287\t1.17920\t0.00006\t0.00006\t2.16614\t5.81268\t0.34688\t0.00006\t5.16094\t1.02993\t0.00006\t0.27826\t0.00006\t0.28252\t3.10296\t1.99864\t31.63126\t(7 component(s) constrained)\n",
      "4\t-1269.75\t0.00006\t0.00006\t2.03792\t0.00006\t0.38364\t17.62258\t0.57156\t0.99734\t0.00006\t0.00006\t2.53135\t6.85490\t3.25692\t0.00006\t5.04119\t3.09077\t0.00006\t1.66850\t0.00006\t1.11196\t2.58837\t1.93610\t34.30378\t(8 component(s) constrained)\n",
      "5\t-1277.65\t0.18220\t0.00006\t2.59760\t0.00006\t0.31778\t17.88512\t0.63172\t1.28094\t0.00006\t0.00006\t2.60569\t6.89340\t0.89929\t0.00006\t5.27981\t1.02945\t0.00006\t1.51994\t0.00006\t0.92639\t2.78834\t1.95803\t35.07728\t(7 component(s) constrained)\n",
      "6\t-1275.80\t0.22102\t0.00006\t2.95269\t0.00006\t0.15052\t17.77251\t0.52178\t1.23470\t0.00006\t0.00006\t2.72643\t7.32411\t3.50996\t0.00006\t5.60771\t3.37864\t0.00006\t1.22454\t0.00006\t0.87657\t3.01691\t2.16861\t35.18061\t(7 component(s) constrained)\n",
      "7\t-1280.74\t0.28785\t0.00006\t3.04144\t0.00006\t0.27019\t18.20071\t0.63898\t1.37743\t0.00006\t0.00006\t2.69249\t7.11865\t0.81318\t0.00006\t5.58635\t0.99569\t0.00006\t1.37753\t0.00006\t0.85585\t3.04801\t2.11363\t35.94228\t(7 component(s) constrained)\n",
      "8\t-1277.70\t0.12971\t0.00006\t2.66705\t0.00006\t0.22085\t17.94282\t0.54267\t1.10825\t0.00006\t0.00006\t2.71634\t7.29458\t3.79416\t0.00006\t5.51139\t3.62114\t0.00006\t1.37266\t0.00006\t0.99427\t2.94600\t2.16953\t35.30478\t(7 component(s) constrained)\n",
      "9\t-1281.18\t0.26124\t0.00006\t2.95479\t0.00006\t0.28853\t18.30237\t0.65871\t1.35454\t0.00006\t0.00006\t2.70078\t7.11080\t0.84880\t0.00006\t5.55551\t1.05870\t0.00006\t1.43278\t0.00006\t0.88634\t3.02170\t2.10124\t36.05284\t(7 component(s) constrained)\n",
      "10\t-1277.89\t0.15540\t0.00006\t2.75320\t0.00006\t0.19546\t17.89422\t0.51879\t1.14812\t0.00006\t0.00006\t2.71891\t7.31065\t3.76856\t0.00006\t5.54418\t3.58692\t0.00006\t1.32710\t0.00006\t0.96010\t2.96668\t2.16867\t35.29636\t(7 component(s) constrained)\n",
      "11\t-1281.15\t0.26981\t0.00006\t2.98515\t0.00006\t0.28340\t18.28151\t0.65093\t1.36618\t0.00006\t0.00006\t2.69819\t7.11514\t0.84740\t0.00006\t5.56801\t1.04798\t0.00006\t1.41752\t0.00006\t0.87678\t3.03022\t2.10343\t36.04731\t(7 component(s) constrained)\n",
      "12\t-1277.89\t0.14743\t0.00006\t2.72635\t0.00006\t0.20424\t17.91216\t0.52846\t1.13493\t0.00006\t0.00006\t2.71864\t7.30495\t3.77405\t0.00006\t5.53458\t3.59636\t0.00006\t1.34217\t0.00006\t0.97144\t2.96089\t2.16954\t35.30042\t(7 component(s) constrained)\n",
      "13\t-1281.16\t0.26714\t0.00006\t2.97581\t0.00006\t0.28520\t18.28950\t0.65383\t1.36211\t0.00006\t0.00006\t2.69933\t7.11364\t0.84695\t0.00006\t5.56454\t1.05069\t0.00006\t1.42260\t0.00006\t0.88011\t3.02800\t2.10314\t36.05031\t(7 component(s) constrained)\n",
      "14\t-1277.89\t0.14988\t0.00006\t2.73481\t0.00006\t0.20119\t17.90594\t0.52489\t1.13918\t0.00006\t0.00006\t2.71868\t7.30721\t3.77336\t0.00006\t5.53755\t3.59421\t0.00006\t1.33711\t0.00006\t0.96774\t2.96261\t2.16920\t35.29908\t(7 component(s) constrained)\n",
      "15\t-1281.16\t0.26798\t0.00006\t2.97877\t0.00006\t0.28457\t18.28679\t0.65283\t1.36349\t0.00006\t0.00006\t2.69893\t7.11428\t0.84731\t0.00006\t5.56559\t1.05010\t0.00006\t1.42090\t0.00006\t0.87899\t3.02864\t2.10317\t36.04942\t(7 component(s) constrained)\n",
      "16\t-1277.89\t0.14913\t0.00006\t2.73210\t0.00006\t0.20224\t17.90808\t0.52612\t1.13782\t0.00006\t0.00006\t2.71867\t7.30634\t3.77339\t0.00006\t5.53662\t3.59470\t0.00006\t1.33883\t0.00006\t0.96895\t2.96209\t2.16932\t35.29958\t(7 component(s) constrained)\n",
      "17\t-1281.16\t0.26772\t0.00006\t2.97783\t0.00006\t0.28479\t18.28772\t0.65316\t1.36304\t0.00006\t0.00006\t2.69907\t7.11402\t0.84715\t0.00006\t5.56526\t1.05022\t0.00006\t1.42148\t0.00006\t0.87936\t3.02845\t2.10317\t36.04972\t(7 component(s) constrained)\n",
      "18\t-1277.89\t0.14936\t0.00006\t2.73298\t0.00006\t0.20188\t17.90735\t0.52571\t1.13825\t0.00006\t0.00006\t2.71868\t7.30667\t3.77342\t0.00006\t5.53692\t3.59460\t0.00006\t1.33824\t0.00006\t0.96855\t2.96225\t2.16928\t35.29939\t(7 component(s) constrained)\n",
      "19\t-1281.16\t0.26780\t0.00006\t2.97813\t0.00006\t0.28472\t18.28740\t0.65305\t1.36318\t0.00006\t0.00006\t2.69902\t7.11412\t0.84721\t0.00006\t5.56537\t1.05020\t0.00006\t1.42128\t0.00006\t0.87924\t3.02851\t2.10317\t36.04962\t(7 component(s) constrained)\n",
      "20\t-1277.89\t0.14929\t0.00006\t2.73269\t0.00006\t0.20200\t17.90760\t0.52585\t1.13811\t0.00006\t0.00006\t2.71867\t7.30655\t3.77340\t0.00006\t5.53682\t3.59462\t0.00006\t1.33844\t0.00006\t0.96868\t2.96220\t2.16929\t35.29946\t(7 component(s) constrained)\n",
      "21\t-1281.16\t0.26777\t0.00006\t2.97803\t0.00006\t0.28474\t18.28751\t0.65309\t1.36313\t0.00006\t0.00006\t2.69904\t7.11408\t0.84719\t0.00006\t5.56533\t1.05020\t0.00006\t1.42135\t0.00006\t0.87928\t3.02849\t2.10317\t36.04965\t(7 component(s) constrained)\n",
      "22\t-1277.89\t0.14931\t0.00006\t2.73279\t0.00006\t0.20196\t17.90752\t0.52580\t1.13816\t0.00006\t0.00006\t2.71868\t7.30659\t3.77341\t0.00006\t5.53685\t3.59461\t0.00006\t1.33837\t0.00006\t0.96864\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "23\t-1281.16\t0.26778\t0.00006\t2.97807\t0.00006\t0.28473\t18.28747\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42132\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "24\t-1277.89\t0.14930\t0.00006\t2.73276\t0.00006\t0.20198\t17.90754\t0.52582\t1.13814\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33840\t0.00006\t0.96866\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "25\t-1281.16\t0.26778\t0.00006\t2.97805\t0.00006\t0.28474\t18.28749\t0.65308\t1.36314\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04965\t(7 component(s) constrained)\n",
      "26\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90753\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "27\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "28\t-1277.89\t0.14931\t0.00006\t2.73276\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "29\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "30\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "31\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "32\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "33\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "34\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "35\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "36\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "37\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "38\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "39\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "40\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "41\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "42\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "43\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "44\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "45\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "46\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "47\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "48\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "49\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "50\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "51\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "52\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "53\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "54\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "55\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "56\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "57\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "58\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "59\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "60\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "61\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "62\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "63\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "64\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "65\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "66\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "67\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "68\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "69\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "70\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "71\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "72\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "73\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "74\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "75\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "76\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "77\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "78\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "79\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "80\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "81\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "82\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "83\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "84\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "85\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "86\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "87\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "88\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "89\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "90\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "91\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "92\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "93\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "94\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "95\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "96\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "97\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "98\t-1277.89\t0.14931\t0.00006\t2.73277\t0.00006\t0.20197\t17.90754\t0.52581\t1.13815\t0.00006\t0.00006\t2.71868\t7.30658\t3.77341\t0.00006\t5.53684\t3.59461\t0.00006\t1.33839\t0.00006\t0.96865\t2.96221\t2.16929\t35.29944\t(7 component(s) constrained)\n",
      "99\t-1281.16\t0.26778\t0.00006\t2.97806\t0.00006\t0.28474\t18.28748\t0.65308\t1.36315\t0.00006\t0.00006\t2.69903\t7.11409\t0.84720\t0.00006\t5.56534\t1.05020\t0.00006\t1.42133\t0.00006\t0.87927\t3.02850\t2.10317\t36.04964\t(7 component(s) constrained)\n",
      "\u001b[0;31mError: \u001b[0mLog-likelihood not converged (stop after 100 iteractions). \n",
      "You can specify the option --reml-maxit to allow for more iterations.\n",
      "\n",
      "\u001b[0m\u001b[0mAn error occurs, please check the options or data\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:37:02 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--mgrm-bin multipleGRMsnot6.txt\n",
      "--make-grm\n",
      "--out GCTAgrmnot6\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "There are 21 GRM file names specified in [multipleGRMsnot6.txt].\n",
      "498 individuals in common in the GRM files.\n",
      "Reading the GRM from the 1th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr1.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr1.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr1.grm.bin].\n",
      "Reading the number of SNPs for the GRM from [GCTAgrmchr1.grm.N.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr1.grm.bin].\n",
      "Reading the GRM from the 2th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr2.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr2.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr2.grm.bin].\n",
      "Reading the number of SNPs for the GRM from [GCTAgrmchr2.grm.N.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr2.grm.bin].\n",
      "Reading the GRM from the 3th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr3.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr3.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr3.grm.bin].\n",
      "Reading the number of SNPs for the GRM from [GCTAgrmchr3.grm.N.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr3.grm.bin].\n",
      "Reading the GRM from the 4th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr4.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr4.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr4.grm.bin].\n",
      "Reading the number of SNPs for the GRM from [GCTAgrmchr4.grm.N.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr4.grm.bin].\n",
      "Reading the GRM from the 5th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr5.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr5.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr5.grm.bin].\n",
      "Reading the number of SNPs for the GRM from [GCTAgrmchr5.grm.N.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr5.grm.bin].\n",
      "Reading the GRM from the 6th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr7.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr7.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr7.grm.bin].\n",
      "Reading the number of SNPs for the GRM from [GCTAgrmchr7.grm.N.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr7.grm.bin].\n",
      "Reading the GRM from the 7th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr8.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr8.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr8.grm.bin].\n",
      "Reading the number of SNPs for the GRM from [GCTAgrmchr8.grm.N.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr8.grm.bin].\n",
      "Reading the GRM from the 8th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr9.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr9.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr9.grm.bin].\n",
      "Reading the number of SNPs for the GRM from [GCTAgrmchr9.grm.N.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr9.grm.bin].\n",
      "Reading the GRM from the 9th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr10.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr10.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr10.grm.bin].\n",
      "Reading the number of SNPs for the GRM from [GCTAgrmchr10.grm.N.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr10.grm.bin].\n",
      "Reading the GRM from the 10th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr11.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr11.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr11.grm.bin].\n",
      "Reading the number of SNPs for the GRM from [GCTAgrmchr11.grm.N.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr11.grm.bin].\n",
      "Reading the GRM from the 11th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr12.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr12.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr12.grm.bin].\n",
      "Reading the number of SNPs for the GRM from [GCTAgrmchr12.grm.N.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr12.grm.bin].\n",
      "Reading the GRM from the 12th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr13.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr13.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr13.grm.bin].\n",
      "Reading the number of SNPs for the GRM from [GCTAgrmchr13.grm.N.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr13.grm.bin].\n",
      "Reading the GRM from the 13th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr14.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr14.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr14.grm.bin].\n",
      "Reading the number of SNPs for the GRM from [GCTAgrmchr14.grm.N.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr14.grm.bin].\n",
      "Reading the GRM from the 14th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr15.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr15.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr15.grm.bin].\n",
      "Reading the number of SNPs for the GRM from [GCTAgrmchr15.grm.N.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr15.grm.bin].\n",
      "Reading the GRM from the 15th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr16.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr16.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr16.grm.bin].\n",
      "Reading the number of SNPs for the GRM from [GCTAgrmchr16.grm.N.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr16.grm.bin].\n",
      "Reading the GRM from the 16th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr17.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr17.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr17.grm.bin].\n",
      "Reading the number of SNPs for the GRM from [GCTAgrmchr17.grm.N.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr17.grm.bin].\n",
      "Reading the GRM from the 17th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr18.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr18.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr18.grm.bin].\n",
      "Reading the number of SNPs for the GRM from [GCTAgrmchr18.grm.N.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr18.grm.bin].\n",
      "Reading the GRM from the 18th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr19.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr19.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr19.grm.bin].\n",
      "Reading the number of SNPs for the GRM from [GCTAgrmchr19.grm.N.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr19.grm.bin].\n",
      "Reading the GRM from the 19th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr20.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr20.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr20.grm.bin].\n",
      "Reading the number of SNPs for the GRM from [GCTAgrmchr20.grm.N.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr20.grm.bin].\n",
      "Reading the GRM from the 20th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr21.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr21.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr21.grm.bin].\n",
      "Reading the number of SNPs for the GRM from [GCTAgrmchr21.grm.N.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr21.grm.bin].\n",
      "Reading the GRM from the 21th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr22.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr22.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr22.grm.bin].\n",
      "Reading the number of SNPs for the GRM from [GCTAgrmchr22.grm.N.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr22.grm.bin].\n",
      "\n",
      "21 GRMs have been merged together.\n",
      "GRM of 498 individuals has been saved in the file [GCTAgrmnot6.grm.bin] (in binary format).\n",
      "Number of SNPs to calculate the genetic relationship between each pair of individuals has been saved in the file [GCTAgrmnot6.grm.N.bin] (in binary format).\n",
      "IDs for the GRM file [GCTAgrmnot6.grm.bin] have been saved in the file [GCTAgrmnot6.grm.id].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:37:02 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.18 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:37:03 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--reml\n",
      "--mgrm-bin multipleGRMs6andnot6.txt\n",
      "--pheno phenos.txt\n",
      "--out GCTAherit6andnot6\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading phenotypes from [phenos.txt].\n",
      "Non-missing phenotypes of 498 individuals are included from [phenos.txt].\n",
      "\n",
      "There are 2 GRM file names specified in the file [multipleGRMs6andnot6.txt].\n",
      "Reading the GRM from the 1th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmchr6.grm.id].\n",
      "498 IDs are read from [GCTAgrmchr6.grm.id].\n",
      "Reading the GRM from [GCTAgrmchr6.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmchr6.grm.bin].\n",
      "Reading the GRM from the 2th file ...\n",
      "Reading IDs of the GRM from [GCTAgrmnot6.grm.id].\n",
      "498 IDs are read from [GCTAgrmnot6.grm.id].\n",
      "Reading the GRM from [GCTAgrmnot6.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrmnot6.grm.bin].\n",
      "498 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "498 observations, 1 fixed effect(s), and 3 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values:  20.748 20.7874 21.1358\n",
      "logL: -1270.78\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G1)\tV(G2)\tV(e)\t\n",
      "1\t-1270.71\t19.93932\t17.28038\t24.98948\t\n",
      "2\t-1270.08\t18.14765\t9.98903\t33.38079\t\n",
      "3\t-1269.60\t18.18990\t10.90080\t32.73385\t\n",
      "4\t-1269.59\t18.20629\t10.78832\t32.82773\t\n",
      "5\t-1269.59\t18.20278\t10.80438\t32.81582\t\n",
      "6\t-1269.59\t18.20343\t10.80209\t32.81739\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 31.04062 31.40201\n",
      "logL: -1274.35341\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G2)\tV(e)\t\n",
      "1\t-1274.34\t28.62580\t33.60718\t\n",
      "2\t-1274.17\t23.65981\t38.22495\t\n",
      "3\t-1274.04\t24.18239\t37.80228\t\n",
      "4\t-1274.04\t24.13438\t37.84796\t\n",
      "5\t-1274.04\t24.13923\t37.84343\t\n",
      "6\t-1274.04\t24.13875\t37.84388\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G1)\t18.203433\t7.809399\n",
      "V(G2)\t10.802094\t10.414431\n",
      "V(e)\t32.817391\t8.614963\n",
      "Vp\t61.822918\t4.024798\n",
      "V(G1)/Vp\t0.294445\t0.121720\n",
      "V(G2)/Vp\t0.174726\t0.167702\n",
      "\n",
      "Sum of V(G)/Vp\t0.469171\t0.141402\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "6.098672e+01\t-4.173938e+01\t-9.459464e+00\t\n",
      "-4.173938e+01\t1.084604e+02\t-6.253400e+01\t\n",
      "-9.459464e+00\t-6.253400e+01\t7.421758e+01\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [GCTAherit6andnot6.hsq].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:37:03 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.13 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:37:03 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32m\u001b[0mOptions: \n",
      "\u001b[0;32m\u001b[0m \n",
      "--grm GCTAgrm \n",
      "--make-bK-sparse 0.05 \n",
      "--out GCTAsparsegrm \n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32m\u001b[0mPruning the GRM to a sparse matrix with a cutoff of 0.050000...\n",
      "\u001b[0;32m\u001b[0mTotal number of parts to be processed: 1\n",
      "    \u001b[0;32m\u001b[0mSaving 498 individual IDs\n",
      "    \u001b[0;32m\u001b[0mProcessing part 1\n",
      "\u001b[0;32m\u001b[0mSaving the sparse GRM (894 pairs) to [GCTAsparsegrm.grm.sp]\n",
      "\u001b[0;32mfinished generating a sparse GRM \u001b[0mSuccess:\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:37:03 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.00 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:37:03 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32m\u001b[0mOptions: \n",
      "\u001b[0;32m\u001b[0m \n",
      "--bfile quantfamdata \n",
      "--make-grm-part 3 1 \n",
      "--thread-num 5 \n",
      "--out test \n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32m\u001b[0mThe program will be running with up to 5 threads.\n",
      "\u001b[0;32m\u001b[0mNote: GRM is computed using the SNPs on the autosomes.\n",
      "\u001b[0;32m\u001b[0mReading PLINK FAM file from [quantfamdata.fam]...\n",
      "\u001b[0;32m\u001b[0m498 individuals to be included from FAM file.\n",
      "498 individuals to be included. 226 males, 272 females, 0 unknown.\n",
      "\u001b[0;32m\u001b[0mReading PLINK BIM file from [quantfamdata.bim]...\n",
      "\u001b[0;32m\u001b[0m134946 SNPs to be included from BIM file(s).\n",
      "\u001b[0;32m\u001b[0mComputing the genetic relationship matrix (GRM) v2 ...\n",
      "\u001b[0;32m\u001b[0mSubset 1/3, no. subject 1-250\n",
      "  \u001b[0;32m\u001b[0m250 samples, 134946 markers, 31375 GRM elements\n",
      "\u001b[0;32m\u001b[0mIDs for the GRM file have been saved in the file [test.part_3_1.grm.id]\n",
      "Computing GRM...\n",
      "  \u001b[0;32m\u001b[0m100% finished in 0.6 sec\n",
      "134946 SNPs have been processed.\n",
      "  Used 134946 valid SNPs.\n",
      "\u001b[0;32m\u001b[0mThe GRM computation is completed.\n",
      "\u001b[0;32m\u001b[0mSaving GRM...\n",
      "\u001b[0;32m\u001b[0mGRM has been saved in the file [test.part_3_1.grm.bin]\n",
      "\u001b[0;32m\u001b[0mNumber of SNPs in each pair of individuals has been saved in the file [test.part_3_1.grm.N.bin]\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:37:04 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.74 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:37:04 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32m\u001b[0mOptions: \n",
      "\u001b[0;32m\u001b[0m \n",
      "--bfile quantfamdata \n",
      "--make-grm-part 3 2 \n",
      "--thread-num 5 \n",
      "--out test \n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32m\u001b[0mThe program will be running with up to 5 threads.\n",
      "\u001b[0;32m\u001b[0mNote: GRM is computed using the SNPs on the autosomes.\n",
      "\u001b[0;32m\u001b[0mReading PLINK FAM file from [quantfamdata.fam]...\n",
      "\u001b[0;32m\u001b[0m498 individuals to be included from FAM file.\n",
      "498 individuals to be included. 226 males, 272 females, 0 unknown.\n",
      "\u001b[0;32m\u001b[0mReading PLINK BIM file from [quantfamdata.bim]...\n",
      "\u001b[0;32m\u001b[0m134946 SNPs to be included from BIM file(s).\n",
      "\u001b[0;32m\u001b[0mComputing the genetic relationship matrix (GRM) v2 ...\n",
      "\u001b[0;32m\u001b[0mSubset 2/3, no. subject 251-392\n",
      "  \u001b[0;32m\u001b[0m142 samples, 134946 markers, 45653 GRM elements\n",
      "\u001b[0;32m\u001b[0mIDs for the GRM file have been saved in the file [test.part_3_2.grm.id]\n",
      "Computing GRM...\n",
      "  \u001b[0;32m\u001b[0m100% finished in 0.6 sec\n",
      "134946 SNPs have been processed.\n",
      "  Used 134946 valid SNPs.\n",
      "\u001b[0;32m\u001b[0mThe GRM computation is completed.\n",
      "\u001b[0;32m\u001b[0mSaving GRM...\n",
      "\u001b[0;32m\u001b[0mGRM has been saved in the file [test.part_3_2.grm.bin]\n",
      "\u001b[0;32m\u001b[0mNumber of SNPs in each pair of individuals has been saved in the file [test.part_3_2.grm.N.bin]\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:37:05 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.81 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:37:05 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32m\u001b[0mOptions: \n",
      "\u001b[0;32m\u001b[0m \n",
      "--bfile quantfamdata \n",
      "--make-grm-part 3 3 \n",
      "--thread-num 5 \n",
      "--out test \n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32m\u001b[0mThe program will be running with up to 5 threads.\n",
      "\u001b[0;32m\u001b[0mNote: GRM is computed using the SNPs on the autosomes.\n",
      "\u001b[0;32m\u001b[0mReading PLINK FAM file from [quantfamdata.fam]...\n",
      "\u001b[0;32m\u001b[0m498 individuals to be included from FAM file.\n",
      "498 individuals to be included. 226 males, 272 females, 0 unknown.\n",
      "\u001b[0;32m\u001b[0mReading PLINK BIM file from [quantfamdata.bim]...\n",
      "\u001b[0;32m\u001b[0m134946 SNPs to be included from BIM file(s).\n",
      "\u001b[0;32m\u001b[0mComputing the genetic relationship matrix (GRM) v2 ...\n",
      "\u001b[0;32m\u001b[0mSubset 3/3, no. subject 393-498\n",
      "  \u001b[0;32m\u001b[0m106 samples, 134946 markers, 47223 GRM elements\n",
      "\u001b[0;32m\u001b[0mIDs for the GRM file have been saved in the file [test.part_3_3.grm.id]\n",
      "Computing GRM...\n",
      "  \u001b[0;32m\u001b[0m100% finished in 0.7 sec\n",
      "134946 SNPs have been processed.\n",
      "  Used 134946 valid SNPs.\n",
      "\u001b[0;32m\u001b[0mThe GRM computation is completed.\n",
      "\u001b[0;32m\u001b[0mSaving GRM...\n",
      "\u001b[0;32m\u001b[0mGRM has been saved in the file [test.part_3_3.grm.bin]\n",
      "\u001b[0;32m\u001b[0mNumber of SNPs in each pair of individuals has been saved in the file [test.part_3_3.grm.N.bin]\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:37:06 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.83 sec.\n"
     ]
    }
   ],
   "source": [
    "gcta64 --bfile quantfamdata --autosome --make-grm-bin --out GCTAgrm \n",
    "\n",
    "gcta64 --reml --grm-bin GCTAgrm --pheno phenos.txt --out GCTAherit \n",
    "\n",
    "gcta64 --bfile quantfamdata --chr 1 --make-grm-bin --out GCTAgrmchr1 \n",
    "gcta64 --reml --grm-bin GCTAgrmchr1 --pheno phenos.txt \\\n",
    "--out GCTAheritchr1\n",
    "\n",
    "gcta64 --bfile quantfamdata --chr 2 --make-grm-bin --out GCTAgrmchr2\n",
    "gcta64 --reml --grm-bin GCTAgrmchr2 --pheno phenos.txt \\\n",
    "--out GCTAheritchr2 \n",
    "\n",
    "gcta64 --bfile quantfamdata --chr 6 --make-grm-bin --out GCTAgrmchr6 \n",
    "gcta64 --reml --grm-bin GCTAgrmchr6 --pheno phenos.txt --out GCTAheritchr6\n",
    "\n",
    "\n",
    "gcta64 --bfile quantfamdata --chr 12 --make-grm-bin --out GCTAgrmchr12\n",
    "gcta64 --reml --grm-bin GCTAgrmchr12 --pheno phenos.txt --out GCTAheritchr12\n",
    "\n",
    "\n",
    "gcta64 --bfile quantfamdata --chr 3 --make-grm-bin --out GCTAgrmchr3\n",
    "gcta64 --bfile quantfamdata --chr 4 --make-grm-bin --out GCTAgrmchr4\n",
    "gcta64 --bfile quantfamdata --chr 5 --make-grm-bin --out GCTAgrmchr5\n",
    "gcta64 --bfile quantfamdata --chr 7 --make-grm-bin --out GCTAgrmchr7\n",
    "gcta64 --bfile quantfamdata --chr 8 --make-grm-bin --out GCTAgrmchr8\n",
    "gcta64 --bfile quantfamdata --chr 9 --make-grm-bin --out GCTAgrmchr9\n",
    "gcta64 --bfile quantfamdata --chr 10 --make-grm-bin --out GCTAgrmchr10\n",
    "gcta64 --bfile quantfamdata --chr 11 --make-grm-bin --out GCTAgrmchr11\n",
    "gcta64 --bfile quantfamdata --chr 13 --make-grm-bin --out GCTAgrmchr13\n",
    "gcta64 --bfile quantfamdata --chr 14 --make-grm-bin --out GCTAgrmchr14\n",
    "gcta64 --bfile quantfamdata --chr 15 --make-grm-bin --out GCTAgrmchr15\n",
    "gcta64 --bfile quantfamdata --chr 16 --make-grm-bin --out GCTAgrmchr16\n",
    "gcta64 --bfile quantfamdata --chr 17 --make-grm-bin --out GCTAgrmchr17\n",
    "gcta64 --bfile quantfamdata --chr 18 --make-grm-bin --out GCTAgrmchr18\n",
    "gcta64 --bfile quantfamdata --chr 19 --make-grm-bin --out GCTAgrmchr19\n",
    "gcta64 --bfile quantfamdata --chr 20 --make-grm-bin --out GCTAgrmchr20\n",
    "gcta64 --bfile quantfamdata --chr 21 --make-grm-bin --out GCTAgrmchr21\n",
    "gcta64 --bfile quantfamdata --chr 22 --make-grm-bin --out GCTAgrmchr22 \n",
    "\n",
    "gcta64 --reml --mgrm-bin multipleGRMs.txt --pheno phenos.txt --out GCTAherit22GRMs \n",
    "\n",
    "gcta64 --mgrm-bin multipleGRMsnot6.txt --make-grm --out GCTAgrmnot6 \n",
    "\n",
    "gcta64 --reml --mgrm-bin multipleGRMs6andnot6.txt --pheno phenos.txt \\\n",
    "--out GCTAherit6andnot6 \n",
    "\n",
    "\n",
    "gcta64 --grm GCTAgrm --make-bK-sparse 0.05 --out GCTAsparsegrm \n",
    "\n",
    "gcta64 --bfile quantfamdata --make-grm-part 3 1 --thread-num 5 \\\n",
    "--out test\n",
    "gcta64 --bfile quantfamdata --make-grm-part 3 2 --thread-num 5 \\\n",
    "--out test\n",
    "gcta64 --bfile quantfamdata --make-grm-part 3 3 --thread-num 5 \\\n",
    "--out test "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "eefdf821-6e01-439f-84c5-a66dbb9e3a47",
   "metadata": {
    "kernel": "Bash",
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:37:06 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32m\u001b[0mOptions: \n",
      "\u001b[0;32m\u001b[0m \n",
      "--grm test \n",
      "--make-bK-sparse 0.05 \n",
      "--out newsparsegrm \n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32m\u001b[0mPruning the GRM to a sparse matrix with a cutoff of 0.050000...\n",
      "\u001b[0;32m\u001b[0mTotal number of parts to be processed: 1\n",
      "    \u001b[0;32m\u001b[0mSaving 498 individual IDs\n",
      "    \u001b[0;32m\u001b[0mProcessing part 1\n",
      "\u001b[0;32m\u001b[0mSaving the sparse GRM (894 pairs) to [newsparsegrm.grm.sp]\n",
      "\u001b[0;32mfinished generating a sparse GRM \u001b[0mSuccess:\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:37:06 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.00 sec.\n",
      "==> GCTAsparsegrm.grm.sp <==\n",
      "0\t0\t0.98112994\n",
      "1\t0\t0.10253318\n",
      "1\t1\t0.99352813\n",
      "2\t2\t1.0000606\n",
      "3\t2\t0.11076995\n",
      "3\t3\t0.9978525\n",
      "4\t1\t0.18634462\n",
      "4\t2\t0.11245476\n",
      "4\t3\t0.21224992\n",
      "4\t4\t0.99371773\n",
      "\n",
      "==> newsparsegrm.grm.sp <==\n",
      "0\t0\t0.98112994\n",
      "1\t0\t0.10253316\n",
      "1\t1\t0.99352795\n",
      "2\t2\t1.0000604\n",
      "3\t2\t0.11076996\n",
      "3\t3\t0.99785256\n",
      "4\t1\t0.18634458\n",
      "4\t2\t0.11245478\n",
      "4\t3\t0.21224986\n",
      "4\t4\t0.99371779\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:37:06 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "Accepted options:\n",
      "--grm-bin GCTAgrm\n",
      "--pca 5\n",
      "--out pcs\n",
      "\n",
      "Note: This is a multi-thread program. You could specify the number of threads by the --thread-num option to speed up the computation if there are multiple processors in your machine.\n",
      "\n",
      "Reading IDs of the GRM from [GCTAgrm.grm.id].\n",
      "498 IDs are read from [GCTAgrm.grm.id].\n",
      "Reading the GRM from [GCTAgrm.grm.bin].\n",
      "GRM for 498 individuals are included from [GCTAgrm.grm.bin].\n",
      "\n",
      "Performing principal component analysis ...\n",
      "Eigenvalues of 498 individuals have been saved in [pcs.eigenval].\n",
      "The first 5 eigenvectors of 498 individuals have been saved in [pcs.eigenvec].\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:37:06 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.08 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:37:06 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32m\u001b[0mOptions: \n",
      "\u001b[0;32m\u001b[0m \n",
      "--bfile quantfamdata \n",
      "--grm-sparse GCTAsparsegrm \n",
      "--fastGWA-mlm \n",
      "--pheno phenos.txt \n",
      "--qcovar pcs.eigenvec \n",
      "--out sparse_assoc \n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32m\u001b[0mReading PLINK FAM file from [quantfamdata.fam]...\n",
      "\u001b[0;32m\u001b[0m498 individuals to be included from FAM file.\n",
      "\u001b[0;32m\u001b[0mReading phenotype data from [phenos.txt]...\n",
      "\u001b[0;32m\u001b[0m498 overlapping individuals with non-missing data to be included from the phenotype file.\n",
      "498 individuals to be included. 226 males, 272 females, 0 unknown.\n",
      "\u001b[0;32m\u001b[0mReading PLINK BIM file from [quantfamdata.bim]...\n",
      "\u001b[0;32m\u001b[0m134946 SNPs to be included from BIM file(s).\n",
      "\u001b[0;32m\u001b[0mReading quantitative covariates from [pcs.eigenvec].\n",
      "\u001b[0;32m\u001b[0m5 covariates of 498 samples to be included.\n",
      "\u001b[0;32m\u001b[0m498 overlapping individuals with non-missing data to be included from the covariate file(s).\n",
      "\u001b[0;32m\u001b[0mReading the sparse GRM file from [GCTAsparsegrm]...\n",
      "\u001b[0;32m\u001b[0mAfter matching all the files, 498 individuals to be included in the analysis.\n",
      "\u001b[0;32m\u001b[0mEstimating the genetic variance (Vg) by fastGWA-REML (grid search)...\n",
      "Iteration 1, step size: 0.974355, logL: -1270.8. Vg: 20.4615, searching range: 19.4871 to 21.4358\n",
      "Iteration 2, step size: 0.129914, logL: -1270.8. Vg: 20.3965, searching range: 20.2666 to 20.5264\n",
      "Iteration 3, step size: 0.0173219, logL: -1270.8. Vg: 20.3532, searching range: 20.3359 to 20.3705\n",
      "Iteration 4, step size: 0.00230958, logL: -1270.8. Vg: 20.359, searching range: 20.3567 to 20.3613\n",
      "Iteration 5, step size: 0.000307944, logL: -1270.8. Vg: 20.3597, searching range: 20.3594 to 20.36\n",
      "Iteration 6, step size: 4.10592e-05, logL: -1270.8. Vg: 20.3597, searching range: 20.3596 to 20.3597\n",
      "Iteration 7, step size: 5.47457e-06, logL: -1270.8. Vg: 20.3597, searching range: 20.3597 to 20.3597\n",
      "Iteration 8, step size: 7.29942e-07, logL: -1270.8. Vg: 20.3597, searching range: 20.3597 to 20.3597\n",
      "Iteration 9, step size: 9.73256e-08, logL: -1270.8. Vg: 20.3597, searching range: 20.3597 to 20.3597\n",
      "Iteration 10, step size: 1.29767e-08, logL: -1270.8. Vg: 20.3597, searching range: 20.3597 to 20.3597\n",
      "Iteration 11, step size: 1.73023e-09, logL: -1270.8. Vg: 20.3597, searching range: 20.3597 to 20.3597\n",
      "Iteration 12, step size: 2.30698e-10, logL: -1270.8. Vg: 20.3597, searching range: 20.3597 to 20.3597\n",
      "Iteration 13, step size: 3.07596e-11, logL: -1270.8. Vg: 20.3597, searching range: 20.3597 to 20.3597\n",
      "fastGWA-REML converged.\n",
      "logL: -1270.8\n",
      "Sampling variance/covariance of the estimates of Vg and Ve:\n",
      " 113.789 -103.598\n",
      "-103.598  108.573\n",
      "\n",
      "Source\tVariance\tSE\n",
      "Vg\t20.3597\t10.6672\n",
      "Ve\t40.5375\t10.4198\n",
      "Vp\t60.8972\n",
      "\n",
      "Heritability = 0.334329 (Pval = 0.0563109)\n",
      "fastGWA-REML runtime: 0.0798168 sec.\n",
      "\u001b[0;33mWarning: \u001b[0mthe estimate of Vg is not statistically significant (i.e., p > 0.05). This is likely because the number of closely related individuals in the sample is not large enough. \n",
      "In this case, the program will use linear regression for association test.\n",
      "\u001b[0;32m\u001b[0m\n",
      "Performing fastGWA linear regression analysis...\n",
      "fastGWA results will be saved in text format to [sparse_assoc.fastGWA].\n",
      "  Filtering out variants with MAF < 0.0001, or customise it with --maf flag.\n",
      "  Filtering out variants with missingness rate > 0.10, or customise it with --geno flag.\n",
      "  \u001b[0;32m\u001b[0m100% finished in 0.5 sec\n",
      "134946 SNPs have been processed.\n",
      "Saved 134946 SNPs.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:37:07 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.77 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:37:07 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32m\u001b[0mOptions: \n",
      "\u001b[0;32m\u001b[0m \n",
      "--bfile quantfamdata \n",
      "--grm-sparse newsparsegrm \n",
      "--fastGWA-mlm \n",
      "--pheno phenos.txt \n",
      "--qcovar pcs.eigenvec \n",
      "--out newsparse_assoc \n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32m\u001b[0mReading PLINK FAM file from [quantfamdata.fam]...\n",
      "\u001b[0;32m\u001b[0m498 individuals to be included from FAM file.\n",
      "\u001b[0;32m\u001b[0mReading phenotype data from [phenos.txt]...\n",
      "\u001b[0;32m\u001b[0m498 overlapping individuals with non-missing data to be included from the phenotype file.\n",
      "498 individuals to be included. 226 males, 272 females, 0 unknown.\n",
      "\u001b[0;32m\u001b[0mReading PLINK BIM file from [quantfamdata.bim]...\n",
      "\u001b[0;32m\u001b[0m134946 SNPs to be included from BIM file(s).\n",
      "\u001b[0;32m\u001b[0mReading quantitative covariates from [pcs.eigenvec].\n",
      "\u001b[0;32m\u001b[0m5 covariates of 498 samples to be included.\n",
      "\u001b[0;32m\u001b[0m498 overlapping individuals with non-missing data to be included from the covariate file(s).\n",
      "\u001b[0;32m\u001b[0mReading the sparse GRM file from [newsparsegrm]...\n",
      "\u001b[0;32m\u001b[0mAfter matching all the files, 498 individuals to be included in the analysis.\n",
      "\u001b[0;32m\u001b[0mEstimating the genetic variance (Vg) by fastGWA-REML (grid search)...\n",
      "Iteration 1, step size: 0.974355, logL: -1270.8. Vg: 20.4615, searching range: 19.4871 to 21.4358\n",
      "Iteration 2, step size: 0.129914, logL: -1270.8. Vg: 20.3965, searching range: 20.2666 to 20.5264\n",
      "Iteration 3, step size: 0.0173219, logL: -1270.8. Vg: 20.3532, searching range: 20.3359 to 20.3705\n",
      "Iteration 4, step size: 0.00230958, logL: -1270.8. Vg: 20.359, searching range: 20.3567 to 20.3613\n",
      "Iteration 5, step size: 0.000307944, logL: -1270.8. Vg: 20.3597, searching range: 20.3594 to 20.36\n",
      "Iteration 6, step size: 4.10592e-05, logL: -1270.8. Vg: 20.3597, searching range: 20.3596 to 20.3597\n",
      "Iteration 7, step size: 5.47457e-06, logL: -1270.8. Vg: 20.3597, searching range: 20.3597 to 20.3597\n",
      "Iteration 8, step size: 7.29942e-07, logL: -1270.8. Vg: 20.3597, searching range: 20.3597 to 20.3597\n",
      "Iteration 9, step size: 9.73256e-08, logL: -1270.8. Vg: 20.3597, searching range: 20.3597 to 20.3597\n",
      "Iteration 10, step size: 1.29767e-08, logL: -1270.8. Vg: 20.3597, searching range: 20.3597 to 20.3597\n",
      "Iteration 11, step size: 1.73023e-09, logL: -1270.8. Vg: 20.3597, searching range: 20.3597 to 20.3597\n",
      "Iteration 12, step size: 2.30698e-10, logL: -1270.8. Vg: 20.3597, searching range: 20.3597 to 20.3597\n",
      "Iteration 13, step size: 3.07596e-11, logL: -1270.8. Vg: 20.3597, searching range: 20.3597 to 20.3597\n",
      "fastGWA-REML converged.\n",
      "logL: -1270.8\n",
      "Sampling variance/covariance of the estimates of Vg and Ve:\n",
      " 113.789 -103.598\n",
      "-103.598  108.573\n",
      "\n",
      "Source\tVariance\tSE\n",
      "Vg\t20.3597\t10.6672\n",
      "Ve\t40.5375\t10.4198\n",
      "Vp\t60.8972\n",
      "\n",
      "Heritability = 0.334329 (Pval = 0.0563108)\n",
      "fastGWA-REML runtime: 0.07881 sec.\n",
      "\u001b[0;33mWarning: \u001b[0mthe estimate of Vg is not statistically significant (i.e., p > 0.05). This is likely because the number of closely related individuals in the sample is not large enough. \n",
      "In this case, the program will use linear regression for association test.\n",
      "\u001b[0;32m\u001b[0m\n",
      "Performing fastGWA linear regression analysis...\n",
      "fastGWA results will be saved in text format to [newsparse_assoc.fastGWA].\n",
      "  Filtering out variants with MAF < 0.0001, or customise it with --maf flag.\n",
      "  Filtering out variants with missingness rate > 0.10, or customise it with --geno flag.\n",
      "  \u001b[0;32m\u001b[0m100% finished in 0.5 sec\n",
      "134946 SNPs have been processed.\n",
      "Saved 134946 SNPs.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:37:08 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.75 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:37:08 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32m\u001b[0mOptions: \n",
      "\u001b[0;32m\u001b[0m \n",
      "--bfile quantfamdata \n",
      "--grm-sparse GCTAsparsegrm \n",
      "--fastGWA-mlm \n",
      "--pheno phenos.txt \n",
      "--out LMMsparse_assoc \n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32m\u001b[0mReading PLINK FAM file from [quantfamdata.fam]...\n",
      "\u001b[0;32m\u001b[0m498 individuals to be included from FAM file.\n",
      "\u001b[0;32m\u001b[0mReading phenotype data from [phenos.txt]...\n",
      "\u001b[0;32m\u001b[0m498 overlapping individuals with non-missing data to be included from the phenotype file.\n",
      "498 individuals to be included. 226 males, 272 females, 0 unknown.\n",
      "\u001b[0;32m\u001b[0mReading PLINK BIM file from [quantfamdata.bim]...\n",
      "\u001b[0;32m\u001b[0m134946 SNPs to be included from BIM file(s).\n",
      "\u001b[0;32m\u001b[0mReading the sparse GRM file from [GCTAsparsegrm]...\n",
      "\u001b[0;32m\u001b[0mAfter matching all the files, 498 individuals to be included in the analysis.\n",
      "\u001b[0;32m\u001b[0mEstimating the genetic variance (Vg) by fastGWA-REML (grid search)...\n",
      "Iteration 1, step size: 0.996879, logL: -1273.59. Vg: 27.9126, searching range: 26.9157 to 28.9095\n",
      "Iteration 2, step size: 0.132917, logL: -1273.59. Vg: 28.3778, searching range: 28.2449 to 28.5108\n",
      "Iteration 3, step size: 0.0177223, logL: -1273.59. Vg: 28.3158, searching range: 28.2981 to 28.3335\n",
      "Iteration 4, step size: 0.00236297, logL: -1273.59. Vg: 28.3217, searching range: 28.3194 to 28.3241\n",
      "Iteration 5, step size: 0.000315063, logL: -1273.59. Vg: 28.3216, searching range: 28.3212 to 28.3219\n",
      "Iteration 6, step size: 4.20084e-05, logL: -1273.59. Vg: 28.3215, searching range: 28.3215 to 28.3215\n",
      "Iteration 7, step size: 5.60112e-06, logL: -1273.59. Vg: 28.3215, searching range: 28.3215 to 28.3215\n",
      "Iteration 8, step size: 7.46816e-07, logL: -1273.59. Vg: 28.3215, searching range: 28.3215 to 28.3215\n",
      "Iteration 9, step size: 9.95755e-08, logL: -1273.59. Vg: 28.3215, searching range: 28.3215 to 28.3215\n",
      "Iteration 10, step size: 6.63837e-09, logL: -1273.59. Vg: 28.3215, searching range: 28.3215 to 28.3215\n",
      "Iteration 11, step size: 8.85116e-10, logL: -1273.59. Vg: 28.3215, searching range: 28.3215 to 28.3215\n",
      "Iteration 12, step size: 5.90077e-11, logL: -1273.59. Vg: 28.3215, searching range: 28.3215 to 28.3215\n",
      "Iteration 13, step size: 3.9338e-12, logL: -1273.59. Vg: 28.3215, searching range: 28.3215 to 28.3215\n",
      "fastGWA-REML converged.\n",
      "logL: -1273.59\n",
      "Sampling variance/covariance of the estimates of Vg and Ve:\n",
      " 108.552 -94.7786\n",
      "-94.7786  97.1234\n",
      "\n",
      "Source\tVariance\tSE\n",
      "Vg\t28.3215\t10.4188\n",
      "Ve\t33.9835\t9.85512\n",
      "Vp\t62.305\n",
      "\n",
      "Heritability = 0.454562 (Pval = 0.00656191)\n",
      "fastGWA-REML runtime: 0.0800791 sec.\n",
      "\u001b[0;32m\u001b[0m\n",
      "Tuning parameters using 2000 null SNPs...\n",
      "  reading genotypes...\n",
      "  \u001b[0;32m\u001b[0m100% finished in 0.2 sec\n",
      "1981 SNPs have been processed.\n",
      "Tuning of Gamma finished at the 1000th SNP.\n",
      "\u001b[0;32m\u001b[0mMean GRAMMAR-Gamma value = 0.015448\n",
      "Tuning of Gamma finished 0.167446 seconds.\n",
      "\u001b[0;32m\u001b[0m\n",
      "Performing fastGWA mixed model association analysis...\n",
      "fastGWA results will be saved in text format to [LMMsparse_assoc.fastGWA].\n",
      "  Filtering out variants with MAF < 0.0001, or customise it with --maf flag.\n",
      "  Filtering out variants with missingness rate > 0.10, or customise it with --geno flag.\n",
      "  \u001b[0;32m\u001b[0m100% finished in 0.5 sec\n",
      "134946 SNPs have been processed.\n",
      "Saved 134946 SNPs.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:37:09 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.90 sec.\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32m\u001b[0m* Genome-wide Complex Trait Analysis (GCTA)\n",
      "\u001b[0;32m\u001b[0m* version v1.94.1 Linux\n",
      "\u001b[0;32m\u001b[0m* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "\u001b[0;32m\u001b[0m* (C) 2010-present, Yang Lab, Westlake University\n",
      "\u001b[0;32m\u001b[0m* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "\u001b[0;32m\u001b[0m*******************************************************************\n",
      "\u001b[0;32mAnalysis started \u001b[0mat 15:37:09 UTC on Thu Apr 18 2024.\n",
      "\u001b[0;32m\u001b[0mHostname: b7b957017671\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32m\u001b[0mOptions: \n",
      "\u001b[0;32m\u001b[0m \n",
      "--bfile quantfamdata \n",
      "--grm-sparse newsparsegrm \n",
      "--fastGWA-mlm \n",
      "--pheno phenos.txt \n",
      "--out LMMnewsparse_assoc \n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32m\u001b[0mReading PLINK FAM file from [quantfamdata.fam]...\n",
      "\u001b[0;32m\u001b[0m498 individuals to be included from FAM file.\n",
      "\u001b[0;32m\u001b[0mReading phenotype data from [phenos.txt]...\n",
      "\u001b[0;32m\u001b[0m498 overlapping individuals with non-missing data to be included from the phenotype file.\n",
      "498 individuals to be included. 226 males, 272 females, 0 unknown.\n",
      "\u001b[0;32m\u001b[0mReading PLINK BIM file from [quantfamdata.bim]...\n",
      "\u001b[0;32m\u001b[0m134946 SNPs to be included from BIM file(s).\n",
      "\u001b[0;32m\u001b[0mReading the sparse GRM file from [newsparsegrm]...\n",
      "\u001b[0;32m\u001b[0mAfter matching all the files, 498 individuals to be included in the analysis.\n",
      "\u001b[0;32m\u001b[0mEstimating the genetic variance (Vg) by fastGWA-REML (grid search)...\n",
      "Iteration 1, step size: 0.996879, logL: -1273.59. Vg: 27.9126, searching range: 26.9157 to 28.9095\n",
      "Iteration 2, step size: 0.132917, logL: -1273.59. Vg: 28.3778, searching range: 28.2449 to 28.5108\n",
      "Iteration 3, step size: 0.0177223, logL: -1273.59. Vg: 28.3158, searching range: 28.2981 to 28.3335\n",
      "Iteration 4, step size: 0.00236297, logL: -1273.59. Vg: 28.3217, searching range: 28.3194 to 28.3241\n",
      "Iteration 5, step size: 0.000315063, logL: -1273.59. Vg: 28.3216, searching range: 28.3212 to 28.3219\n",
      "Iteration 6, step size: 4.20084e-05, logL: -1273.59. Vg: 28.3215, searching range: 28.3215 to 28.3215\n",
      "Iteration 7, step size: 5.60112e-06, logL: -1273.59. Vg: 28.3215, searching range: 28.3215 to 28.3215\n",
      "Iteration 8, step size: 7.46816e-07, logL: -1273.59. Vg: 28.3215, searching range: 28.3215 to 28.3215\n",
      "Iteration 9, step size: 9.95755e-08, logL: -1273.59. Vg: 28.3215, searching range: 28.3215 to 28.3215\n",
      "Iteration 10, step size: 1.32767e-08, logL: -1273.59. Vg: 28.3215, searching range: 28.3215 to 28.3215\n",
      "Iteration 11, step size: 1.77023e-09, logL: -1273.59. Vg: 28.3215, searching range: 28.3215 to 28.3215\n",
      "Iteration 12, step size: 2.36031e-10, logL: -1273.59. Vg: 28.3215, searching range: 28.3215 to 28.3215\n",
      "Iteration 13, step size: 3.14709e-11, logL: -1273.59. Vg: 28.3215, searching range: 28.3215 to 28.3215\n",
      "fastGWA-REML converged.\n",
      "logL: -1273.59\n",
      "Sampling variance/covariance of the estimates of Vg and Ve:\n",
      " 108.552 -94.7786\n",
      "-94.7786  97.1234\n",
      "\n",
      "Source\tVariance\tSE\n",
      "Vg\t28.3215\t10.4188\n",
      "Ve\t33.9835\t9.85512\n",
      "Vp\t62.305\n",
      "\n",
      "Heritability = 0.454563 (Pval = 0.00656189)\n",
      "fastGWA-REML runtime: 0.0900761 sec.\n",
      "\u001b[0;32m\u001b[0m\n",
      "Tuning parameters using 2000 null SNPs...\n",
      "  reading genotypes...\n",
      "  \u001b[0;32m\u001b[0m100% finished in 0.2 sec\n",
      "1981 SNPs have been processed.\n",
      "Tuning of Gamma finished at the 1000th SNP.\n",
      "\u001b[0;32m\u001b[0mMean GRAMMAR-Gamma value = 0.015448\n",
      "Tuning of Gamma finished 0.171788 seconds.\n",
      "\u001b[0;32m\u001b[0m\n",
      "Performing fastGWA mixed model association analysis...\n",
      "fastGWA results will be saved in text format to [LMMnewsparse_assoc.fastGWA].\n",
      "  Filtering out variants with MAF < 0.0001, or customise it with --maf flag.\n",
      "  Filtering out variants with missingness rate > 0.10, or customise it with --geno flag.\n",
      "  \u001b[0;32m\u001b[0m100% finished in 0.5 sec\n",
      "134946 SNPs have been processed.\n",
      "Saved 134946 SNPs.\n",
      "\u001b[0;32m\u001b[0m\n",
      "\u001b[0;32mAnalysis finished \u001b[0mat 15:37:10 UTC on Thu Apr 18 2024\n",
      "Overall computational time: 0.92 sec.\n"
     ]
    }
   ],
   "source": [
    "cat test.part_3_*.grm.id > test.grm.id\n",
    "cat test.part_3_*.grm.bin > test.grm.bin\n",
    "cat test.part_3_*.grm.N.bin > test.grm.N.bin"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4a18f65b-5d52-46b3-a66e-0deb68e9d8f2",
   "metadata": {
    "kernel": "Bash",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "gcta64 --grm test --make-bK-sparse 0.05 --out newsparsegrm \n",
    "\n",
    "head *.sp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "086c5917-89e5-43be-9f8c-e15e94765fc2",
   "metadata": {
    "kernel": "Bash",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "gcta64 --grm-bin GCTAgrm --pca 5 --out pcs \n",
    "\n",
    "gcta64 --bfile quantfamdata --grm-sparse GCTAsparsegrm \\\n",
    "--fastGWA-mlm --pheno phenos.txt --qcovar pcs.eigenvec \\\n",
    "--out sparse_assoc \n",
    "\n",
    "gcta64 --bfile quantfamdata --grm-sparse newsparsegrm \\\n",
    "--fastGWA-mlm --pheno phenos.txt --qcovar pcs.eigenvec \\\n",
    "--out newsparse_assoc \n",
    "\n",
    "gcta64 --bfile quantfamdata --grm-sparse GCTAsparsegrm \\\n",
    "--fastGWA-mlm --pheno phenos.txt --out LMMsparse_assoc \n",
    "\n",
    "gcta64 --bfile quantfamdata --grm-sparse newsparsegrm \\\n",
    "--fastGWA-mlm --pheno phenos.txt --out LMMnewsparse_assoc "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "b998c07f-163c-4036-bcfe-c1e2327c4e86",
   "metadata": {
    "kernel": "R",
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>CHR</th><th scope=col>SNP</th><th scope=col>POS</th><th scope=col>A1</th><th scope=col>A2</th><th scope=col>N</th><th scope=col>AF1</th><th scope=col>BETA</th><th scope=col>SE</th><th scope=col>P</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>rs9326624 </td><td> 558615</td><td>G</td><td>A</td><td>498</td><td>0.0913655</td><td> 0.590324</td><td>0.905413</td><td>0.514405</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1</td><td>rs3094315 </td><td> 742429</td><td>G</td><td>A</td><td>498</td><td>0.3162650</td><td> 0.490011</td><td>0.531356</td><td>0.356429</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1</td><td>rs12562034</td><td> 758311</td><td>A</td><td>G</td><td>495</td><td>0.0818182</td><td> 0.722314</td><td>0.874147</td><td>0.408630</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1</td><td>rs3934834 </td><td> 995669</td><td>A</td><td>G</td><td>498</td><td>0.1817270</td><td> 0.168089</td><td>0.626957</td><td>0.788620</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1</td><td>rs3737728 </td><td>1011278</td><td>A</td><td>G</td><td>498</td><td>0.1957830</td><td> 0.376003</td><td>0.632034</td><td>0.551904</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1</td><td>rs4970405 </td><td>1038818</td><td>G</td><td>A</td><td>498</td><td>0.1194780</td><td>-0.737276</td><td>0.748977</td><td>0.324930</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 10\n",
       "\\begin{tabular}{r|llllllllll}\n",
       "  & CHR & SNP & POS & A1 & A2 & N & AF1 & BETA & SE & P\\\\\n",
       "  & <int> & <chr> & <int> & <chr> & <chr> & <int> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & rs9326624  &  558615 & G & A & 498 & 0.0913655 &  0.590324 & 0.905413 & 0.514405\\\\\n",
       "\t2 & 1 & rs3094315  &  742429 & G & A & 498 & 0.3162650 &  0.490011 & 0.531356 & 0.356429\\\\\n",
       "\t3 & 1 & rs12562034 &  758311 & A & G & 495 & 0.0818182 &  0.722314 & 0.874147 & 0.408630\\\\\n",
       "\t4 & 1 & rs3934834  &  995669 & A & G & 498 & 0.1817270 &  0.168089 & 0.626957 & 0.788620\\\\\n",
       "\t5 & 1 & rs3737728  & 1011278 & A & G & 498 & 0.1957830 &  0.376003 & 0.632034 & 0.551904\\\\\n",
       "\t6 & 1 & rs4970405  & 1038818 & G & A & 498 & 0.1194780 & -0.737276 & 0.748977 & 0.324930\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 10\n",
       "\n",
       "| <!--/--> | CHR &lt;int&gt; | SNP &lt;chr&gt; | POS &lt;int&gt; | A1 &lt;chr&gt; | A2 &lt;chr&gt; | N &lt;int&gt; | AF1 &lt;dbl&gt; | BETA &lt;dbl&gt; | SE &lt;dbl&gt; | P &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | rs9326624  |  558615 | G | A | 498 | 0.0913655 |  0.590324 | 0.905413 | 0.514405 |\n",
       "| 2 | 1 | rs3094315  |  742429 | G | A | 498 | 0.3162650 |  0.490011 | 0.531356 | 0.356429 |\n",
       "| 3 | 1 | rs12562034 |  758311 | A | G | 495 | 0.0818182 |  0.722314 | 0.874147 | 0.408630 |\n",
       "| 4 | 1 | rs3934834  |  995669 | A | G | 498 | 0.1817270 |  0.168089 | 0.626957 | 0.788620 |\n",
       "| 5 | 1 | rs3737728  | 1011278 | A | G | 498 | 0.1957830 |  0.376003 | 0.632034 | 0.551904 |\n",
       "| 6 | 1 | rs4970405  | 1038818 | G | A | 498 | 0.1194780 | -0.737276 | 0.748977 | 0.324930 |\n",
       "\n"
      ],
      "text/plain": [
       "  CHR SNP        POS     A1 A2 N   AF1       BETA      SE       P       \n",
       "1 1   rs9326624   558615 G  A  498 0.0913655  0.590324 0.905413 0.514405\n",
       "2 1   rs3094315   742429 G  A  498 0.3162650  0.490011 0.531356 0.356429\n",
       "3 1   rs12562034  758311 A  G  495 0.0818182  0.722314 0.874147 0.408630\n",
       "4 1   rs3934834   995669 A  G  498 0.1817270  0.168089 0.626957 0.788620\n",
       "5 1   rs3737728  1011278 A  G  498 0.1957830  0.376003 0.632034 0.551904\n",
       "6 1   rs4970405  1038818 G  A  498 0.1194780 -0.737276 0.748977 0.324930"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>CHR</th><th scope=col>SNP</th><th scope=col>POS</th><th scope=col>A1</th><th scope=col>A2</th><th scope=col>N</th><th scope=col>AF1</th><th scope=col>BETA</th><th scope=col>SE</th><th scope=col>P</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>rs9326624 </td><td> 558615</td><td>G</td><td>A</td><td>498</td><td>0.0913655</td><td> 0.590324</td><td>0.905413</td><td>0.514405</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1</td><td>rs3094315 </td><td> 742429</td><td>G</td><td>A</td><td>498</td><td>0.3162650</td><td> 0.490011</td><td>0.531356</td><td>0.356429</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1</td><td>rs12562034</td><td> 758311</td><td>A</td><td>G</td><td>495</td><td>0.0818182</td><td> 0.722314</td><td>0.874147</td><td>0.408630</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1</td><td>rs3934834 </td><td> 995669</td><td>A</td><td>G</td><td>498</td><td>0.1817270</td><td> 0.168089</td><td>0.626957</td><td>0.788620</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1</td><td>rs3737728 </td><td>1011278</td><td>A</td><td>G</td><td>498</td><td>0.1957830</td><td> 0.376003</td><td>0.632034</td><td>0.551904</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1</td><td>rs4970405 </td><td>1038818</td><td>G</td><td>A</td><td>498</td><td>0.1194780</td><td>-0.737276</td><td>0.748977</td><td>0.324930</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 10\n",
       "\\begin{tabular}{r|llllllllll}\n",
       "  & CHR & SNP & POS & A1 & A2 & N & AF1 & BETA & SE & P\\\\\n",
       "  & <int> & <chr> & <int> & <chr> & <chr> & <int> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & rs9326624  &  558615 & G & A & 498 & 0.0913655 &  0.590324 & 0.905413 & 0.514405\\\\\n",
       "\t2 & 1 & rs3094315  &  742429 & G & A & 498 & 0.3162650 &  0.490011 & 0.531356 & 0.356429\\\\\n",
       "\t3 & 1 & rs12562034 &  758311 & A & G & 495 & 0.0818182 &  0.722314 & 0.874147 & 0.408630\\\\\n",
       "\t4 & 1 & rs3934834  &  995669 & A & G & 498 & 0.1817270 &  0.168089 & 0.626957 & 0.788620\\\\\n",
       "\t5 & 1 & rs3737728  & 1011278 & A & G & 498 & 0.1957830 &  0.376003 & 0.632034 & 0.551904\\\\\n",
       "\t6 & 1 & rs4970405  & 1038818 & G & A & 498 & 0.1194780 & -0.737276 & 0.748977 & 0.324930\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 10\n",
       "\n",
       "| <!--/--> | CHR &lt;int&gt; | SNP &lt;chr&gt; | POS &lt;int&gt; | A1 &lt;chr&gt; | A2 &lt;chr&gt; | N &lt;int&gt; | AF1 &lt;dbl&gt; | BETA &lt;dbl&gt; | SE &lt;dbl&gt; | P &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | rs9326624  |  558615 | G | A | 498 | 0.0913655 |  0.590324 | 0.905413 | 0.514405 |\n",
       "| 2 | 1 | rs3094315  |  742429 | G | A | 498 | 0.3162650 |  0.490011 | 0.531356 | 0.356429 |\n",
       "| 3 | 1 | rs12562034 |  758311 | A | G | 495 | 0.0818182 |  0.722314 | 0.874147 | 0.408630 |\n",
       "| 4 | 1 | rs3934834  |  995669 | A | G | 498 | 0.1817270 |  0.168089 | 0.626957 | 0.788620 |\n",
       "| 5 | 1 | rs3737728  | 1011278 | A | G | 498 | 0.1957830 |  0.376003 | 0.632034 | 0.551904 |\n",
       "| 6 | 1 | rs4970405  | 1038818 | G | A | 498 | 0.1194780 | -0.737276 | 0.748977 | 0.324930 |\n",
       "\n"
      ],
      "text/plain": [
       "  CHR SNP        POS     A1 A2 N   AF1       BETA      SE       P       \n",
       "1 1   rs9326624   558615 G  A  498 0.0913655  0.590324 0.905413 0.514405\n",
       "2 1   rs3094315   742429 G  A  498 0.3162650  0.490011 0.531356 0.356429\n",
       "3 1   rs12562034  758311 A  G  495 0.0818182  0.722314 0.874147 0.408630\n",
       "4 1   rs3934834   995669 A  G  498 0.1817270  0.168089 0.626957 0.788620\n",
       "5 1   rs3737728  1011278 A  G  498 0.1957830  0.376003 0.632034 0.551904\n",
       "6 1   rs4970405  1038818 G  A  498 0.1194780 -0.737276 0.748977 0.324930"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>CHR</th><th scope=col>SNP</th><th scope=col>POS</th><th scope=col>A1</th><th scope=col>A2</th><th scope=col>N</th><th scope=col>AF1</th><th scope=col>BETA</th><th scope=col>SE</th><th scope=col>P</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>rs9326624 </td><td> 558615</td><td>G</td><td>A</td><td>498</td><td>0.0913655</td><td> 0.6800840</td><td>0.932956</td><td>0.466028</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1</td><td>rs3094315 </td><td> 742429</td><td>G</td><td>A</td><td>498</td><td>0.3162650</td><td> 0.5165540</td><td>0.547754</td><td>0.345661</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1</td><td>rs12562034</td><td> 758311</td><td>A</td><td>G</td><td>495</td><td>0.0818182</td><td> 0.8136900</td><td>0.900972</td><td>0.366460</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1</td><td>rs3934834 </td><td> 995669</td><td>A</td><td>G</td><td>498</td><td>0.1817270</td><td>-0.0701892</td><td>0.645799</td><td>0.913452</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1</td><td>rs3737728 </td><td>1011278</td><td>A</td><td>G</td><td>498</td><td>0.1957830</td><td> 0.4090710</td><td>0.651214</td><td>0.529894</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1</td><td>rs4970405 </td><td>1038818</td><td>G</td><td>A</td><td>498</td><td>0.1194780</td><td>-0.7021480</td><td>0.772184</td><td>0.363191</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 10\n",
       "\\begin{tabular}{r|llllllllll}\n",
       "  & CHR & SNP & POS & A1 & A2 & N & AF1 & BETA & SE & P\\\\\n",
       "  & <int> & <chr> & <int> & <chr> & <chr> & <int> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & rs9326624  &  558615 & G & A & 498 & 0.0913655 &  0.6800840 & 0.932956 & 0.466028\\\\\n",
       "\t2 & 1 & rs3094315  &  742429 & G & A & 498 & 0.3162650 &  0.5165540 & 0.547754 & 0.345661\\\\\n",
       "\t3 & 1 & rs12562034 &  758311 & A & G & 495 & 0.0818182 &  0.8136900 & 0.900972 & 0.366460\\\\\n",
       "\t4 & 1 & rs3934834  &  995669 & A & G & 498 & 0.1817270 & -0.0701892 & 0.645799 & 0.913452\\\\\n",
       "\t5 & 1 & rs3737728  & 1011278 & A & G & 498 & 0.1957830 &  0.4090710 & 0.651214 & 0.529894\\\\\n",
       "\t6 & 1 & rs4970405  & 1038818 & G & A & 498 & 0.1194780 & -0.7021480 & 0.772184 & 0.363191\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 10\n",
       "\n",
       "| <!--/--> | CHR &lt;int&gt; | SNP &lt;chr&gt; | POS &lt;int&gt; | A1 &lt;chr&gt; | A2 &lt;chr&gt; | N &lt;int&gt; | AF1 &lt;dbl&gt; | BETA &lt;dbl&gt; | SE &lt;dbl&gt; | P &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | rs9326624  |  558615 | G | A | 498 | 0.0913655 |  0.6800840 | 0.932956 | 0.466028 |\n",
       "| 2 | 1 | rs3094315  |  742429 | G | A | 498 | 0.3162650 |  0.5165540 | 0.547754 | 0.345661 |\n",
       "| 3 | 1 | rs12562034 |  758311 | A | G | 495 | 0.0818182 |  0.8136900 | 0.900972 | 0.366460 |\n",
       "| 4 | 1 | rs3934834  |  995669 | A | G | 498 | 0.1817270 | -0.0701892 | 0.645799 | 0.913452 |\n",
       "| 5 | 1 | rs3737728  | 1011278 | A | G | 498 | 0.1957830 |  0.4090710 | 0.651214 | 0.529894 |\n",
       "| 6 | 1 | rs4970405  | 1038818 | G | A | 498 | 0.1194780 | -0.7021480 | 0.772184 | 0.363191 |\n",
       "\n"
      ],
      "text/plain": [
       "  CHR SNP        POS     A1 A2 N   AF1       BETA       SE       P       \n",
       "1 1   rs9326624   558615 G  A  498 0.0913655  0.6800840 0.932956 0.466028\n",
       "2 1   rs3094315   742429 G  A  498 0.3162650  0.5165540 0.547754 0.345661\n",
       "3 1   rs12562034  758311 A  G  495 0.0818182  0.8136900 0.900972 0.366460\n",
       "4 1   rs3934834   995669 A  G  498 0.1817270 -0.0701892 0.645799 0.913452\n",
       "5 1   rs3737728  1011278 A  G  498 0.1957830  0.4090710 0.651214 0.529894\n",
       "6 1   rs4970405  1038818 G  A  498 0.1194780 -0.7021480 0.772184 0.363191"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 10</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>CHR</th><th scope=col>SNP</th><th scope=col>POS</th><th scope=col>A1</th><th scope=col>A2</th><th scope=col>N</th><th scope=col>AF1</th><th scope=col>BETA</th><th scope=col>SE</th><th scope=col>P</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>rs9326624 </td><td> 558615</td><td>G</td><td>A</td><td>498</td><td>0.0913655</td><td> 0.6800840</td><td>0.932956</td><td>0.466028</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1</td><td>rs3094315 </td><td> 742429</td><td>G</td><td>A</td><td>498</td><td>0.3162650</td><td> 0.5165540</td><td>0.547754</td><td>0.345661</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1</td><td>rs12562034</td><td> 758311</td><td>A</td><td>G</td><td>495</td><td>0.0818182</td><td> 0.8136900</td><td>0.900972</td><td>0.366460</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1</td><td>rs3934834 </td><td> 995669</td><td>A</td><td>G</td><td>498</td><td>0.1817270</td><td>-0.0701892</td><td>0.645799</td><td>0.913452</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1</td><td>rs3737728 </td><td>1011278</td><td>A</td><td>G</td><td>498</td><td>0.1957830</td><td> 0.4090710</td><td>0.651214</td><td>0.529894</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1</td><td>rs4970405 </td><td>1038818</td><td>G</td><td>A</td><td>498</td><td>0.1194780</td><td>-0.7021490</td><td>0.772184</td><td>0.363191</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 10\n",
       "\\begin{tabular}{r|llllllllll}\n",
       "  & CHR & SNP & POS & A1 & A2 & N & AF1 & BETA & SE & P\\\\\n",
       "  & <int> & <chr> & <int> & <chr> & <chr> & <int> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & rs9326624  &  558615 & G & A & 498 & 0.0913655 &  0.6800840 & 0.932956 & 0.466028\\\\\n",
       "\t2 & 1 & rs3094315  &  742429 & G & A & 498 & 0.3162650 &  0.5165540 & 0.547754 & 0.345661\\\\\n",
       "\t3 & 1 & rs12562034 &  758311 & A & G & 495 & 0.0818182 &  0.8136900 & 0.900972 & 0.366460\\\\\n",
       "\t4 & 1 & rs3934834  &  995669 & A & G & 498 & 0.1817270 & -0.0701892 & 0.645799 & 0.913452\\\\\n",
       "\t5 & 1 & rs3737728  & 1011278 & A & G & 498 & 0.1957830 &  0.4090710 & 0.651214 & 0.529894\\\\\n",
       "\t6 & 1 & rs4970405  & 1038818 & G & A & 498 & 0.1194780 & -0.7021490 & 0.772184 & 0.363191\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 10\n",
       "\n",
       "| <!--/--> | CHR &lt;int&gt; | SNP &lt;chr&gt; | POS &lt;int&gt; | A1 &lt;chr&gt; | A2 &lt;chr&gt; | N &lt;int&gt; | AF1 &lt;dbl&gt; | BETA &lt;dbl&gt; | SE &lt;dbl&gt; | P &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | rs9326624  |  558615 | G | A | 498 | 0.0913655 |  0.6800840 | 0.932956 | 0.466028 |\n",
       "| 2 | 1 | rs3094315  |  742429 | G | A | 498 | 0.3162650 |  0.5165540 | 0.547754 | 0.345661 |\n",
       "| 3 | 1 | rs12562034 |  758311 | A | G | 495 | 0.0818182 |  0.8136900 | 0.900972 | 0.366460 |\n",
       "| 4 | 1 | rs3934834  |  995669 | A | G | 498 | 0.1817270 | -0.0701892 | 0.645799 | 0.913452 |\n",
       "| 5 | 1 | rs3737728  | 1011278 | A | G | 498 | 0.1957830 |  0.4090710 | 0.651214 | 0.529894 |\n",
       "| 6 | 1 | rs4970405  | 1038818 | G | A | 498 | 0.1194780 | -0.7021490 | 0.772184 | 0.363191 |\n",
       "\n"
      ],
      "text/plain": [
       "  CHR SNP        POS     A1 A2 N   AF1       BETA       SE       P       \n",
       "1 1   rs9326624   558615 G  A  498 0.0913655  0.6800840 0.932956 0.466028\n",
       "2 1   rs3094315   742429 G  A  498 0.3162650  0.5165540 0.547754 0.345661\n",
       "3 1   rs12562034  758311 A  G  495 0.0818182  0.8136900 0.900972 0.366460\n",
       "4 1   rs3934834   995669 A  G  498 0.1817270 -0.0701892 0.645799 0.913452\n",
       "5 1   rs3737728  1011278 A  G  498 0.1957830  0.4090710 0.651214 0.529894\n",
       "6 1   rs4970405  1038818 G  A  498 0.1194780 -0.7021490 0.772184 0.363191"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "res4<-read.table(\"sparse_assoc.fastGWA\", header=T)\n",
    "res5<-read.table(\"newsparse_assoc.fastGWA\", header=T)\n",
    "res6<-read.table(\"LMMsparse_assoc.fastGWA\", header=T)\n",
    "res7<-read.table(\"LMMnewsparse_assoc.fastGWA\", header=T) \n",
    "\n",
    "head(res4)\n",
    "head(res5)\n",
    "head(res6) \n",
    "head(res7)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2e5c0402-e92f-4fcb-bd0e-a659fc026583",
   "metadata": {
    "kernel": "R",
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "1.06102625393192"
      ],
      "text/latex": [
       "1.06102625393192"
      ],
      "text/markdown": [
       "1.06102625393192"
      ],
      "text/plain": [
       "[1] 1.061026"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "1.06102625393192"
      ],
      "text/latex": [
       "1.06102625393192"
      ],
      "text/markdown": [
       "1.06102625393192"
      ],
      "text/plain": [
       "[1] 1.061026"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "1.00438438440012"
      ],
      "text/latex": [
       "1.00438438440012"
      ],
      "text/markdown": [
       "1.00438438440012"
      ],
      "text/plain": [
       "[1] 1.004384"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "1.00438438440012"
      ],
      "text/latex": [
       "1.00438438440012"
      ],
      "text/markdown": [
       "1.00438438440012"
      ],
      "text/plain": [
       "[1] 1.004384"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "chi<-(qchisq(1-res4$P,1))\n",
    "lambda=median(chi)/0.456\n",
    "lambda\n",
    "chi<-(qchisq(1-res5$P,1))\n",
    "lambda=median(chi)/0.456\n",
    "lambda\n",
    "chi<-(qchisq(1-res6$P,1))\n",
    "lambda=median(chi)/0.456\n",
    "lambda\n",
    "chi<-(qchisq(1-res7$P,1))\n",
    "lambda=median(chi)/0.456\n",
    "lambda"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "75201e27-e880-4bb7-a035-073e9cf0944c",
   "metadata": {
    "kernel": "R"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 x 9</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Chr</th><th scope=col>SNP</th><th scope=col>bp</th><th scope=col>A1</th><th scope=col>A2</th><th scope=col>Freq</th><th scope=col>b</th><th scope=col>se</th><th scope=col>p</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>rs9326624 </td><td> 558615</td><td>G</td><td>A</td><td>0.0913655</td><td> 0.693811</td><td>0.924665</td><td>0.453052</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1</td><td>rs3094315 </td><td> 742429</td><td>G</td><td>A</td><td>0.3162650</td><td> 0.494021</td><td>0.550792</td><td>0.369758</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1</td><td>rs12562034</td><td> 758311</td><td>A</td><td>G</td><td>0.0818182</td><td> 0.827582</td><td>0.882891</td><td>0.348576</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1</td><td>rs3934834 </td><td> 995669</td><td>A</td><td>G</td><td>0.1817270</td><td>-0.121518</td><td>0.640661</td><td>0.849563</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1</td><td>rs3737728 </td><td>1011278</td><td>A</td><td>G</td><td>0.1957830</td><td> 0.513895</td><td>0.644486</td><td>0.425236</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1</td><td>rs4970405 </td><td>1038818</td><td>G</td><td>A</td><td>0.1194780</td><td>-0.697621</td><td>0.764350</td><td>0.361401</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 x 9\n",
       "\\begin{tabular}{r|lllllllll}\n",
       "  & Chr & SNP & bp & A1 & A2 & Freq & b & se & p\\\\\n",
       "  & <int> & <chr> & <int> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & rs9326624  &  558615 & G & A & 0.0913655 &  0.693811 & 0.924665 & 0.453052\\\\\n",
       "\t2 & 1 & rs3094315  &  742429 & G & A & 0.3162650 &  0.494021 & 0.550792 & 0.369758\\\\\n",
       "\t3 & 1 & rs12562034 &  758311 & A & G & 0.0818182 &  0.827582 & 0.882891 & 0.348576\\\\\n",
       "\t4 & 1 & rs3934834  &  995669 & A & G & 0.1817270 & -0.121518 & 0.640661 & 0.849563\\\\\n",
       "\t5 & 1 & rs3737728  & 1011278 & A & G & 0.1957830 &  0.513895 & 0.644486 & 0.425236\\\\\n",
       "\t6 & 1 & rs4970405  & 1038818 & G & A & 0.1194780 & -0.697621 & 0.764350 & 0.361401\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 x 9\n",
       "\n",
       "| <!--/--> | Chr &lt;int&gt; | SNP &lt;chr&gt; | bp &lt;int&gt; | A1 &lt;chr&gt; | A2 &lt;chr&gt; | Freq &lt;dbl&gt; | b &lt;dbl&gt; | se &lt;dbl&gt; | p &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | rs9326624  |  558615 | G | A | 0.0913655 |  0.693811 | 0.924665 | 0.453052 |\n",
       "| 2 | 1 | rs3094315  |  742429 | G | A | 0.3162650 |  0.494021 | 0.550792 | 0.369758 |\n",
       "| 3 | 1 | rs12562034 |  758311 | A | G | 0.0818182 |  0.827582 | 0.882891 | 0.348576 |\n",
       "| 4 | 1 | rs3934834  |  995669 | A | G | 0.1817270 | -0.121518 | 0.640661 | 0.849563 |\n",
       "| 5 | 1 | rs3737728  | 1011278 | A | G | 0.1957830 |  0.513895 | 0.644486 | 0.425236 |\n",
       "| 6 | 1 | rs4970405  | 1038818 | G | A | 0.1194780 | -0.697621 | 0.764350 | 0.361401 |\n",
       "\n"
      ],
      "text/plain": [
       "  Chr SNP        bp      A1 A2 Freq      b         se       p       \n",
       "1 1   rs9326624   558615 G  A  0.0913655  0.693811 0.924665 0.453052\n",
       "2 1   rs3094315   742429 G  A  0.3162650  0.494021 0.550792 0.369758\n",
       "3 1   rs12562034  758311 A  G  0.0818182  0.827582 0.882891 0.348576\n",
       "4 1   rs3934834   995669 A  G  0.1817270 -0.121518 0.640661 0.849563\n",
       "5 1   rs3737728  1011278 A  G  0.1957830  0.513895 0.644486 0.425236\n",
       "6 1   rs4970405  1038818 G  A  0.1194780 -0.697621 0.764350 0.361401"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 x 10</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>CHR</th><th scope=col>SNP</th><th scope=col>POS</th><th scope=col>A1</th><th scope=col>A2</th><th scope=col>N</th><th scope=col>AF1</th><th scope=col>BETA</th><th scope=col>SE</th><th scope=col>P</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1</td><td>rs9326624 </td><td> 558615</td><td>G</td><td>A</td><td>498</td><td>0.0913655</td><td> 0.6800840</td><td>0.932956</td><td>0.466028</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1</td><td>rs3094315 </td><td> 742429</td><td>G</td><td>A</td><td>498</td><td>0.3162650</td><td> 0.5165540</td><td>0.547754</td><td>0.345661</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1</td><td>rs12562034</td><td> 758311</td><td>A</td><td>G</td><td>495</td><td>0.0818182</td><td> 0.8136900</td><td>0.900972</td><td>0.366460</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1</td><td>rs3934834 </td><td> 995669</td><td>A</td><td>G</td><td>498</td><td>0.1817270</td><td>-0.0701892</td><td>0.645799</td><td>0.913452</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1</td><td>rs3737728 </td><td>1011278</td><td>A</td><td>G</td><td>498</td><td>0.1957830</td><td> 0.4090710</td><td>0.651214</td><td>0.529894</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1</td><td>rs4970405 </td><td>1038818</td><td>G</td><td>A</td><td>498</td><td>0.1194780</td><td>-0.7021490</td><td>0.772184</td><td>0.363191</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 x 10\n",
       "\\begin{tabular}{r|llllllllll}\n",
       "  & CHR & SNP & POS & A1 & A2 & N & AF1 & BETA & SE & P\\\\\n",
       "  & <int> & <chr> & <int> & <chr> & <chr> & <int> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & 1 & rs9326624  &  558615 & G & A & 498 & 0.0913655 &  0.6800840 & 0.932956 & 0.466028\\\\\n",
       "\t2 & 1 & rs3094315  &  742429 & G & A & 498 & 0.3162650 &  0.5165540 & 0.547754 & 0.345661\\\\\n",
       "\t3 & 1 & rs12562034 &  758311 & A & G & 495 & 0.0818182 &  0.8136900 & 0.900972 & 0.366460\\\\\n",
       "\t4 & 1 & rs3934834  &  995669 & A & G & 498 & 0.1817270 & -0.0701892 & 0.645799 & 0.913452\\\\\n",
       "\t5 & 1 & rs3737728  & 1011278 & A & G & 498 & 0.1957830 &  0.4090710 & 0.651214 & 0.529894\\\\\n",
       "\t6 & 1 & rs4970405  & 1038818 & G & A & 498 & 0.1194780 & -0.7021490 & 0.772184 & 0.363191\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 x 10\n",
       "\n",
       "| <!--/--> | CHR &lt;int&gt; | SNP &lt;chr&gt; | POS &lt;int&gt; | A1 &lt;chr&gt; | A2 &lt;chr&gt; | N &lt;int&gt; | AF1 &lt;dbl&gt; | BETA &lt;dbl&gt; | SE &lt;dbl&gt; | P &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1 | rs9326624  |  558615 | G | A | 498 | 0.0913655 |  0.6800840 | 0.932956 | 0.466028 |\n",
       "| 2 | 1 | rs3094315  |  742429 | G | A | 498 | 0.3162650 |  0.5165540 | 0.547754 | 0.345661 |\n",
       "| 3 | 1 | rs12562034 |  758311 | A | G | 495 | 0.0818182 |  0.8136900 | 0.900972 | 0.366460 |\n",
       "| 4 | 1 | rs3934834  |  995669 | A | G | 498 | 0.1817270 | -0.0701892 | 0.645799 | 0.913452 |\n",
       "| 5 | 1 | rs3737728  | 1011278 | A | G | 498 | 0.1957830 |  0.4090710 | 0.651214 | 0.529894 |\n",
       "| 6 | 1 | rs4970405  | 1038818 | G | A | 498 | 0.1194780 | -0.7021490 | 0.772184 | 0.363191 |\n",
       "\n"
      ],
      "text/plain": [
       "  CHR SNP        POS     A1 A2 N   AF1       BETA       SE       P       \n",
       "1 1   rs9326624   558615 G  A  498 0.0913655  0.6800840 0.932956 0.466028\n",
       "2 1   rs3094315   742429 G  A  498 0.3162650  0.5165540 0.547754 0.345661\n",
       "3 1   rs12562034  758311 A  G  495 0.0818182  0.8136900 0.900972 0.366460\n",
       "4 1   rs3934834   995669 A  G  498 0.1817270 -0.0701892 0.645799 0.913452\n",
       "5 1   rs3737728  1011278 A  G  498 0.1957830  0.4090710 0.651214 0.529894\n",
       "6 1   rs4970405  1038818 G  A  498 0.1194780 -0.7021490 0.772184 0.363191"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "source(\"qqmanHJCupdated.R\") \n",
    "\n",
    "new4<-data.frame(res4$SNP, res4$CHR, res4$POS, res4$P)\n",
    "names(new4)<-c(\"SNP\", \"CHR\", \"BP\", \"P\") \n",
    "\n",
    "png(\"qq4.png\")\n",
    "qq(new4$P)\n",
    "dev.off() \n",
    "\n",
    "png(\"mh4.png\")\n",
    "manhattan(new4, pch=20, suggestiveline=F, genomewideline=F, ymin=2,\n",
    "cex.x.axis=0.65, colors=c(\"black\",\"dodgerblue\"), cex=0.5)\n",
    "dev.off() \n",
    "\n",
    "new5<-data.frame(res5$SNP, res5$CHR, res5$POS, res5$P)\n",
    "names(new5)<-c(\"SNP\", \"CHR\", \"BP\", \"P\") \n",
    "\n",
    "png(\"qq5.png\")\n",
    "qq(new5$P)\n",
    "dev.off() \n",
    "\n",
    "png(\"mh5.png\")\n",
    "manhattan(new5, pch=20, suggestiveline=F, genomewideline=F, ymin=2,\n",
    "cex.x.axis=0.65, colors=c(\"black\",\"dodgerblue\"), cex=0.5)\n",
    "dev.off() \n",
    "\n",
    "new6<-data.frame(res6$SNP, res6$CHR, res6$POS, res6$P)\n",
    "names(new6)<-c(\"SNP\", \"CHR\", \"BP\", \"P\")\n",
    "\n",
    "png(\"qq6.png\")\n",
    "qq(new6$P)\n",
    "dev.off() \n",
    "\n",
    "png(\"mh6.png\")\n",
    "manhattan(new6, pch=20, suggestiveline=F, genomewideline=F, ymin=2,\n",
    "cex.x.axis=0.65, colors=c(\"black\",\"dodgerblue\"), cex=0.5)\n",
    "dev.off() \n",
    "\n",
    "new7<-data.frame(res7$SNP, res7$CHR, res7$POS, res7$P)\n",
    "names(new7)<-c(\"SNP\", \"CHR\", \"BP\", \"P\")\n",
    "\n",
    "png(\"qq7.png\")\n",
    "qq(new7$P)\n",
    "dev.off() \n",
    "\n",
    "png(\"mh7.png\")\n",
    "manhattan(new7, pch=20, suggestiveline=F, genomewideline=F, ymin=2,\n",
    "cex.x.axis=0.65, colors=c(\"black\",\"dodgerblue\"), cex=0.5)\n",
    "dev.off()\n",
    "\n",
    "res3<-read.table(\"GCTAresults.mlma\", header=T)\n",
    "res7<-read.table(\"LMMnewsparse_assoc.fastGWA\", header=T)\n",
    "head(res3)\n",
    "head(res7) \n",
    "\n",
    "png(\"compareGCTAfastGWA.png\")\n",
    "plot(-log10(res3$p),-log10(res7$P))\n",
    "abline(0,1)\n",
    "dev.off()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.3"
  },
  "sos": {
   "kernels": [
    [
     "Bash",
     "calysto_bash",
     "Bash",
     "#E6EEFF",
     "shell"
    ],
    [
     "R",
     "ir",
     "R",
     "#DCDCDA",
     "r"
    ]
   ],
   "version": ""
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
