{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "a3009f0f-0638-498b-953a-0c0d264e1cf6",
   "metadata": {
    "kernel": "Bash",
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.9.0-b.8 64-bit (22 Oct 2024)              cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to logreg.add.log.\n",
      "Options in effect:\n",
      "  --logistic\n",
      "  --map ./data/dbp.map\n",
      "  --out logreg.add\n",
      "  --ped ./data/dbp.cc.ped\n",
      "\n",
      "15485 MB RAM detected; reserving 7742 MB for main workspace.\n",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (20 variants, 600 people).\n",
      "--file: logreg.add-temporary.bed + logreg.add-temporary.bim +\n",
      "logreg.add-temporary.fam written.\n",
      "20 variants loaded from .bim file.\n",
      "600 people (329 males, 271 females) loaded from .fam.\n",
      "600 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 600 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate is 0.988333.\n",
      "20 variants and 600 people pass filters and QC.\n",
      "Among remaining phenotypes, 300 are cases and 300 are controls.\n",
      "Writing logistic model association results to logreg.add.assoc.logistic ...\n",
      "done.\n",
      " CHR      SNP         BP   A1       TEST    NMISS         OR         STAT            P \n",
      "  11   rs1101       1021    1        ADD      600     0.9518      -0.4159       0.6775\n",
      "  11   rs1102       3886    2        ADD      600     0.9728      -0.2034       0.8388\n",
      "  11   rs1103      15023    2        ADD      600     0.9011      -0.8828       0.3774\n",
      "  11   rs1104      15788    1        ADD      508      1.316        1.879      0.06025\n",
      "  11   rs1105      21702    1        ADD      600       1.05       0.4151       0.6781\n",
      "  11   rs1106      23505    2        ADD      552      1.083       0.4735       0.6358\n",
      "  11   rs1107      23508    2        ADD      600      1.155        0.888       0.3745\n",
      "  11   rs1108      28769    2        ADD      600      1.173        1.382        0.167\n",
      "  11   rs1109      31385    1        ADD      600      1.007      0.05903       0.9529\n",
      "  11   rs1110      33198    1        ADD      600      0.936      -0.5453       0.5856\n",
      "  11   rs1111    1245388    2        ADD      600      1.041       0.3465        0.729\n",
      "  11   rs1112    1245604    2        ADD      600      2.149        5.642    1.683e-08\n",
      "  11   rs1113    1246723    2        ADD      600      1.654        3.744    0.0001809\n",
      "  11   rs1114    1246765    2        ADD      600     0.8826       -1.088       0.2767\n",
      "  11   rs1115    1247100    1        ADD      600      1.948        4.889    1.016e-06\n",
      "  11   rs1116    1257557    1        ADD      600      1.046       0.2124       0.8318\n",
      "  11   rs1117    1258119    2        ADD      600      2.224        4.864    1.151e-06\n",
      "  11   rs1118    1258732    2        ADD      600      1.625        3.652    0.0002604\n",
      "  11   rs1119    1259178    1        ADD      600      1.778        3.763    0.0001677\n",
      "  11   rs1120    1259848    1        ADD      600      1.569        3.523    0.0004259\n",
      "PLINK v1.9.0-b.8 64-bit (22 Oct 2024)              cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to logreg.add.ci.log.\n",
      "Options in effect:\n",
      "  --ci 0.95\n",
      "  --logistic\n",
      "  --map ./data/dbp.map\n",
      "  --out logreg.add.ci\n",
      "  --ped ./data/dbp.cc.ped\n",
      "\n",
      "15485 MB RAM detected; reserving 7742 MB for main workspace.\n",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (20 variants, 600 people).\n",
      "--file: logreg.add.ci-temporary.bed + logreg.add.ci-temporary.bim +\n",
      "logreg.add.ci-temporary.fam written.\n",
      "20 variants loaded from .bim file.\n",
      "600 people (329 males, 271 females) loaded from .fam.\n",
      "600 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 600 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate is 0.988333.\n",
      "20 variants and 600 people pass filters and QC.\n",
      "Among remaining phenotypes, 300 are cases and 300 are controls.\n",
      "Writing logistic model association results to logreg.add.ci.assoc.logistic ...\n",
      "done.\n",
      " CHR      SNP         BP   A1       TEST    NMISS         OR       SE      L95      U95         STAT            P \n",
      "  11   rs1101       1021    1        ADD      600     0.9518   0.1189   0.7539    1.201      -0.4159       0.6775\n",
      "  11   rs1102       3886    2        ADD      600     0.9728   0.1356   0.7457    1.269      -0.2034       0.8388\n",
      "  11   rs1103      15023    2        ADD      600     0.9011   0.1179   0.7152    1.135      -0.8828       0.3774\n",
      "  11   rs1104      15788    1        ADD      508      1.316   0.1462   0.9882    1.753        1.879      0.06025\n",
      "  11   rs1105      21702    1        ADD      600       1.05   0.1186   0.8325    1.326       0.4151       0.6781\n",
      "  11   rs1106      23505    2        ADD      552      1.083   0.1693   0.7775     1.51       0.4735       0.6358\n",
      "  11   rs1107      23508    2        ADD      600      1.155    0.162   0.8406    1.586        0.888       0.3745\n",
      "  11   rs1108      28769    2        ADD      600      1.173   0.1156   0.9353    1.472        1.382        0.167\n",
      "  11   rs1109      31385    1        ADD      600      1.007    0.118    0.799    1.269      0.05903       0.9529\n",
      "  11   rs1110      33198    1        ADD      600      0.936   0.1213    0.738    1.187      -0.5453       0.5856\n",
      "  11   rs1111    1245388    2        ADD      600      1.041   0.1155   0.8299    1.305       0.3465        0.729\n",
      "  11   rs1112    1245604    2        ADD      600      2.149   0.1356    1.648    2.804        5.642    1.683e-08\n",
      "  11   rs1113    1246723    2        ADD      600      1.654   0.1344    1.271    2.153        3.744    0.0001809\n",
      "  11   rs1114    1246765    2        ADD      600     0.8826   0.1148   0.7048    1.105       -1.088       0.2767\n",
      "  11   rs1115    1247100    1        ADD      600      1.948   0.1364    1.491    2.545        4.889    1.016e-06\n",
      "  11   rs1116    1257557    1        ADD      600      1.046   0.2125   0.6898    1.587       0.2124       0.8318\n",
      "  11   rs1117    1258119    2        ADD      600      2.224   0.1643    1.612    3.069        4.864    1.151e-06\n",
      "  11   rs1118    1258732    2        ADD      600      1.625    0.133    1.252    2.109        3.652    0.0002604\n",
      "  11   rs1119    1259178    1        ADD      600      1.778   0.1529    1.318    2.399        3.763    0.0001677\n",
      "  11   rs1120    1259848    1        ADD      600      1.569   0.1278    1.221    2.016        3.523    0.0004259\n"
     ]
    }
   ],
   "source": [
    "# I. Logistic regression on a single SNP under an allelic model\n",
    "plink --ped ./data/dbp.cc.ped --map ./data/dbp.map --out logreg.add  --logistic\n",
    "cat logreg.add.assoc.logistic\n",
    "\n",
    "plink --ped ./data/dbp.cc.ped --map ./data/dbp.map --out logreg.add.ci  --logistic  --ci 0.95\n",
    "cat logreg.add.ci.assoc.logistic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "479d430b-fca9-42d7-ae70-5e00c2a8289a",
   "metadata": {
    "kernel": "Bash",
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.9.0-b.8 64-bit (22 Oct 2024)              cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to logreg.age.add.log.\n",
      "Options in effect:\n",
      "  --covar ./data/dbp.age.pheno\n",
      "  --logistic\n",
      "  --map ./data/dbp.map\n",
      "  --out logreg.age.add\n",
      "  --ped ./data/dbp.cc.ped\n",
      "\n",
      "15485 MB RAM detected; reserving 7742 MB for main workspace.\n",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (20 variants, 600 people).\n",
      "--file: logreg.age.add-temporary.bed + logreg.age.add-temporary.bim +\n",
      "logreg.age.add-temporary.fam written.\n",
      "20 variants loaded from .bim file.\n",
      "600 people (329 males, 271 females) loaded from .fam.\n",
      "600 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 1 covariate loaded.\n",
      "Before main variant filters, 600 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate is 0.988333.\n",
      "20 variants and 600 people pass filters and QC.\n",
      "Among remaining phenotypes, 300 are cases and 300 are controls.\n",
      "Writing logistic model association results to logreg.age.add.assoc.logistic ...\n",
      "done.\n",
      " CHR      SNP         BP   A1       TEST    NMISS         OR         STAT            P \n",
      "  11   rs1101       1021    1        ADD      600     0.9506      -0.4262         0.67\n",
      "  11   rs1101       1021    1       COV1      600      1.001       0.3412       0.7329\n",
      "  11   rs1102       3886    2        ADD      600     0.9733      -0.1994       0.8419\n",
      "  11   rs1102       3886    2       COV1      600      1.001       0.3259       0.7445\n",
      "  11   rs1103      15023    2        ADD      600     0.9004      -0.8896       0.3737\n",
      "  11   rs1103      15023    2       COV1      600      1.001       0.3464        0.729\n",
      "  11   rs1104      15788    1        ADD      508      1.326        1.924      0.05441\n",
      "  11   rs1104      15788    1       COV1      508      1.003       0.7744       0.4387\n",
      "  11   rs1105      21702    1        ADD      600      1.052       0.4241       0.6715\n",
      "  11   rs1105      21702    1       COV1      600      1.001       0.3396       0.7341\n",
      "  11   rs1106      23505    2        ADD      552      1.084       0.4753       0.6346\n",
      "  11   rs1106      23505    2       COV1      552      1.002       0.3678        0.713\n",
      "  11   rs1107      23508    2        ADD      600      1.151       0.8654       0.3868\n",
      "  11   rs1107      23508    2       COV1      600      1.001       0.2608       0.7943\n",
      "  11   rs1108      28769    2        ADD      600      1.175        1.394       0.1635\n",
      "  11   rs1108      28769    2       COV1      600      1.001       0.3751       0.7076\n",
      "  11   rs1109      31385    1        ADD      600      1.007      0.05813       0.9536\n",
      "  11   rs1109      31385    1       COV1      600      1.001       0.3282       0.7428\n",
      "  11   rs1110      33198    1        ADD      600     0.9355      -0.5499       0.5824\n",
      "  11   rs1110      33198    1       COV1      600      1.001       0.3359        0.737\n",
      "  11   rs1111    1245388    2        ADD      600      1.042       0.3573       0.7209\n",
      "  11   rs1111    1245388    2       COV1      600      1.001       0.3397       0.7341\n",
      "  11   rs1112    1245604    2        ADD      600      2.149        5.642    1.681e-08\n",
      "  11   rs1112    1245604    2       COV1      600      1.001       0.3289       0.7423\n",
      "  11   rs1113    1246723    2        ADD      600      1.653        3.733    0.0001892\n",
      "  11   rs1113    1246723    2       COV1      600      1.001       0.1346       0.8929\n",
      "  11   rs1114    1246765    2        ADD      600     0.8828       -1.086       0.2777\n",
      "  11   rs1114    1246765    2       COV1      600      1.001       0.3207       0.7485\n",
      "  11   rs1115    1247100    1        ADD      600      1.949        4.892    9.996e-07\n",
      "  11   rs1115    1247100    1       COV1      600      1.001       0.3734       0.7088\n",
      "  11   rs1116    1257557    1        ADD      600      1.048       0.2203       0.8256\n",
      "  11   rs1116    1257557    1       COV1      600      1.001       0.3335       0.7388\n",
      "  11   rs1117    1258119    2        ADD      600      2.227        4.872    1.103e-06\n",
      "  11   rs1117    1258119    2       COV1      600      1.002       0.4376       0.6617\n",
      "  11   rs1118    1258732    2        ADD      600      1.624        3.641    0.0002716\n",
      "  11   rs1118    1258732    2       COV1      600      1.001       0.1575       0.8749\n",
      "  11   rs1119    1259178    1        ADD      600      1.784        3.779    0.0001576\n",
      "  11   rs1119    1259178    1       COV1      600      1.002        0.486       0.6269\n",
      "  11   rs1120    1259848    1        ADD      600       1.57        3.527    0.0004209\n",
      "  11   rs1120    1259848    1       COV1      600      1.001       0.3619       0.7174\n",
      "PLINK v1.9.0-b.8 64-bit (22 Oct 2024)              cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to logreg.sex.add.log.\n",
      "Options in effect:\n",
      "  --logistic\n",
      "  --map ./data/dbp.map\n",
      "  --out logreg.sex.add\n",
      "  --ped ./data/dbp.cc.ped\n",
      "  --sex\n",
      "\n",
      "Note: --sex flag deprecated.  Use e.g. \"--linear sex\".\n",
      "15485 MB RAM detected; reserving 7742 MB for main workspace.\n",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (20 variants, 600 people).\n",
      "--file: logreg.sex.add-temporary.bed + logreg.sex.add-temporary.bim +\n",
      "logreg.sex.add-temporary.fam written.\n",
      "20 variants loaded from .bim file.\n",
      "600 people (329 males, 271 females) loaded from .fam.\n",
      "600 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 600 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate is 0.988333.\n",
      "20 variants and 600 people pass filters and QC.\n",
      "Among remaining phenotypes, 300 are cases and 300 are controls.\n",
      "Writing logistic model association results to logreg.sex.add.assoc.logistic ...\n",
      "done.\n",
      " CHR      SNP         BP   A1       TEST    NMISS         OR         STAT            P \n",
      "  11   rs1101       1021    1        ADD      600     0.9855      -0.1201       0.9044\n",
      "  11   rs1101       1021    1        SEX      600      2.234        4.791    1.663e-06\n",
      "  11   rs1102       3886    2        ADD      600     0.9934     -0.04765        0.962\n",
      "  11   rs1102       3886    2        SEX      600      2.236        4.803    1.567e-06\n",
      "  11   rs1103      15023    2        ADD      600     0.9171      -0.7195       0.4718\n",
      "  11   rs1103      15023    2        SEX      600      2.229        4.781    1.747e-06\n",
      "  11   rs1104      15788    1        ADD      508      1.319        1.858      0.06315\n",
      "  11   rs1104      15788    1        SEX      508      2.247        4.423    9.718e-06\n",
      "  11   rs1105      21702    1        ADD      600      1.028         0.23       0.8181\n",
      "  11   rs1105      21702    1        SEX      600      2.234        4.795    1.629e-06\n",
      "  11   rs1106      23505    2        ADD      552      1.061       0.3429       0.7316\n",
      "  11   rs1106      23505    2        SEX      552      2.325        4.827    1.384e-06\n",
      "  11   rs1107      23508    2        ADD      600      1.259         1.38       0.1675\n",
      "  11   rs1107      23508    2        SEX      600      2.293        4.912    8.999e-07\n",
      "  11   rs1108      28769    2        ADD      600       1.13        1.036       0.3001\n",
      "  11   rs1108      28769    2        SEX      600      2.211        4.723    2.322e-06\n",
      "  11   rs1109      31385    1        ADD      600     0.9622      -0.3193       0.7495\n",
      "  11   rs1109      31385    1        SEX      600      2.246        4.815     1.47e-06\n",
      "  11   rs1110      33198    1        ADD      600     0.9277      -0.6069       0.5439\n",
      "  11   rs1110      33198    1        SEX      600       2.24        4.813    1.485e-06\n",
      "  11   rs1111    1245388    2        ADD      600      1.023       0.1952       0.8452\n",
      "  11   rs1111    1245388    2        SEX      600      2.235        4.798      1.6e-06\n",
      "  11   rs1112    1245604    2        ADD      600      2.163        5.574    2.495e-08\n",
      "  11   rs1112    1245604    2        SEX      600      2.257        4.719    2.373e-06\n",
      "  11   rs1113    1246723    2        ADD      600      1.661        3.705    0.0002117\n",
      "  11   rs1113    1246723    2        SEX      600      2.247        4.773    1.819e-06\n",
      "  11   rs1114    1246765    2        ADD      600     0.8582       -1.303       0.1927\n",
      "  11   rs1114    1246765    2        SEX      600      2.261        4.856      1.2e-06\n",
      "  11   rs1115    1247100    1        ADD      600      1.938        4.758    1.952e-06\n",
      "  11   rs1115    1247100    1        SEX      600      2.221        4.667    3.059e-06\n",
      "  11   rs1116    1257557    1        ADD      600     0.9877     -0.05695       0.9546\n",
      "  11   rs1116    1257557    1        SEX      600      2.238        4.802    1.571e-06\n",
      "  11   rs1117    1258119    2        ADD      600      2.177        4.655    3.239e-06\n",
      "  11   rs1117    1258119    2        SEX      600      2.186        4.581    4.633e-06\n",
      "  11   rs1118    1258732    2        ADD      600      1.633        3.622    0.0002926\n",
      "  11   rs1118    1258732    2        SEX      600      2.249         4.78     1.75e-06\n",
      "  11   rs1119    1259178    1        ADD      600      1.782        3.688    0.0002263\n",
      "  11   rs1119    1259178    1        SEX      600      2.237        4.748    2.051e-06\n",
      "  11   rs1120    1259848    1        ADD      600      1.532         3.28     0.001038\n",
      "  11   rs1120    1259848    1        SEX      600      2.188        4.631    3.643e-06\n",
      "PLINK v1.9.0-b.8 64-bit (22 Oct 2024)              cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to logreg.sexage.add.log.\n",
      "Options in effect:\n",
      "  --covar ./data/dbp.age.pheno\n",
      "  --logistic\n",
      "  --map ./data/dbp.map\n",
      "  --out logreg.sexage.add\n",
      "  --ped ./data/dbp.cc.ped\n",
      "  --sex\n",
      "\n",
      "Note: --sex flag deprecated.  Use e.g. \"--linear sex\".\n",
      "15485 MB RAM detected; reserving 7742 MB for main workspace.\n",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (20 variants, 600 people).\n",
      "--file: logreg.sexage.add-temporary.bed + logreg.sexage.add-temporary.bim +\n",
      "logreg.sexage.add-temporary.fam written.\n",
      "20 variants loaded from .bim file.\n",
      "600 people (329 males, 271 females) loaded from .fam.\n",
      "600 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 1 covariate loaded.\n",
      "Before main variant filters, 600 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate is 0.988333.\n",
      "20 variants and 600 people pass filters and QC.\n",
      "Among remaining phenotypes, 300 are cases and 300 are controls.\n",
      "Writing logistic model association results to logreg.sexage.add.assoc.logistic\n",
      "... done.\n",
      " CHR      SNP         BP   A1       TEST    NMISS         OR         STAT            P \n",
      "  11   rs1101       1021    1        ADD      600     0.9838      -0.1345        0.893\n",
      "  11   rs1101       1021    1       COV1      600      1.002       0.5076       0.6118\n",
      "  11   rs1101       1021    1        SEX      600      2.241        4.804    1.554e-06\n",
      "  11   rs1102       3886    2        ADD      600     0.9943     -0.04129       0.9671\n",
      "  11   rs1102       3886    2       COV1      600      1.002       0.5034       0.6147\n",
      "  11   rs1102       3886    2        SEX      600      2.244        4.817     1.46e-06\n",
      "  11   rs1103      15023    2        ADD      600      0.916      -0.7294       0.4658\n",
      "  11   rs1103      15023    2       COV1      600      1.002       0.5179       0.6045\n",
      "  11   rs1103      15023    2        SEX      600      2.236        4.795    1.627e-06\n",
      "  11   rs1104      15788    1        ADD      508      1.329        1.902      0.05717\n",
      "  11   rs1104      15788    1       COV1      508      1.003       0.7708       0.4408\n",
      "  11   rs1104      15788    1        SEX      508      2.248        4.423    9.749e-06\n",
      "  11   rs1105      21702    1        ADD      600       1.03       0.2433       0.8078\n",
      "  11   rs1105      21702    1       COV1      600      1.002       0.5101         0.61\n",
      "  11   rs1105      21702    1        SEX      600      2.241        4.809     1.52e-06\n",
      "  11   rs1106      23505    2        ADD      552      1.062       0.3449       0.7302\n",
      "  11   rs1106      23505    2       COV1      552      1.002        0.481       0.6305\n",
      "  11   rs1106      23505    2        SEX      552       2.33        4.836    1.323e-06\n",
      "  11   rs1107      23508    2        ADD      600      1.253        1.347       0.1778\n",
      "  11   rs1107      23508    2       COV1      600      1.002       0.4045       0.6859\n",
      "  11   rs1107      23508    2        SEX      600      2.298        4.921    8.601e-07\n",
      "  11   rs1108      28769    2        ADD      600      1.132        1.052       0.2927\n",
      "  11   rs1108      28769    2       COV1      600      1.002       0.5364       0.5917\n",
      "  11   rs1108      28769    2        SEX      600      2.218        4.738    2.163e-06\n",
      "  11   rs1109      31385    1        ADD      600     0.9619      -0.3218       0.7476\n",
      "  11   rs1109      31385    1       COV1      600      1.002       0.5056       0.6132\n",
      "  11   rs1109      31385    1        SEX      600      2.254        4.829     1.37e-06\n",
      "  11   rs1110      33198    1        ADD      600     0.9269      -0.6137       0.5394\n",
      "  11   rs1110      33198    1       COV1      600      1.002       0.5121       0.6086\n",
      "  11   rs1110      33198    1        SEX      600      2.248        4.828    1.382e-06\n",
      "  11   rs1111    1245388    2        ADD      600      1.025       0.2112       0.8327\n",
      "  11   rs1111    1245388    2       COV1      600      1.002       0.5103       0.6098\n",
      "  11   rs1111    1245388    2        SEX      600      2.242        4.812    1.492e-06\n",
      "  11   rs1112    1245604    2        ADD      600      2.163        5.574    2.485e-08\n",
      "  11   rs1112    1245604    2       COV1      600      1.002       0.5076       0.6117\n",
      "  11   rs1112    1245604    2        SEX      600      2.265        4.733     2.21e-06\n",
      "  11   rs1113    1246723    2        ADD      600      1.658        3.685    0.0002287\n",
      "  11   rs1113    1246723    2       COV1      600      1.001       0.3139       0.7536\n",
      "  11   rs1113    1246723    2        SEX      600      2.251         4.78    1.751e-06\n",
      "  11   rs1114    1246765    2        ADD      600     0.8584         -1.3       0.1937\n",
      "  11   rs1114    1246765    2       COV1      600      1.002       0.4967       0.6194\n",
      "  11   rs1114    1246765    2        SEX      600      2.269        4.869     1.12e-06\n",
      "  11   rs1115    1247100    1        ADD      600      1.939        4.762    1.912e-06\n",
      "  11   rs1115    1247100    1       COV1      600      1.002        0.542       0.5878\n",
      "  11   rs1115    1247100    1        SEX      600      2.229        4.682     2.84e-06\n",
      "  11   rs1116    1257557    1        ADD      600     0.9901     -0.04589       0.9634\n",
      "  11   rs1116    1257557    1       COV1      600      1.002       0.5028       0.6151\n",
      "  11   rs1116    1257557    1        SEX      600      2.245        4.816    1.468e-06\n",
      "  11   rs1117    1258119    2        ADD      600      2.181        4.667    3.062e-06\n",
      "  11   rs1117    1258119    2       COV1      600      1.002       0.6017       0.5474\n",
      "  11   rs1117    1258119    2        SEX      600      2.195        4.598    4.267e-06\n",
      "  11   rs1118    1258732    2        ADD      600       1.63        3.603    0.0003141\n",
      "  11   rs1118    1258732    2       COV1      600      1.001       0.3356       0.7371\n",
      "  11   rs1118    1258732    2        SEX      600      2.254        4.789    1.678e-06\n",
      "  11   rs1119    1259178    1        ADD      600      1.789        3.708    0.0002086\n",
      "  11   rs1119    1259178    1       COV1      600      1.003       0.6508       0.5151\n",
      "  11   rs1119    1259178    1        SEX      600      2.246        4.767    1.874e-06\n",
      "  11   rs1120    1259848    1        ADD      600      1.533        3.284     0.001024\n",
      "  11   rs1120    1259848    1       COV1      600      1.002       0.5287        0.597\n",
      "  11   rs1120    1259848    1        SEX      600      2.195        4.646     3.39e-06\n",
      "PLINK v1.9.0-b.8 64-bit (22 Oct 2024)              cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to logreg.snp1112.add.log.\n",
      "Options in effect:\n",
      "  --condition rs1112\n",
      "  --logistic\n",
      "  --map ./data/dbp.map\n",
      "  --out logreg.snp1112.add\n",
      "  --ped ./data/dbp.cc.ped\n",
      "\n",
      "15485 MB RAM detected; reserving 7742 MB for main workspace.\n",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (20 variants, 600 people).\n",
      "--file: logreg.snp1112.add-temporary.bed + logreg.snp1112.add-temporary.bim +\n",
      "logreg.snp1112.add-temporary.fam written.\n",
      "20 variants loaded from .bim file.\n",
      "600 people (329 males, 271 females) loaded from .fam.\n",
      "600 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 600 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate is 0.988333.\n",
      "20 variants and 600 people pass filters and QC.\n",
      "Among remaining phenotypes, 300 are cases and 300 are controls.\n",
      "Writing logistic model association results to logreg.snp1112.add.assoc.logistic\n",
      "... done.\n",
      " CHR      SNP         BP   A1       TEST    NMISS         OR         STAT            P \n",
      "  11   rs1101       1021    1        ADD      600     0.9607      -0.3278       0.7431\n",
      "  11   rs1101       1021    1     rs1112      600      2.149        5.636    1.738e-08\n",
      "  11   rs1102       3886    2        ADD      600     0.9603      -0.2902       0.7717\n",
      "  11   rs1102       3886    2     rs1112      600      2.151        5.646    1.647e-08\n",
      "  11   rs1103      15023    2        ADD      600     0.9089      -0.7873       0.4311\n",
      "  11   rs1103      15023    2     rs1112      600      2.147        5.628    1.819e-08\n",
      "  11   rs1104      15788    1        ADD      508      1.308        1.787      0.07398\n",
      "  11   rs1104      15788    1     rs1112      508      2.186        5.314    1.073e-07\n",
      "  11   rs1105      21702    1        ADD      600      1.052       0.4154       0.6778\n",
      "  11   rs1105      21702    1     rs1112      600       2.15        5.642    1.681e-08\n",
      "  11   rs1106      23505    2        ADD      552      1.128       0.6908       0.4897\n",
      "  11   rs1106      23505    2     rs1112      552      2.175        5.505    3.686e-08\n",
      "  11   rs1107      23508    2        ADD      600      1.213        1.152       0.2493\n",
      "  11   rs1107      23508    2     rs1112      600      2.168        5.679    1.358e-08\n",
      "  11   rs1108      28769    2        ADD      600      1.182        1.401       0.1612\n",
      "  11   rs1108      28769    2     rs1112      600      2.154        5.645    1.656e-08\n",
      "  11   rs1109      31385    1        ADD      600      1.018       0.1466       0.8835\n",
      "  11   rs1109      31385    1     rs1112      600       2.15        5.643    1.667e-08\n",
      "  11   rs1110      33198    1        ADD      600     0.9452       -0.451        0.652\n",
      "  11   rs1110      33198    1     rs1112      600      2.147        5.635    1.753e-08\n",
      "  11   rs1111    1245388    2        ADD      600     0.8861        -0.99       0.3222\n",
      "  11   rs1111    1245388    2     rs1112      600      2.216        5.707    1.149e-08\n",
      "  11   rs1112    1245604    2        ADD      600         NA           NA           NA\n",
      "  11   rs1112    1245604    2     rs1112      600         NA           NA           NA\n",
      "  11   rs1113    1246723    2        ADD      600     0.6134       -1.897       0.0578\n",
      "  11   rs1113    1246723    2     rs1112      600      3.215        4.571    4.853e-06\n",
      "  11   rs1114    1246765    2        ADD      600      0.818       -1.685      0.09197\n",
      "  11   rs1114    1246765    2     rs1112      600      2.201        5.772    7.824e-09\n",
      "  11   rs1115    1247100    1        ADD      600     0.7118      -0.9279       0.3534\n",
      "  11   rs1115    1247100    1     rs1112      600      2.927        2.973     0.002946\n",
      "  11   rs1116    1257557    1        ADD      600      1.075       0.3324       0.7396\n",
      "  11   rs1116    1257557    1     rs1112      600      2.152        5.647    1.637e-08\n",
      "  11   rs1117    1258119    2        ADD      600       1.33        1.242       0.2143\n",
      "  11   rs1117    1258119    2     rs1112      600      1.822        3.186      0.00144\n",
      "  11   rs1118    1258732    2        ADD      600     0.7151       -1.485       0.1375\n",
      "  11   rs1118    1258732    2     rs1112      600      2.798        4.558    5.153e-06\n",
      "  11   rs1119    1259178    1        ADD      600      1.031       0.1516       0.8795\n",
      "  11   rs1119    1259178    1     rs1112      600      2.114        4.303    1.682e-05\n",
      "  11   rs1120    1259848    1        ADD      600     0.8174       -1.043       0.2968\n",
      "  11   rs1120    1259848    1     rs1112      600      2.502        4.578    4.684e-06\n",
      "PLINK v1.9.0-b.8 64-bit (22 Oct 2024)              cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to logreg.snp1117.add.log.\n",
      "Options in effect:\n",
      "  --condition rs1117\n",
      "  --logistic\n",
      "  --map ./data/dbp.map\n",
      "  --out logreg.snp1117.add\n",
      "  --ped ./data/dbp.cc.ped\n",
      "\n",
      "15485 MB RAM detected; reserving 7742 MB for main workspace.\n",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (20 variants, 600 people).\n",
      "--file: logreg.snp1117.add-temporary.bed + logreg.snp1117.add-temporary.bim +\n",
      "logreg.snp1117.add-temporary.fam written.\n",
      "20 variants loaded from .bim file.\n",
      "600 people (329 males, 271 females) loaded from .fam.\n",
      "600 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 600 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate is 0.988333.\n",
      "20 variants and 600 people pass filters and QC.\n",
      "Among remaining phenotypes, 300 are cases and 300 are controls.\n",
      "Writing logistic model association results to logreg.snp1117.add.assoc.logistic\n",
      "... done.\n",
      " CHR      SNP         BP   A1       TEST    NMISS         OR         STAT            P \n",
      "  11   rs1101       1021    1        ADD      600     0.9468      -0.4501       0.6527\n",
      "  11   rs1101       1021    1     rs1117      600      2.226        4.865    1.142e-06\n",
      "  11   rs1102       3886    2        ADD      600     0.9323       -0.504       0.6142\n",
      "  11   rs1102       3886    2     rs1117      600      2.234        4.886    1.028e-06\n",
      "  11   rs1103      15023    2        ADD      600     0.9055      -0.8243       0.4098\n",
      "  11   rs1103      15023    2     rs1117      600      2.223        4.854    1.211e-06\n",
      "  11   rs1104      15788    1        ADD      508      1.267        1.595       0.1106\n",
      "  11   rs1104      15788    1     rs1117      508      2.023        4.029    5.596e-05\n",
      "  11   rs1105      21702    1        ADD      600      1.053       0.4298       0.6674\n",
      "  11   rs1105      21702    1     rs1117      600      2.224        4.866    1.141e-06\n",
      "  11   rs1106      23505    2        ADD      552      1.077       0.4278       0.6688\n",
      "  11   rs1106      23505    2     rs1117      552      2.251        4.762    1.919e-06\n",
      "  11   rs1107      23508    2        ADD      600      1.204        1.117       0.2639\n",
      "  11   rs1107      23508    2     rs1117      600      2.245        4.905    9.332e-07\n",
      "  11   rs1108      28769    2        ADD      600      1.199        1.536       0.1246\n",
      "  11   rs1108      28769    2     rs1117      600      2.246        4.904    9.405e-07\n",
      "  11   rs1109      31385    1        ADD      600      1.016       0.1352       0.8924\n",
      "  11   rs1109      31385    1     rs1117      600      2.224        4.866     1.14e-06\n",
      "  11   rs1110      33198    1        ADD      600     0.9575      -0.3514       0.7253\n",
      "  11   rs1110      33198    1     rs1117      600      2.219        4.848    1.247e-06\n",
      "  11   rs1111    1245388    2        ADD      600      1.094       0.7599       0.4473\n",
      "  11   rs1111    1245388    2     rs1117      600      2.248        4.904    9.375e-07\n",
      "  11   rs1112    1245604    2        ADD      600      1.822        3.186      0.00144\n",
      "  11   rs1112    1245604    2     rs1117      600       1.33        1.242       0.2143\n",
      "  11   rs1113    1246723    2        ADD      600      1.283        1.645          0.1\n",
      "  11   rs1113    1246723    2     rs1117      600      1.936        3.594    0.0003253\n",
      "  11   rs1114    1246765    2        ADD      600      1.024       0.1982       0.8429\n",
      "  11   rs1114    1246765    2     rs1117      600      2.243        4.758    1.956e-06\n",
      "  11   rs1115    1247100    1        ADD      600      1.533        2.619      0.00882\n",
      "  11   rs1115    1247100    1     rs1117      600      1.667        2.594     0.009491\n",
      "  11   rs1116    1257557    1        ADD      600     0.9933     -0.03105       0.9752\n",
      "  11   rs1116    1257557    1     rs1117      600      2.224         4.86    1.172e-06\n",
      "  11   rs1117    1258119    2        ADD      600         NA           NA           NA\n",
      "  11   rs1117    1258119    2     rs1117      600         NA           NA           NA\n",
      "  11   rs1118    1258732    2        ADD      600      1.287        1.717      0.08598\n",
      "  11   rs1118    1258732    2     rs1117      600      1.954        3.721    0.0001984\n",
      "  11   rs1119    1259178    1        ADD      600       1.66        3.259     0.001116\n",
      "  11   rs1119    1259178    1     rs1117      600      2.116        4.497    6.886e-06\n",
      "  11   rs1120    1259848    1        ADD      600      1.277        1.763      0.07797\n",
      "  11   rs1120    1259848    1     rs1117      600      1.979        3.863    0.0001121\n"
     ]
    }
   ],
   "source": [
    "# II. Adjustment for the effects of covariates and of other SNPs\n",
    "plink --ped ./data/dbp.cc.ped --map ./data/dbp.map --out logreg.age.add  --logistic  --covar ./data/dbp.age.pheno\n",
    "cat logreg.age.add.assoc.logistic\n",
    "\n",
    "plink --ped ./data/dbp.cc.ped --map ./data/dbp.map --out logreg.sex.add  --logistic --sex\n",
    "cat logreg.sex.add.assoc.logistic\n",
    "\n",
    "plink --ped ./data/dbp.cc.ped --map ./data/dbp.map --out logreg.sexage.add  --logistic --sex  --covar ./data/dbp.age.pheno\n",
    "cat logreg.sexage.add.assoc.logistic\n",
    "\n",
    "plink --ped ./data/dbp.cc.ped --map ./data/dbp.map --out logreg.snp1112.add  --logistic  --condition rs1112\n",
    "cat logreg.snp1112.add.assoc.logistic\n",
    "\n",
    "plink --ped ./data/dbp.cc.ped --map ./data/dbp.map --out logreg.snp1117.add  --logistic  --condition rs1117\n",
    "cat logreg.snp1117.add.assoc.logistic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "c62e8d14-097d-45ec-bf76-43293fefd6c2",
   "metadata": {
    "kernel": "Bash",
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.9.0-b.8 64-bit (22 Oct 2024)              cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to linreg.sex.add.log.\n",
      "Options in effect:\n",
      "  --linear\n",
      "  --map ./data/dbp.map\n",
      "  --out linreg.sex.add\n",
      "  --ped ./data/dbp.qt.ped\n",
      "  --sex\n",
      "\n",
      "Note: --sex flag deprecated.  Use e.g. \"--linear sex\".\n",
      "15485 MB RAM detected; reserving 7742 MB for main workspace.\n",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (20 variants, 600 people).\n",
      "--file: linreg.sex.add-temporary.bed + linreg.sex.add-temporary.bim +\n",
      "linreg.sex.add-temporary.fam written.\n",
      "20 variants loaded from .bim file.\n",
      "600 people (329 males, 271 females) loaded from .fam.\n",
      "600 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 600 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate is 0.988333.\n",
      "20 variants and 600 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to linreg.sex.add.assoc.linear ...\n",
      "done.\n",
      " CHR      SNP         BP   A1       TEST    NMISS       BETA         STAT            P \n",
      "  11   rs1101       1021    1        ADD      600   -0.05063      -0.1401       0.8887\n",
      "  11   rs1101       1021    1        SEX      600       3.02         6.05    2.553e-09\n",
      "  11   rs1102       3886    2        ADD      600     0.1476       0.3584       0.7201\n",
      "  11   rs1102       3886    2        SEX      600       3.03        6.079    2.152e-09\n",
      "  11   rs1103      15023    2        ADD      600    -0.3833       -1.072        0.284\n",
      "  11   rs1103      15023    2        SEX      600      3.004        6.033     2.83e-09\n",
      "  11   rs1104      15788    1        ADD      508     0.9312        2.119      0.03454\n",
      "  11   rs1104      15788    1        SEX      508      3.085        5.701    2.028e-08\n",
      "  11   rs1105      21702    1        ADD      600     0.3752        1.042       0.2979\n",
      "  11   rs1105      21702    1        SEX      600      3.004         6.03    2.869e-09\n",
      "  11   rs1106      23505    2        ADD      552   -0.03967     -0.07807       0.9378\n",
      "  11   rs1106      23505    2        SEX      552      3.244        6.326    5.221e-10\n",
      "  11   rs1107      23508    2        ADD      600     0.5266         1.07       0.2853\n",
      "  11   rs1107      23508    2        SEX      600      3.075         6.15    1.416e-09\n",
      "  11   rs1108      28769    2        ADD      600     0.1506       0.4292       0.6679\n",
      "  11   rs1108      28769    2        SEX      600      3.008        6.021    3.033e-09\n",
      "  11   rs1109      31385    1        ADD      600     0.1926        0.536       0.5921\n",
      "  11   rs1109      31385    1        SEX      600      3.003        6.013    3.176e-09\n",
      "  11   rs1110      33198    1        ADD      600   -0.01093      -0.0297       0.9763\n",
      "  11   rs1110      33198    1        SEX      600      3.024         6.07     2.27e-09\n",
      "  11   rs1111    1245388    2        ADD      600     0.4724        1.348       0.1781\n",
      "  11   rs1111    1245388    2        SEX      600      3.002        6.033    2.821e-09\n",
      "  11   rs1112    1245604    2        ADD      600      2.868         7.82    2.406e-14\n",
      "  11   rs1112    1245604    2        SEX      600      2.882        6.071    2.268e-09\n",
      "  11   rs1113    1246723    2        ADD      600      1.958        5.048    5.927e-07\n",
      "  11   rs1113    1246723    2        SEX      600      2.966        6.077    2.183e-09\n",
      "  11   rs1114    1246765    2        ADD      600    -0.2197      -0.6312       0.5281\n",
      "  11   rs1114    1246765    2        SEX      600      3.036        6.092    1.995e-09\n",
      "  11   rs1115    1247100    1        ADD      600      2.498        6.586    9.946e-11\n",
      "  11   rs1115    1247100    1        SEX      600       2.87        5.959    4.336e-09\n",
      "  11   rs1116    1257557    1        ADD      600     0.2256       0.3493        0.727\n",
      "  11   rs1116    1257557    1        SEX      600      3.014        6.042    2.675e-09\n",
      "  11   rs1117    1258119    2        ADD      600      3.125        6.889    1.426e-11\n",
      "  11   rs1117    1258119    2        SEX      600      2.804        5.835    8.814e-09\n",
      "  11   rs1118    1258732    2        ADD      600      1.914        4.973    8.633e-07\n",
      "  11   rs1118    1258732    2        SEX      600      2.972        6.087    2.054e-09\n",
      "  11   rs1119    1259178    1        ADD      600      2.097        4.795    2.058e-06\n",
      "  11   rs1119    1259178    1        SEX      600       2.95        6.031    2.858e-09\n",
      "  11   rs1120    1259848    1        ADD      600       1.78        4.782    2.194e-06\n",
      "  11   rs1120    1259848    1        SEX      600      2.869        5.855     7.87e-09\n"
     ]
    }
   ],
   "source": [
    "# III. Analysis of quantitative instead of dichotomized trait\n",
    "plink --ped ./data/dbp.qt.ped --map ./data/dbp.map --out linreg.sex.add  --linear   --sex\n",
    "cat linreg.sex.add.assoc.linear"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "593cb0eb-2bf8-4465-886a-2ba68a6c7ceb",
   "metadata": {
    "kernel": "Bash",
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.9.0-b.8 64-bit (22 Oct 2024)              cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to logreg.sex.inter.add.log.\n",
      "Options in effect:\n",
      "  --interaction\n",
      "  --logistic\n",
      "  --map ./data/dbp.map\n",
      "  --out logreg.sex.inter.add\n",
      "  --ped ./data/dbp.cc.ped\n",
      "  --sex\n",
      "\n",
      "Note: --interaction flag deprecated.  Use e.g. \"--linear interaction\".\n",
      "Note: --sex flag deprecated.  Use e.g. \"--linear sex\".\n",
      "15485 MB RAM detected; reserving 7742 MB for main workspace.\n",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (20 variants, 600 people).\n",
      "--file: logreg.sex.inter.add-temporary.bed + logreg.sex.inter.add-temporary.bim\n",
      "+ logreg.sex.inter.add-temporary.fam written.\n",
      "20 variants loaded from .bim file.\n",
      "600 people (329 males, 271 females) loaded from .fam.\n",
      "600 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 600 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate is 0.988333.\n",
      "20 variants and 600 people pass filters and QC.\n",
      "Among remaining phenotypes, 300 are cases and 300 are controls.\n",
      "Writing logistic model association results to\n",
      "logreg.sex.inter.add.assoc.logistic ... done.\n",
      " CHR      SNP         BP   A1       TEST    NMISS         OR         STAT            P \n",
      "  11   rs1101       1021    1        ADD      600      1.124        0.636       0.5248\n",
      "  11   rs1101       1021    1        SEX      600      2.769         3.62    0.0002948\n",
      "  11   rs1101       1021    1    ADDxSEX      600     0.7912      -0.9564       0.3389\n",
      "  11   rs1102       3886    2        ADD      600       1.22       0.9702        0.332\n",
      "  11   rs1102       3886    2        SEX      600      2.737         4.46    8.201e-06\n",
      "  11   rs1102       3886    2    ADDxSEX      600     0.6871       -1.354       0.1757\n",
      "  11   rs1103      15023    2        ADD      600     0.8432      -0.9292       0.3528\n",
      "  11   rs1103      15023    2        SEX      600      1.941        2.352      0.01867\n",
      "  11   rs1103      15023    2    ADDxSEX      600      1.159       0.6073       0.5436\n",
      "  11   rs1104      15788    1        ADD      508      1.462        1.701      0.08885\n",
      "  11   rs1104      15788    1        SEX      508      2.483        3.717    0.0002019\n",
      "  11   rs1104      15788    1    ADDxSEX      508     0.8308       -0.619       0.5359\n",
      "  11   rs1105      21702    1        ADD      600      1.017      0.09422       0.9249\n",
      "  11   rs1105      21702    1        SEX      600      2.195        2.836     0.004563\n",
      "  11   rs1105      21702    1    ADDxSEX      600      1.019      0.07824       0.9376\n",
      "  11   rs1106      23505    2        ADD      552      1.162       0.5807       0.5615\n",
      "  11   rs1106      23505    2        SEX      552      2.441        4.384    1.163e-05\n",
      "  11   rs1106      23505    2    ADDxSEX      552     0.8494      -0.4706        0.638\n",
      "  11   rs1107      23508    2        ADD      600      1.636        2.106      0.03516\n",
      "  11   rs1107      23508    2        SEX      600      2.711        4.992     5.99e-07\n",
      "  11   rs1107      23508    2    ADDxSEX      600     0.5881       -1.608       0.1079\n",
      "  11   rs1108      28769    2        ADD      600     0.9862     -0.07823       0.9376\n",
      "  11   rs1108      28769    2        SEX      600      1.757        2.014      0.04404\n",
      "  11   rs1108      28769    2    ADDxSEX      600      1.276        1.021       0.3072\n",
      "  11   rs1109      31385    1        ADD      600     0.9181      -0.4721       0.6368\n",
      "  11   rs1109      31385    1        SEX      600      2.087        2.723     0.006467\n",
      "  11   rs1109      31385    1    ADDxSEX      600      1.088       0.3483       0.7276\n",
      "  11   rs1110      33198    1        ADD      600      1.006      0.03105       0.9752\n",
      "  11   rs1110      33198    1        SEX      600      2.566        3.335    0.0008532\n",
      "  11   rs1110      33198    1    ADDxSEX      600     0.8617      -0.5991       0.5491\n",
      "  11   rs1111    1245388    2        ADD      600       1.21        1.106       0.2689\n",
      "  11   rs1111    1245388    2        SEX      600      2.945        4.033    5.506e-05\n",
      "  11   rs1111    1245388    2    ADDxSEX      600     0.7295       -1.334       0.1822\n",
      "  11   rs1112    1245604    2        ADD      600      2.115        3.681    0.0002325\n",
      "  11   rs1112    1245604    2        SEX      600      2.204        3.361    0.0007774\n",
      "  11   rs1112    1245604    2    ADDxSEX      600      1.042       0.1491       0.8815\n",
      "  11   rs1113    1246723    2        ADD      600      1.662        2.534      0.01129\n",
      "  11   rs1113    1246723    2        SEX      600      2.248        3.622    0.0002919\n",
      "  11   rs1113    1246723    2    ADDxSEX      600     0.9991     -0.00342       0.9973\n",
      "  11   rs1114    1246765    2        ADD      600     0.9352      -0.3893       0.6971\n",
      "  11   rs1114    1246765    2        SEX      600      2.584        3.672    0.0002407\n",
      "  11   rs1114    1246765    2    ADDxSEX      600     0.8518      -0.6819       0.4953\n",
      "  11   rs1115    1247100    1        ADD      600      1.822        2.919     0.003514\n",
      "  11   rs1115    1247100    1        SEX      600      2.096        3.329    0.0008719\n",
      "  11   rs1115    1247100    1    ADDxSEX      600       1.12       0.4047       0.6857\n",
      "  11   rs1116    1257557    1        ADD      600      0.619       -1.251       0.2111\n",
      "  11   rs1116    1257557    1        SEX      600      2.016        3.885    0.0001021\n",
      "  11   rs1116    1257557    1    ADDxSEX      600      2.056        1.521       0.1283\n",
      "  11   rs1117    1258119    2        ADD      600       2.26        3.269     0.001078\n",
      "  11   rs1117    1258119    2        SEX      600      2.238        3.905    9.431e-05\n",
      "  11   rs1117    1258119    2    ADDxSEX      600     0.9343      -0.2024       0.8396\n",
      "  11   rs1118    1258732    2        ADD      600      1.544        2.206      0.02739\n",
      "  11   rs1118    1258732    2        SEX      600      2.117        3.293    0.0009904\n",
      "  11   rs1118    1258732    2    ADDxSEX      600      1.113       0.3942       0.6934\n",
      "  11   rs1119    1259178    1        ADD      600      1.622        2.064      0.03902\n",
      "  11   rs1119    1259178    1        SEX      600      2.106        3.671    0.0002416\n",
      "  11   rs1119    1259178    1    ADDxSEX      600      1.185       0.5369       0.5913\n",
      "  11   rs1120    1259848    1        ADD      600      1.391        1.716      0.08623\n",
      "  11   rs1120    1259848    1        SEX      600      1.969        2.961     0.003069\n",
      "  11   rs1120    1259848    1    ADDxSEX      600      1.194       0.6785       0.4974\n",
      "PLINK v1.9.0-b.8 64-bit (22 Oct 2024)              cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to logreg.snp1112.inter.add.log.\n",
      "Options in effect:\n",
      "  --condition rs1112\n",
      "  --interaction\n",
      "  --logistic\n",
      "  --map ./data/dbp.map\n",
      "  --out logreg.snp1112.inter.add\n",
      "  --ped ./data/dbp.cc.ped\n",
      "\n",
      "Note: --interaction flag deprecated.  Use e.g. \"--linear interaction\".\n",
      "15485 MB RAM detected; reserving 7742 MB for main workspace.\n",
      ".ped scan complete (for binary autoconversion).\n",
      "Performing single-pass .bed write (20 variants, 600 people).\n",
      "--file: logreg.snp1112.inter.add-temporary.bed +\n",
      "logreg.snp1112.inter.add-temporary.bim + logreg.snp1112.inter.add-temporary.fam\n",
      "written.\n",
      "20 variants loaded from .bim file.\n",
      "600 people (329 males, 271 females) loaded from .fam.\n",
      "600 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 600 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... done.\n",
      "Total genotyping rate is 0.988333.\n",
      "20 variants and 600 people pass filters and QC.\n",
      "Among remaining phenotypes, 300 are cases and 300 are controls.\n",
      "Writing logistic model association results to\n",
      "logreg.snp1112.inter.add.assoc.logistic ... done.\n",
      " CHR      SNP         BP   A1       TEST    NMISS         OR         STAT            P \n",
      "  11   rs1101       1021    1        ADD      600     0.7121       -2.034      0.04195\n",
      "  11   rs1101       1021    1     rs1112      600      1.377        1.535       0.1247\n",
      "  11   rs1101       1021    1  ADDxCSNP1      600      1.721        2.686     0.007232\n",
      "  11   rs1102       3886    2        ADD      600     0.9292        -0.39       0.6965\n",
      "  11   rs1102       3886    2     rs1112      600      2.083        4.036    5.442e-05\n",
      "  11   rs1102       3886    2  ADDxCSNP1      600       1.06       0.2609       0.7942\n",
      "  11   rs1103      15023    2        ADD      600     0.6364        -2.64     0.008293\n",
      "  11   rs1103      15023    2     rs1112      600      1.276        1.143       0.2529\n",
      "  11   rs1103      15023    2  ADDxCSNP1      600      1.833         3.02      0.00253\n",
      "  11   rs1104      15788    1        ADD      508      1.288         1.26       0.2077\n",
      "  11   rs1104      15788    1     rs1112      508      2.154        3.935    8.335e-05\n",
      "  11   rs1104      15788    1  ADDxCSNP1      508      1.029       0.1163       0.9074\n",
      "  11   rs1105      21702    1        ADD      600      1.442        2.165      0.03039\n",
      "  11   rs1105      21702    1     rs1112      600       3.59        5.344    9.116e-08\n",
      "  11   rs1105      21702    1  ADDxCSNP1      600     0.5797       -2.725     0.006423\n",
      "  11   rs1106      23505    2        ADD      552      1.217       0.8359       0.4032\n",
      "  11   rs1106      23505    2     rs1112      552      2.258        5.036    4.765e-07\n",
      "  11   rs1106      23505    2  ADDxCSNP1      552     0.8713      -0.4816       0.6301\n",
      "  11   rs1107      23508    2        ADD      600      1.096       0.4091       0.6824\n",
      "  11   rs1107      23508    2     rs1112      600      2.049        4.521     6.16e-06\n",
      "  11   rs1107      23508    2  ADDxCSNP1      600      1.204        0.677       0.4984\n",
      "  11   rs1108      28769    2        ADD      600      1.564        2.713      0.00666\n",
      "  11   rs1108      28769    2     rs1112      600      3.494        5.153     2.56e-07\n",
      "  11   rs1108      28769    2  ADDxCSNP1      600     0.6175       -2.506       0.0122\n",
      "  11   rs1109      31385    1        ADD      600      1.447         2.22      0.02641\n",
      "  11   rs1109      31385    1     rs1112      600       3.78        5.668    1.445e-08\n",
      "  11   rs1109      31385    1  ADDxCSNP1      600      0.537       -3.119     0.001812\n",
      "  11   rs1110      33198    1        ADD      600      1.217        1.184       0.2364\n",
      "  11   rs1110      33198    1     rs1112      600      3.355         5.02    5.177e-07\n",
      "  11   rs1110      33198    1  ADDxCSNP1      600     0.6276       -2.325      0.02007\n",
      "  11   rs1111    1245388    2        ADD      600     0.8582      -0.9292       0.3528\n",
      "  11   rs1111    1245388    2     rs1112      600      2.096        3.124     0.001782\n",
      "  11   rs1111    1245388    2  ADDxCSNP1      600      1.059       0.2905       0.7714\n",
      "  11   rs1112    1245604    2        ADD      600         NA           NA           NA\n",
      "  11   rs1112    1245604    2     rs1112      600         NA           NA           NA\n",
      "  11   rs1112    1245604    2  ADDxCSNP1      600         NA           NA           NA\n",
      "  11   rs1113    1246723    2        ADD      600     0.6978       -1.104       0.2696\n",
      "  11   rs1113    1246723    2     rs1112      600      3.471        4.406    1.053e-05\n",
      "  11   rs1113    1246723    2  ADDxCSNP1      600     0.8706      -0.6449        0.519\n",
      "  11   rs1114    1246765    2        ADD      600     0.7975       -1.401       0.1613\n",
      "  11   rs1114    1246765    2     rs1112      600      2.113        3.374    0.0007418\n",
      "  11   rs1114    1246765    2  ADDxCSNP1      600      1.046       0.2333       0.8155\n",
      "  11   rs1115    1247100    1        ADD      600     0.6187       -1.019       0.3082\n",
      "  11   rs1115    1247100    1     rs1112      600      2.866        2.893     0.003811\n",
      "  11   rs1115    1247100    1  ADDxCSNP1      600      1.113       0.4712       0.6375\n",
      "  11   rs1116    1257557    1        ADD      600     0.9862     -0.04721       0.9623\n",
      "  11   rs1116    1257557    1     rs1112      600      2.098        5.058    4.243e-07\n",
      "  11   rs1116    1257557    1  ADDxCSNP1      600      1.176       0.4419       0.6586\n",
      "  11   rs1117    1258119    2        ADD      600      1.451       0.8555       0.3923\n",
      "  11   rs1117    1258119    2     rs1112      600      1.846        3.125     0.001779\n",
      "  11   rs1117    1258119    2  ADDxCSNP1      600     0.9281      -0.2363       0.8132\n",
      "  11   rs1118    1258732    2        ADD      600     0.8964      -0.3916       0.6953\n",
      "  11   rs1118    1258732    2     rs1112      600      3.407        4.543    5.538e-06\n",
      "  11   rs1118    1258732    2  ADDxCSNP1      600     0.7544       -1.371       0.1703\n",
      "  11   rs1119    1259178    1        ADD      600     0.7733      -0.7527       0.4516\n",
      "  11   rs1119    1259178    1     rs1112      600       1.93        3.377    0.0007321\n",
      "  11   rs1119    1259178    1  ADDxCSNP1      600      1.321        1.028       0.3039\n",
      "  11   rs1120    1259848    1        ADD      600     0.9116      -0.3828       0.7018\n",
      "  11   rs1120    1259848    1     rs1112      600      2.811        4.054    5.036e-05\n",
      "  11   rs1120    1259848    1  ADDxCSNP1      600     0.8617      -0.7491       0.4538\n"
     ]
    }
   ],
   "source": [
    "# IV. Gene-environment (GxE) and gene-gene (GxG) interaction\n",
    "plink --ped ./data/dbp.cc.ped --map ./data/dbp.map --out logreg.sex.inter.add  --logistic  --sex  --interaction\n",
    "cat logreg.sex.inter.add.assoc.logistic\n",
    "\n",
    "plink --ped ./data/dbp.cc.ped --map ./data/dbp.map --out logreg.snp1112.inter.add  --logistic  --condition rs1112  --interaction\n",
    "cat logreg.snp1112.inter.add.assoc.logistic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "171be2b3-38d7-486c-b4db-10a71118a07e",
   "metadata": {
    "kernel": "R"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "'dbp'"
      ],
      "text/latex": [
       "'dbp'"
      ],
      "text/markdown": [
       "'dbp'"
      ],
      "text/plain": [
       "[1] \"dbp\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 5 × 28</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>fam</th><th scope=col>pid</th><th scope=col>fid</th><th scope=col>mid</th><th scope=col>sex</th><th scope=col>trait</th><th scope=col>affection</th><th scope=col>age</th><th scope=col>rs1101</th><th scope=col>rs1102</th><th scope=col>⋯</th><th scope=col>rs1111</th><th scope=col>rs1112</th><th scope=col>rs1113</th><th scope=col>rs1114</th><th scope=col>rs1115</th><th scope=col>rs1116</th><th scope=col>rs1117</th><th scope=col>rs1118</th><th scope=col>rs1119</th><th scope=col>rs1120</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>4928</th><td>4928</td><td>1</td><td>0</td><td>0</td><td>1</td><td>85.51</td><td>1</td><td>66</td><td>4</td><td>2</td><td>⋯</td><td>3</td><td>3</td><td>3</td><td>3</td><td>3</td><td>4</td><td>2</td><td>3</td><td>3</td><td>3</td></tr>\n",
       "\t<tr><th scope=row>1838</th><td>1838</td><td>1</td><td>0</td><td>0</td><td>1</td><td>84.51</td><td>1</td><td>67</td><td>2</td><td>2</td><td>⋯</td><td>4</td><td>2</td><td>2</td><td>4</td><td>4</td><td>4</td><td>2</td><td>2</td><td>4</td><td>4</td></tr>\n",
       "\t<tr><th scope=row>2450</th><td>2450</td><td>1</td><td>0</td><td>0</td><td>1</td><td>84.30</td><td>1</td><td>89</td><td>2</td><td>2</td><td>⋯</td><td>4</td><td>3</td><td>3</td><td>4</td><td>3</td><td>4</td><td>2</td><td>3</td><td>3</td><td>3</td></tr>\n",
       "\t<tr><th scope=row>647</th><td> 647</td><td>1</td><td>0</td><td>0</td><td>2</td><td>89.14</td><td>1</td><td>36</td><td>4</td><td>4</td><td>⋯</td><td>2</td><td>3</td><td>3</td><td>2</td><td>3</td><td>4</td><td>3</td><td>3</td><td>4</td><td>3</td></tr>\n",
       "\t<tr><th scope=row>2772</th><td>2772</td><td>1</td><td>0</td><td>0</td><td>1</td><td>90.39</td><td>1</td><td>54</td><td>3</td><td>2</td><td>⋯</td><td>2</td><td>2</td><td>2</td><td>2</td><td>4</td><td>4</td><td>2</td><td>2</td><td>4</td><td>4</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 5 × 28\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & fam & pid & fid & mid & sex & trait & affection & age & rs1101 & rs1102 & ⋯ & rs1111 & rs1112 & rs1113 & rs1114 & rs1115 & rs1116 & rs1117 & rs1118 & rs1119 & rs1120\\\\\n",
       "  & <int> & <int> & <int> & <int> & <fct> & <dbl> & <fct> & <int> & <fct> & <fct> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t4928 & 4928 & 1 & 0 & 0 & 1 & 85.51 & 1 & 66 & 4 & 2 & ⋯ & 3 & 3 & 3 & 3 & 3 & 4 & 2 & 3 & 3 & 3\\\\\n",
       "\t1838 & 1838 & 1 & 0 & 0 & 1 & 84.51 & 1 & 67 & 2 & 2 & ⋯ & 4 & 2 & 2 & 4 & 4 & 4 & 2 & 2 & 4 & 4\\\\\n",
       "\t2450 & 2450 & 1 & 0 & 0 & 1 & 84.30 & 1 & 89 & 2 & 2 & ⋯ & 4 & 3 & 3 & 4 & 3 & 4 & 2 & 3 & 3 & 3\\\\\n",
       "\t647 &  647 & 1 & 0 & 0 & 2 & 89.14 & 1 & 36 & 4 & 4 & ⋯ & 2 & 3 & 3 & 2 & 3 & 4 & 3 & 3 & 4 & 3\\\\\n",
       "\t2772 & 2772 & 1 & 0 & 0 & 1 & 90.39 & 1 & 54 & 3 & 2 & ⋯ & 2 & 2 & 2 & 2 & 4 & 4 & 2 & 2 & 4 & 4\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 5 × 28\n",
       "\n",
       "| <!--/--> | fam &lt;int&gt; | pid &lt;int&gt; | fid &lt;int&gt; | mid &lt;int&gt; | sex &lt;fct&gt; | trait &lt;dbl&gt; | affection &lt;fct&gt; | age &lt;int&gt; | rs1101 &lt;fct&gt; | rs1102 &lt;fct&gt; | ⋯ ⋯ | rs1111 &lt;fct&gt; | rs1112 &lt;fct&gt; | rs1113 &lt;fct&gt; | rs1114 &lt;fct&gt; | rs1115 &lt;fct&gt; | rs1116 &lt;fct&gt; | rs1117 &lt;fct&gt; | rs1118 &lt;fct&gt; | rs1119 &lt;fct&gt; | rs1120 &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 4928 | 4928 | 1 | 0 | 0 | 1 | 85.51 | 1 | 66 | 4 | 2 | ⋯ | 3 | 3 | 3 | 3 | 3 | 4 | 2 | 3 | 3 | 3 |\n",
       "| 1838 | 1838 | 1 | 0 | 0 | 1 | 84.51 | 1 | 67 | 2 | 2 | ⋯ | 4 | 2 | 2 | 4 | 4 | 4 | 2 | 2 | 4 | 4 |\n",
       "| 2450 | 2450 | 1 | 0 | 0 | 1 | 84.30 | 1 | 89 | 2 | 2 | ⋯ | 4 | 3 | 3 | 4 | 3 | 4 | 2 | 3 | 3 | 3 |\n",
       "| 647 |  647 | 1 | 0 | 0 | 2 | 89.14 | 1 | 36 | 4 | 4 | ⋯ | 2 | 3 | 3 | 2 | 3 | 4 | 3 | 3 | 4 | 3 |\n",
       "| 2772 | 2772 | 1 | 0 | 0 | 1 | 90.39 | 1 | 54 | 3 | 2 | ⋯ | 2 | 2 | 2 | 2 | 4 | 4 | 2 | 2 | 4 | 4 |\n",
       "\n"
      ],
      "text/plain": [
       "     fam  pid fid mid sex trait affection age rs1101 rs1102 ⋯ rs1111 rs1112\n",
       "4928 4928 1   0   0   1   85.51 1         66  4      2      ⋯ 3      3     \n",
       "1838 1838 1   0   0   1   84.51 1         67  2      2      ⋯ 4      2     \n",
       "2450 2450 1   0   0   1   84.30 1         89  2      2      ⋯ 4      3     \n",
       "647   647 1   0   0   2   89.14 1         36  4      4      ⋯ 2      3     \n",
       "2772 2772 1   0   0   1   90.39 1         54  3      2      ⋯ 2      2     \n",
       "     rs1113 rs1114 rs1115 rs1116 rs1117 rs1118 rs1119 rs1120\n",
       "4928 3      3      3      4      2      3      3      3     \n",
       "1838 2      4      4      4      2      2      4      4     \n",
       "2450 3      4      3      4      2      3      3      3     \n",
       "647  3      2      3      4      3      3      4      3     \n",
       "2772 2      2      4      4      2      2      4      4     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "      fam            pid         fid         mid    sex         trait       \n",
       " Min.   :   1   Min.   :1   Min.   :0   Min.   :0   1:329   Min.   : 60.50  \n",
       " 1st Qu.:1250   1st Qu.:1   1st Qu.:0   1st Qu.:0   2:271   1st Qu.: 77.44  \n",
       " Median :2600   Median :1   Median :0   Median :0           Median : 82.00  \n",
       " Mean   :2536   Mean   :1   Mean   :0   Mean   :0           Mean   : 81.85  \n",
       " 3rd Qu.:3729   3rd Qu.:1   3rd Qu.:0   3rd Qu.:0           3rd Qu.: 86.09  \n",
       " Max.   :5000   Max.   :1   Max.   :0   Max.   :0           Max.   :101.49  \n",
       " affection      age        rs1101  rs1102  rs1103   rs1104    rs1105 \n",
       " 0:300     Min.   :18.00   2:115   2:317   2:165   2   : 32   2:118  \n",
       " 1:300     1st Qu.:38.00   3:311   3:249   3:309   3   :208   3:311  \n",
       "           Median :55.00   4:174   4: 34   4:126   4   :268   4:171  \n",
       "           Mean   :55.49                           NA's: 92          \n",
       "           3rd Qu.:74.00                                             \n",
       "           Max.   :90.00                                             \n",
       "  rs1106    rs1107  rs1108  rs1109  rs1110  rs1111  rs1112  rs1113  rs1114 \n",
       " 2   :399   2:430   2:163   2:113   2:112   2:194   2:297   2:321   2:210  \n",
       " 3   :141   3:157   3:298   3:305   3:323   3:290   3:251   3:236   3:279  \n",
       " 4   : 12   4: 13   4:139   4:182   4:165   4:116   4: 52   4: 43   4:111  \n",
       " NA's: 48                                                                  \n",
       "                                                                           \n",
       "                                                                           \n",
       " rs1115  rs1116  rs1117  rs1118  rs1119  rs1120 \n",
       " 2: 45   2:  4   2:396   2:305   2: 23   2: 57  \n",
       " 3:227   3: 88   3:190   3:249   3:176   3:249  \n",
       " 4:328   4:508   4: 14   4: 46   4:401   4:294  \n",
       "                                                \n",
       "                                                \n",
       "                                                "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "load (\"./data/dbp.R\")\n",
    "ls ()\n",
    "dbp[1:5,]\n",
    "\n",
    "summary(dbp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "e39e663d-93d7-4d25-9660-9c34981f93aa",
   "metadata": {
    "kernel": "R",
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Call:  glm(formula = affection ~ rs1112, family = binomial(\"logit\"), \n",
      "    data = dbp)\n",
      "\n",
      "Coefficients:\n",
      "(Intercept)      rs11123      rs11124  \n",
      "    -0.4449       0.7582       1.5435  \n",
      "\n",
      "Degrees of Freedom: 599 Total (i.e. Null);  597 Residual\n",
      "Null Deviance:\t    831.8 \n",
      "Residual Deviance: 797.7 \tAIC: 803.7\n",
      "[1] \"glm\" \"lm\" \n",
      "\n",
      "Call:\n",
      "glm(formula = affection ~ rs1112, family = binomial(\"logit\"), \n",
      "    data = dbp)\n",
      "\n",
      "Coefficients:\n",
      "            Estimate Std. Error z value Pr(>|z|)    \n",
      "(Intercept)  -0.4449     0.1189  -3.741 0.000183 ***\n",
      "rs11123       0.7582     0.1746   4.343 1.40e-05 ***\n",
      "rs11124       1.5435     0.3416   4.518 6.24e-06 ***\n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "(Dispersion parameter for binomial family taken to be 1)\n",
      "\n",
      "    Null deviance: 831.78  on 599  degrees of freedom\n",
      "Residual deviance: 797.75  on 597  degrees of freedom\n",
      "AIC: 803.75\n",
      "\n",
      "Number of Fisher Scoring iterations: 4\n",
      "\n",
      "[1] 4.077856e-08\n",
      "              Estimate Std. Error   z value     Pr(>|z|)\n",
      "(Intercept) -0.4449068  0.1189351 -3.740754 1.834691e-04\n",
      "rs11123      0.7582015  0.1745740  4.343154 1.404519e-05\n",
      "rs11124      1.5435191  0.3416277  4.518132 6.238747e-06\n",
      "  rs11123   rs11124 \n",
      "0.7582015 1.5435191 \n",
      " rs11123  rs11124 \n",
      "2.134434 4.681034 \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Waiting for profiling to be done...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                 2.5 %     97.5 %\n",
      "(Intercept) -0.6802726 -0.2135169\n",
      "rs11123      0.4176220  1.1023701\n",
      "rs11124      0.8984800  2.2475097\n",
      "                2.5 %    97.5 %\n",
      "(Intercept) 0.5064789 0.8077385\n",
      "rs11123     1.5183466 3.0112947\n",
      "rs11124     2.4558674 9.4641382\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       " affection rs1112 \n",
       " 0:300     2:297  \n",
       " 1:300     3:251  \n",
       "           4: 52  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       " affection     rs1112      \n",
       " 0:300     Min.   :0.0000  \n",
       " 1:300     1st Qu.:0.0000  \n",
       "           Median :1.0000  \n",
       "           Mean   :0.5917  \n",
       "           3rd Qu.:1.0000  \n",
       "           Max.   :2.0000  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "glm(formula = affection ~ rs1112, family = binomial(\"logit\"), \n",
       "    data = snp.data)\n",
       "\n",
       "Coefficients:\n",
       "            Estimate Std. Error z value Pr(>|z|)    \n",
       "(Intercept)  -0.4470     0.1142  -3.913 9.10e-05 ***\n",
       "rs1112        0.7652     0.1356   5.642 1.68e-08 ***\n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "(Dispersion parameter for binomial family taken to be 1)\n",
       "\n",
       "    Null deviance: 831.78  on 599  degrees of freedom\n",
       "Residual deviance: 797.75  on 598  degrees of freedom\n",
       "AIC: 801.75\n",
       "\n",
       "Number of Fisher Scoring iterations: 4\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis of Deviance Table\n",
      "\n",
      "Model: binomial, link: logit\n",
      "\n",
      "Response: affection\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "\n",
      "       Df Deviance Resid. Df Resid. Dev  Pr(>Chi)    \n",
      "NULL                     599     831.78              \n",
      "rs1112  1   34.026       598     797.75 5.438e-09 ***\n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       " affection     trait        sex          age        rs1112  rs1117 \n",
       " 0:300     Min.   : 60.50   1:329   Min.   :18.00   2:297   2:396  \n",
       " 1:300     1st Qu.: 77.44   2:271   1st Qu.:38.00   3:251   3:190  \n",
       "           Median : 82.00           Median :55.00   4: 52   4: 14  \n",
       "           Mean   : 81.85           Mean   :55.49                  \n",
       "           3rd Qu.: 86.09           3rd Qu.:74.00                  \n",
       "           Max.   :101.49           Max.   :90.00                  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# --- I. Logistic regression on a single SNP genotype --- #\n",
    "\n",
    "# Regression + Wald test #\n",
    "result.snp12 = glm (affection ~ rs1112, family=binomial(\"logit\"), data=dbp)\n",
    "print ( result.snp12 )\n",
    "print ( class (result.snp12) )\n",
    "print ( summary(result.snp12) )\n",
    "\n",
    "# LRT #\n",
    "dev.geno = anova (result.snp12, test=\"Chi\")\n",
    "  lrt.pvalue = pchisq(dev.geno[dim(dev.geno)[1],\"Deviance\"], df=2, ncp=0, FALSE) \n",
    "  print ( lrt.pvalue )\n",
    "\n",
    "# OR + CI #\n",
    "print ( summary(result.snp12)$coefficients )\n",
    "snp.beta = summary(result.snp12)$coefficients[2:3,1]\n",
    "  print ( snp.beta )\n",
    "  print ( exp(snp.beta) )\n",
    "ci = confint (result.snp12)\n",
    "  print (ci) \n",
    "  print ( exp(ci) )\n",
    "\n",
    "# Allelic model #\n",
    "snp.data = dbp[,c(\"affection\", \"rs1112\")]\n",
    "  summary(snp.data)\n",
    "snp.data[,\"rs1112\"] <- as.numeric(snp.data[,\"rs1112\"]) - 1\n",
    "  summary(snp.data)\n",
    "\n",
    "# Results for allelic model (allele 2) #\n",
    "result.all = glm (affection ~ rs1112, family=binomial(\"logit\"), data=snp.data)\n",
    "dev.all    = anova (result.all, test=\"Chi\")\n",
    "summary(result.all)   \n",
    "print(dev.all) \n",
    "\n",
    "\n",
    "# --- Excerpt  --- #\n",
    "snp.data = dbp[,c(\"affection\", \"trait\",\"sex\", \"age\", \"rs1112\", \"rs1117\")]\n",
    "  summary(snp.data)\n",
    "\n",
    "# Allelic model #\n",
    "snp.data[,\"rs1112\"] <- as.numeric(snp.data[,\"rs1112\"]) - 1\n",
    "snp.data[,\"rs1117\"] <- as.numeric(snp.data[,\"rs1117\"]) - 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "22c5133e-9971-4116-9015-ac3628a856a4",
   "metadata": {
    "kernel": "R",
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "glm(formula = affection ~ sex + rs1112, family = binomial(\"logit\"), \n",
       "    data = snp.data)\n",
       "\n",
       "Coefficients:\n",
       "            Estimate Std. Error z value Pr(>|z|)    \n",
       "(Intercept) -0.08386    0.13730  -0.611    0.541    \n",
       "sex2        -0.81412    0.17253  -4.719 2.37e-06 ***\n",
       "rs1112       0.77139    0.13840   5.574 2.49e-08 ***\n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "(Dispersion parameter for binomial family taken to be 1)\n",
       "\n",
       "    Null deviance: 831.78  on 599  degrees of freedom\n",
       "Residual deviance: 774.98  on 597  degrees of freedom\n",
       "AIC: 780.98\n",
       "\n",
       "Number of Fisher Scoring iterations: 4\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "glm(formula = affection ~ age + rs1112, family = binomial(\"logit\"), \n",
       "    data = snp.data)\n",
       "\n",
       "Coefficients:\n",
       "             Estimate Std. Error z value Pr(>|z|)    \n",
       "(Intercept) -0.520422   0.250956  -2.074   0.0381 *  \n",
       "age          0.001322   0.004020   0.329   0.7423    \n",
       "rs1112       0.765189   0.135624   5.642 1.68e-08 ***\n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "(Dispersion parameter for binomial family taken to be 1)\n",
       "\n",
       "    Null deviance: 831.78  on 599  degrees of freedom\n",
       "Residual deviance: 797.64  on 597  degrees of freedom\n",
       "AIC: 803.64\n",
       "\n",
       "Number of Fisher Scoring iterations: 4\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "glm(formula = affection ~ sex + age + rs1112, family = binomial(\"logit\"), \n",
       "    data = snp.data)\n",
       "\n",
       "Coefficients:\n",
       "             Estimate Std. Error z value Pr(>|z|)    \n",
       "(Intercept) -0.198133   0.263732  -0.751    0.452    \n",
       "sex2        -0.817603   0.172736  -4.733 2.21e-06 ***\n",
       "age          0.002084   0.004105   0.508    0.612    \n",
       "rs1112       0.771546   0.138411   5.574 2.48e-08 ***\n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "(Dispersion parameter for binomial family taken to be 1)\n",
       "\n",
       "    Null deviance: 831.78  on 599  degrees of freedom\n",
       "Residual deviance: 774.72  on 596  degrees of freedom\n",
       "AIC: 782.72\n",
       "\n",
       "Number of Fisher Scoring iterations: 4\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "glm(formula = affection ~ rs1117 + rs1112, family = binomial(\"logit\"), \n",
       "    data = snp.data)\n",
       "\n",
       "Coefficients:\n",
       "            Estimate Std. Error z value Pr(>|z|)    \n",
       "(Intercept)  -0.4523     0.1144  -3.955 7.66e-05 ***\n",
       "rs1117        0.2853     0.2297   1.242  0.21431    \n",
       "rs1112        0.5999     0.1883   3.186  0.00144 ** \n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "(Dispersion parameter for binomial family taken to be 1)\n",
       "\n",
       "    Null deviance: 831.78  on 599  degrees of freedom\n",
       "Residual deviance: 796.21  on 597  degrees of freedom\n",
       "AIC: 802.21\n",
       "\n",
       "Number of Fisher Scoring iterations: 4\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A anova: 3 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Df</th><th scope=col>Deviance</th><th scope=col>Resid. Df</th><th scope=col>Resid. Dev</th><th scope=col>Pr(&gt;Chi)</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>NULL</th><td>NA</td><td>      NA</td><td>599</td><td>831.7766</td><td>          NA</td></tr>\n",
       "\t<tr><th scope=row>rs1117</th><td> 1</td><td>25.06376</td><td>598</td><td>806.7129</td><td>5.546551e-07</td></tr>\n",
       "\t<tr><th scope=row>rs1112</th><td> 1</td><td>10.50081</td><td>597</td><td>796.2120</td><td>1.193220e-03</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A anova: 3 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & Df & Deviance & Resid. Df & Resid. Dev & Pr(>Chi)\\\\\n",
       "  & <int> & <dbl> & <int> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tNULL & NA &       NA & 599 & 831.7766 &           NA\\\\\n",
       "\trs1117 &  1 & 25.06376 & 598 & 806.7129 & 5.546551e-07\\\\\n",
       "\trs1112 &  1 & 10.50081 & 597 & 796.2120 & 1.193220e-03\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A anova: 3 × 5\n",
       "\n",
       "| <!--/--> | Df &lt;int&gt; | Deviance &lt;dbl&gt; | Resid. Df &lt;int&gt; | Resid. Dev &lt;dbl&gt; | Pr(&gt;Chi) &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| NULL | NA |       NA | 599 | 831.7766 |           NA |\n",
       "| rs1117 |  1 | 25.06376 | 598 | 806.7129 | 5.546551e-07 |\n",
       "| rs1112 |  1 | 10.50081 | 597 | 796.2120 | 1.193220e-03 |\n",
       "\n"
      ],
      "text/plain": [
       "       Df Deviance Resid. Df Resid. Dev Pr(>Chi)    \n",
       "NULL   NA       NA 599       831.7766             NA\n",
       "rs1117  1 25.06376 598       806.7129   5.546551e-07\n",
       "rs1112  1 10.50081 597       796.2120   1.193220e-03"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "glm(formula = affection ~ rs1112 + rs1117, family = binomial(\"logit\"), \n",
       "    data = snp.data)\n",
       "\n",
       "Coefficients:\n",
       "            Estimate Std. Error z value Pr(>|z|)    \n",
       "(Intercept)  -0.4523     0.1144  -3.955 7.66e-05 ***\n",
       "rs1112        0.5999     0.1883   3.186  0.00144 ** \n",
       "rs1117        0.2853     0.2297   1.242  0.21431    \n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "(Dispersion parameter for binomial family taken to be 1)\n",
       "\n",
       "    Null deviance: 831.78  on 599  degrees of freedom\n",
       "Residual deviance: 796.21  on 597  degrees of freedom\n",
       "AIC: 802.21\n",
       "\n",
       "Number of Fisher Scoring iterations: 4\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A anova: 3 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Df</th><th scope=col>Deviance</th><th scope=col>Resid. Df</th><th scope=col>Resid. Dev</th><th scope=col>Pr(&gt;Chi)</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>NULL</th><td>NA</td><td>       NA</td><td>599</td><td>831.7766</td><td>          NA</td></tr>\n",
       "\t<tr><th scope=row>rs1112</th><td> 1</td><td>34.026174</td><td>598</td><td>797.7504</td><td>5.437568e-09</td></tr>\n",
       "\t<tr><th scope=row>rs1117</th><td> 1</td><td> 1.538401</td><td>597</td><td>796.2120</td><td>2.148560e-01</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A anova: 3 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & Df & Deviance & Resid. Df & Resid. Dev & Pr(>Chi)\\\\\n",
       "  & <int> & <dbl> & <int> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tNULL & NA &        NA & 599 & 831.7766 &           NA\\\\\n",
       "\trs1112 &  1 & 34.026174 & 598 & 797.7504 & 5.437568e-09\\\\\n",
       "\trs1117 &  1 &  1.538401 & 597 & 796.2120 & 2.148560e-01\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A anova: 3 × 5\n",
       "\n",
       "| <!--/--> | Df &lt;int&gt; | Deviance &lt;dbl&gt; | Resid. Df &lt;int&gt; | Resid. Dev &lt;dbl&gt; | Pr(&gt;Chi) &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| NULL | NA |        NA | 599 | 831.7766 |           NA |\n",
       "| rs1112 |  1 | 34.026174 | 598 | 797.7504 | 5.437568e-09 |\n",
       "| rs1117 |  1 |  1.538401 | 597 | 796.2120 | 2.148560e-01 |\n",
       "\n"
      ],
      "text/plain": [
       "       Df Deviance  Resid. Df Resid. Dev Pr(>Chi)    \n",
       "NULL   NA        NA 599       831.7766             NA\n",
       "rs1112  1 34.026174 598       797.7504   5.437568e-09\n",
       "rs1117  1  1.538401 597       796.2120   2.148560e-01"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# --- II. Adjustment for the effects of covariates and of other SNPs --- #\n",
    "\n",
    "# Covariates #\n",
    "result.adj = glm (affection ~ sex + rs1112      , family=binomial(\"logit\"), data=snp.data)\n",
    "  summary(result.adj)\n",
    "result.adj = glm (affection ~ age + rs1112      , family=binomial(\"logit\"), data=snp.data)\n",
    "  summary(result.adj)\n",
    "result.adj = glm (affection ~ sex + age + rs1112, family=binomial(\"logit\"), data=snp.data)\n",
    "  summary(result.adj)\n",
    "\n",
    "\n",
    "# Effect of SNP2 given SNP 1 # \n",
    "result.adj = glm (affection ~ rs1117 + rs1112, family=binomial(\"logit\"), data=snp.data)\n",
    "  summary(result.adj)\n",
    "  anova (result.adj, test=\"Chi\")\n",
    "result.adj = glm (affection ~ rs1112 + rs1117, family=binomial(\"logit\"), data=snp.data)\n",
    "  summary(result.adj)\n",
    "  anova (result.adj, test=\"Chi\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "2f090b46-61b7-4a60-a78c-330cba6664ae",
   "metadata": {
    "kernel": "R",
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "lm(formula = trait ~ rs1112, data = snp.data)\n",
       "\n",
       "Residuals:\n",
       "     Min       1Q   Median       3Q      Max \n",
       "-22.5556  -3.9106   0.2194   4.0144  15.4809 \n",
       "\n",
       "Coefficients:\n",
       "            Estimate Std. Error t value Pr(>|t|)    \n",
       "(Intercept)  80.1021     0.3301 242.680  < 2e-16 ***\n",
       "rs1112        2.9535     0.3774   7.826 2.29e-14 ***\n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "Residual standard error: 5.954 on 598 degrees of freedom\n",
       "Multiple R-squared:  0.09291,\tAdjusted R-squared:  0.09139 \n",
       "F-statistic: 61.25 on 1 and 598 DF,  p-value: 2.292e-14\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "lm(formula = trait ~ sex + rs1112, data = snp.data)\n",
       "\n",
       "Residuals:\n",
       "     Min       1Q   Median       3Q      Max \n",
       "-20.9404  -3.6272   0.2234   3.7815  16.3480 \n",
       "\n",
       "Coefficients:\n",
       "            Estimate Std. Error t value Pr(>|t|)    \n",
       "(Intercept)  81.4542     0.3904 208.654  < 2e-16 ***\n",
       "sex2         -2.8823     0.4748  -6.071 2.27e-09 ***\n",
       "rs1112        2.8685     0.3668   7.820 2.41e-14 ***\n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "Residual standard error: 5.784 on 597 degrees of freedom\n",
       "Multiple R-squared:  0.1456,\tAdjusted R-squared:  0.1428 \n",
       "F-statistic: 50.89 on 2 and 597 DF,  p-value: < 2.2e-16\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# --- III. Analysis of quantitative instead of dichotomized trait --- #\n",
    "\n",
    "# Consideration of QT instead of binary affection status (dichotomized QT) #\n",
    "result.adj = lm (trait ~ rs1112      , data=snp.data)\n",
    "  summary(result.adj)\n",
    "result.adj = lm (trait ~ sex + rs1112, data=snp.data)\n",
    "  summary(result.adj)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "3fe03329-a2fc-4ade-8e5a-22b1232c2ea5",
   "metadata": {
    "kernel": "R",
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "glm(formula = affection ~ sex * rs1112, family = binomial(\"logit\"), \n",
       "    data = snp.data)\n",
       "\n",
       "Coefficients:\n",
       "            Estimate Std. Error z value Pr(>|z|)    \n",
       "(Intercept) -0.09415    0.15371  -0.613 0.540174    \n",
       "sex2        -0.79026    0.23515  -3.361 0.000777 ***\n",
       "rs1112       0.79049    0.18896   4.183 2.87e-05 ***\n",
       "sex2:rs1112 -0.04141    0.27771  -0.149 0.881472    \n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "(Dispersion parameter for binomial family taken to be 1)\n",
       "\n",
       "    Null deviance: 831.78  on 599  degrees of freedom\n",
       "Residual deviance: 774.96  on 596  degrees of freedom\n",
       "AIC: 782.96\n",
       "\n",
       "Number of Fisher Scoring iterations: 4\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "glm(formula = affection ~ age * rs1112, family = binomial(\"logit\"), \n",
       "    data = snp.data)\n",
       "\n",
       "Coefficients:\n",
       "             Estimate Std. Error z value Pr(>|z|)   \n",
       "(Intercept) -0.764365   0.328207  -2.329  0.01986 * \n",
       "age          0.005719   0.005508   1.038  0.29909   \n",
       "rs1112       1.193715   0.393377   3.035  0.00241 **\n",
       "age:rs1112  -0.007716   0.006585  -1.172  0.24130   \n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "(Dispersion parameter for binomial family taken to be 1)\n",
       "\n",
       "    Null deviance: 831.78  on 599  degrees of freedom\n",
       "Residual deviance: 796.26  on 596  degrees of freedom\n",
       "AIC: 804.26\n",
       "\n",
       "Number of Fisher Scoring iterations: 4\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "glm(formula = affection ~ rs1112 * rs1117, family = binomial(\"logit\"), \n",
       "    data = snp.data)\n",
       "\n",
       "Coefficients:\n",
       "              Estimate Std. Error z value Pr(>|z|)    \n",
       "(Intercept)   -0.45855    0.11749  -3.903  9.5e-05 ***\n",
       "rs1112         0.61285    0.19612   3.125  0.00178 ** \n",
       "rs1117         0.37232    0.43522   0.855  0.39228    \n",
       "rs1112:rs1117 -0.07464    0.31590  -0.236  0.81323    \n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "(Dispersion parameter for binomial family taken to be 1)\n",
       "\n",
       "    Null deviance: 831.78  on 599  degrees of freedom\n",
       "Residual deviance: 796.16  on 596  degrees of freedom\n",
       "AIC: 804.16\n",
       "\n",
       "Number of Fisher Scoring iterations: 4\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# --- IV. Gene-environment (GxE) and gene-gene (GxG) interaction --- #\n",
    "\n",
    "# GxE #\n",
    "result.inter = glm (affection ~ sex * rs1112, family=binomial(\"logit\"), data=snp.data)\n",
    "  summary(result.inter)\n",
    "result.inter = glm (affection ~ age * rs1112, family=binomial(\"logit\"), data=snp.data)\n",
    "  summary(result.inter)\n",
    "\n",
    "# GxG #\n",
    "result.inter = glm (affection ~ rs1112 * rs1117, family=binomial(\"logit\"), data=snp.data)\n",
    "  summary(result.inter)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "987393f8-c492-4089-8c8f-bf7a1874883e",
   "metadata": {
    "kernel": "R",
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "glm(formula = affection ~ sex + age + rs1112 + rs1117, family = binomial(\"logit\"), \n",
       "    data = snp.data)\n",
       "\n",
       "Coefficients:\n",
       "             Estimate Std. Error z value Pr(>|z|)    \n",
       "(Intercept) -0.211378   0.264220  -0.800  0.42371    \n",
       "sex2        -0.808941   0.173019  -4.675 2.93e-06 ***\n",
       "age          0.002183   0.004109   0.531  0.59519    \n",
       "rs1112       0.635080   0.193437   3.283  0.00103 ** \n",
       "rs1117       0.233915   0.235438   0.994  0.32045    \n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "(Dispersion parameter for binomial family taken to be 1)\n",
       "\n",
       "    Null deviance: 831.78  on 599  degrees of freedom\n",
       "Residual deviance: 773.74  on 595  degrees of freedom\n",
       "AIC: 783.74\n",
       "\n",
       "Number of Fisher Scoring iterations: 4\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Start:  AIC=783.74\n",
      "affection ~ sex + age + rs1112 + rs1117\n",
      "\n",
      "         Df Deviance    AIC\n",
      "- age     1   774.02 782.02\n",
      "- rs1117  1   774.72 782.72\n",
      "<none>        773.74 783.74\n",
      "- rs1112  1   784.97 792.97\n",
      "- sex     1   796.08 804.08\n",
      "\n",
      "Step:  AIC=782.02\n",
      "affection ~ sex + rs1112 + rs1117\n",
      "\n",
      "         Df Deviance    AIC\n",
      "- rs1117  1   774.98 780.98\n",
      "<none>        774.02 782.02\n",
      "- rs1112  1   785.34 791.34\n",
      "- sex     1   796.21 802.21\n",
      "\n",
      "Step:  AIC=780.98\n",
      "affection ~ sex + rs1112\n",
      "\n",
      "         Df Deviance    AIC\n",
      "<none>        774.98 780.98\n",
      "- sex     1   797.75 801.75\n",
      "- rs1112  1   808.19 812.19\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "glm(formula = affection ~ sex + rs1112, family = binomial(\"logit\"), \n",
       "    data = snp.data)\n",
       "\n",
       "Coefficients:\n",
       "            Estimate Std. Error z value Pr(>|z|)    \n",
       "(Intercept) -0.08386    0.13730  -0.611    0.541    \n",
       "sex2        -0.81412    0.17253  -4.719 2.37e-06 ***\n",
       "rs1112       0.77139    0.13840   5.574 2.49e-08 ***\n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "(Dispersion parameter for binomial family taken to be 1)\n",
       "\n",
       "    Null deviance: 831.78  on 599  degrees of freedom\n",
       "Residual deviance: 774.98  on 597  degrees of freedom\n",
       "AIC: 780.98\n",
       "\n",
       "Number of Fisher Scoring iterations: 4\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# --- V. Model selection --- #\n",
    "\n",
    "result.reg = glm (affection ~ sex + age + rs1112 + rs1117, family=binomial(\"logit\"), data=snp.data)\n",
    "  summary(result.reg)\n",
    "\n",
    "modelchoice.result <- step (result.reg)\n",
    "  summary(modelchoice.result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "13e3669e-8102-4b16-b0ff-3c2148837260",
   "metadata": {
    "kernel": "R"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'ci'</li><li>'dbp'</li><li>'dev.all'</li><li>'dev.geno'</li><li>'lrt.pvalue'</li><li>'modelchoice.result'</li><li>'result.adj'</li><li>'result.all'</li><li>'result.inter'</li><li>'result.reg'</li><li>'result.snp12'</li><li>'snp.beta'</li><li>'snp.data'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'ci'\n",
       "\\item 'dbp'\n",
       "\\item 'dev.all'\n",
       "\\item 'dev.geno'\n",
       "\\item 'lrt.pvalue'\n",
       "\\item 'modelchoice.result'\n",
       "\\item 'result.adj'\n",
       "\\item 'result.all'\n",
       "\\item 'result.inter'\n",
       "\\item 'result.reg'\n",
       "\\item 'result.snp12'\n",
       "\\item 'snp.beta'\n",
       "\\item 'snp.data'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'ci'\n",
       "2. 'dbp'\n",
       "3. 'dev.all'\n",
       "4. 'dev.geno'\n",
       "5. 'lrt.pvalue'\n",
       "6. 'modelchoice.result'\n",
       "7. 'result.adj'\n",
       "8. 'result.all'\n",
       "9. 'result.inter'\n",
       "10. 'result.reg'\n",
       "11. 'result.snp12'\n",
       "12. 'snp.beta'\n",
       "13. 'snp.data'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"ci\"                 \"dbp\"                \"dev.all\"           \n",
       " [4] \"dev.geno\"           \"lrt.pvalue\"         \"modelchoice.result\"\n",
       " [7] \"result.adj\"         \"result.all\"         \"result.inter\"      \n",
       "[10] \"result.reg\"         \"result.snp12\"       \"snp.beta\"          \n",
       "[13] \"snp.data\"          "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 5 × 28</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>fam</th><th scope=col>pid</th><th scope=col>fid</th><th scope=col>mid</th><th scope=col>sex</th><th scope=col>trait</th><th scope=col>affection</th><th scope=col>age</th><th scope=col>rs1101</th><th scope=col>rs1102</th><th scope=col>⋯</th><th scope=col>rs1111</th><th scope=col>rs1112</th><th scope=col>rs1113</th><th scope=col>rs1114</th><th scope=col>rs1115</th><th scope=col>rs1116</th><th scope=col>rs1117</th><th scope=col>rs1118</th><th scope=col>rs1119</th><th scope=col>rs1120</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>⋯</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>4928</th><td>4928</td><td>1</td><td>0</td><td>0</td><td>1</td><td>85.51</td><td>1</td><td>66</td><td>4</td><td>2</td><td>⋯</td><td>3</td><td>3</td><td>3</td><td>3</td><td>3</td><td>4</td><td>2</td><td>3</td><td>3</td><td>3</td></tr>\n",
       "\t<tr><th scope=row>1838</th><td>1838</td><td>1</td><td>0</td><td>0</td><td>1</td><td>84.51</td><td>1</td><td>67</td><td>2</td><td>2</td><td>⋯</td><td>4</td><td>2</td><td>2</td><td>4</td><td>4</td><td>4</td><td>2</td><td>2</td><td>4</td><td>4</td></tr>\n",
       "\t<tr><th scope=row>2450</th><td>2450</td><td>1</td><td>0</td><td>0</td><td>1</td><td>84.30</td><td>1</td><td>89</td><td>2</td><td>2</td><td>⋯</td><td>4</td><td>3</td><td>3</td><td>4</td><td>3</td><td>4</td><td>2</td><td>3</td><td>3</td><td>3</td></tr>\n",
       "\t<tr><th scope=row>647</th><td> 647</td><td>1</td><td>0</td><td>0</td><td>2</td><td>89.14</td><td>1</td><td>36</td><td>4</td><td>4</td><td>⋯</td><td>2</td><td>3</td><td>3</td><td>2</td><td>3</td><td>4</td><td>3</td><td>3</td><td>4</td><td>3</td></tr>\n",
       "\t<tr><th scope=row>2772</th><td>2772</td><td>1</td><td>0</td><td>0</td><td>1</td><td>90.39</td><td>1</td><td>54</td><td>3</td><td>2</td><td>⋯</td><td>2</td><td>2</td><td>2</td><td>2</td><td>4</td><td>4</td><td>2</td><td>2</td><td>4</td><td>4</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 5 × 28\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & fam & pid & fid & mid & sex & trait & affection & age & rs1101 & rs1102 & ⋯ & rs1111 & rs1112 & rs1113 & rs1114 & rs1115 & rs1116 & rs1117 & rs1118 & rs1119 & rs1120\\\\\n",
       "  & <int> & <int> & <int> & <int> & <fct> & <dbl> & <fct> & <int> & <fct> & <fct> & ⋯ & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct> & <fct>\\\\\n",
       "\\hline\n",
       "\t4928 & 4928 & 1 & 0 & 0 & 1 & 85.51 & 1 & 66 & 4 & 2 & ⋯ & 3 & 3 & 3 & 3 & 3 & 4 & 2 & 3 & 3 & 3\\\\\n",
       "\t1838 & 1838 & 1 & 0 & 0 & 1 & 84.51 & 1 & 67 & 2 & 2 & ⋯ & 4 & 2 & 2 & 4 & 4 & 4 & 2 & 2 & 4 & 4\\\\\n",
       "\t2450 & 2450 & 1 & 0 & 0 & 1 & 84.30 & 1 & 89 & 2 & 2 & ⋯ & 4 & 3 & 3 & 4 & 3 & 4 & 2 & 3 & 3 & 3\\\\\n",
       "\t647 &  647 & 1 & 0 & 0 & 2 & 89.14 & 1 & 36 & 4 & 4 & ⋯ & 2 & 3 & 3 & 2 & 3 & 4 & 3 & 3 & 4 & 3\\\\\n",
       "\t2772 & 2772 & 1 & 0 & 0 & 1 & 90.39 & 1 & 54 & 3 & 2 & ⋯ & 2 & 2 & 2 & 2 & 4 & 4 & 2 & 2 & 4 & 4\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 5 × 28\n",
       "\n",
       "| <!--/--> | fam &lt;int&gt; | pid &lt;int&gt; | fid &lt;int&gt; | mid &lt;int&gt; | sex &lt;fct&gt; | trait &lt;dbl&gt; | affection &lt;fct&gt; | age &lt;int&gt; | rs1101 &lt;fct&gt; | rs1102 &lt;fct&gt; | ⋯ ⋯ | rs1111 &lt;fct&gt; | rs1112 &lt;fct&gt; | rs1113 &lt;fct&gt; | rs1114 &lt;fct&gt; | rs1115 &lt;fct&gt; | rs1116 &lt;fct&gt; | rs1117 &lt;fct&gt; | rs1118 &lt;fct&gt; | rs1119 &lt;fct&gt; | rs1120 &lt;fct&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 4928 | 4928 | 1 | 0 | 0 | 1 | 85.51 | 1 | 66 | 4 | 2 | ⋯ | 3 | 3 | 3 | 3 | 3 | 4 | 2 | 3 | 3 | 3 |\n",
       "| 1838 | 1838 | 1 | 0 | 0 | 1 | 84.51 | 1 | 67 | 2 | 2 | ⋯ | 4 | 2 | 2 | 4 | 4 | 4 | 2 | 2 | 4 | 4 |\n",
       "| 2450 | 2450 | 1 | 0 | 0 | 1 | 84.30 | 1 | 89 | 2 | 2 | ⋯ | 4 | 3 | 3 | 4 | 3 | 4 | 2 | 3 | 3 | 3 |\n",
       "| 647 |  647 | 1 | 0 | 0 | 2 | 89.14 | 1 | 36 | 4 | 4 | ⋯ | 2 | 3 | 3 | 2 | 3 | 4 | 3 | 3 | 4 | 3 |\n",
       "| 2772 | 2772 | 1 | 0 | 0 | 1 | 90.39 | 1 | 54 | 3 | 2 | ⋯ | 2 | 2 | 2 | 2 | 4 | 4 | 2 | 2 | 4 | 4 |\n",
       "\n"
      ],
      "text/plain": [
       "     fam  pid fid mid sex trait affection age rs1101 rs1102 ⋯ rs1111 rs1112\n",
       "4928 4928 1   0   0   1   85.51 1         66  4      2      ⋯ 3      3     \n",
       "1838 1838 1   0   0   1   84.51 1         67  2      2      ⋯ 4      2     \n",
       "2450 2450 1   0   0   1   84.30 1         89  2      2      ⋯ 4      3     \n",
       "647   647 1   0   0   2   89.14 1         36  4      4      ⋯ 2      3     \n",
       "2772 2772 1   0   0   1   90.39 1         54  3      2      ⋯ 2      2     \n",
       "     rs1113 rs1114 rs1115 rs1116 rs1117 rs1118 rs1119 rs1120\n",
       "4928 3      3      3      4      2      3      3      3     \n",
       "1838 2      4      4      4      2      2      4      4     \n",
       "2450 3      4      3      4      2      3      3      3     \n",
       "647  3      2      3      4      3      3      4      3     \n",
       "2772 2      2      4      4      2      2      4      4     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "# Data set import\n",
    "\n",
    "load(\"./data/dbp.R\")\n",
    "ls()\n",
    "dbp[1:5,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "8d3629ea-b954-4221-83df-ca2ba38040e4",
   "metadata": {
    "kernel": "R",
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Call:  glm(formula = affection ~ rs1112, family = binomial(\"logit\"), \n",
      "    data = dbp)\n",
      "\n",
      "Coefficients:\n",
      "(Intercept)      rs11123      rs11124  \n",
      "    -0.4449       0.7582       1.5435  \n",
      "\n",
      "Degrees of Freedom: 599 Total (i.e. Null);  597 Residual\n",
      "Null Deviance:\t    831.8 \n",
      "Residual Deviance: 797.7 \tAIC: 803.7\n",
      "[1] \"glm\" \"lm\" \n",
      "\n",
      "Call:\n",
      "glm(formula = affection ~ rs1112, family = binomial(\"logit\"), \n",
      "    data = dbp)\n",
      "\n",
      "Coefficients:\n",
      "            Estimate Std. Error z value Pr(>|z|)    \n",
      "(Intercept)  -0.4449     0.1189  -3.741 0.000183 ***\n",
      "rs11123       0.7582     0.1746   4.343 1.40e-05 ***\n",
      "rs11124       1.5435     0.3416   4.518 6.24e-06 ***\n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "\n",
      "(Dispersion parameter for binomial family taken to be 1)\n",
      "\n",
      "    Null deviance: 831.78  on 599  degrees of freedom\n",
      "Residual deviance: 797.75  on 597  degrees of freedom\n",
      "AIC: 803.75\n",
      "\n",
      "Number of Fisher Scoring iterations: 4\n",
      "\n",
      "[1] 4.077856e-08\n",
      "              Estimate Std. Error   z value     Pr(>|z|)\n",
      "(Intercept) -0.4449068  0.1189351 -3.740754 1.834691e-04\n",
      "rs11123      0.7582015  0.1745740  4.343154 1.404519e-05\n",
      "rs11124      1.5435191  0.3416277  4.518132 6.238747e-06\n",
      "  rs11123   rs11124 \n",
      "0.7582015 1.5435191 \n",
      " rs11123  rs11124 \n",
      "2.134434 4.681034 \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Waiting for profiling to be done...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                 2.5 %     97.5 %\n",
      "(Intercept) -0.6802726 -0.2135169\n",
      "rs11123      0.4176220  1.1023701\n",
      "rs11124      0.8984800  2.2475097\n",
      "                2.5 %    97.5 %\n",
      "(Intercept) 0.5064789 0.8077385\n",
      "rs11123     1.5183466 3.0112947\n",
      "rs11124     2.4558674 9.4641382\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       " affection rs1112 \n",
       " 0:300     2:297  \n",
       " 1:300     3:251  \n",
       "           4: 52  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       " affection     rs1112      \n",
       " 0:300     Min.   :0.0000  \n",
       " 1:300     1st Qu.:0.0000  \n",
       "           Median :1.0000  \n",
       "           Mean   :0.5917  \n",
       "           3rd Qu.:1.0000  \n",
       "           Max.   :2.0000  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "glm(formula = affection ~ rs1112, family = binomial(\"logit\"), \n",
       "    data = snp.data)\n",
       "\n",
       "Coefficients:\n",
       "            Estimate Std. Error z value Pr(>|z|)    \n",
       "(Intercept)  -0.4470     0.1142  -3.913 9.10e-05 ***\n",
       "rs1112        0.7652     0.1356   5.642 1.68e-08 ***\n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "(Dispersion parameter for binomial family taken to be 1)\n",
       "\n",
       "    Null deviance: 831.78  on 599  degrees of freedom\n",
       "Residual deviance: 797.75  on 598  degrees of freedom\n",
       "AIC: 801.75\n",
       "\n",
       "Number of Fisher Scoring iterations: 4\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis of Deviance Table\n",
      "\n",
      "Model: binomial, link: logit\n",
      "\n",
      "Response: affection\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "\n",
      "       Df Deviance Resid. Df Resid. Dev  Pr(>Chi)    \n",
      "NULL                     599     831.78              \n",
      "rs1112  1   34.026       598     797.75 5.438e-09 ***\n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n"
     ]
    }
   ],
   "source": [
    "# I. Logistic regression on a single SNP genotype\n",
    "\n",
    "result.snp12 = glm (affection ~ rs1112, family=binomial(\"logit\"), data=dbp)\n",
    "print (result.snp12)\n",
    "print ( class  (result.snp12) )\n",
    "print ( summary(result.snp12) )\n",
    "\n",
    "dev.geno = anova (result.snp12, test=\"Chi\")\n",
    "lrt.pvalue = pchisq(dev.geno[dim(dev.geno)[1],\"Deviance\"], \n",
    "             df=2, ncp=0, FALSE) \n",
    "print ( lrt.pvalue )\n",
    "\n",
    "print ( summary(result.snp12)$coefficients )\n",
    "snp.beta = summary(result.snp12)$coefficients[2:3,1]\n",
    "print ( snp.beta )\n",
    "print ( exp(snp.beta) )\n",
    "\n",
    "ci = confint (result.snp12)\n",
    "print (ci) \n",
    "print ( exp(ci) )\n",
    "\n",
    "snp.data = dbp[,c(\"affection\", \"rs1112\")]\n",
    "summary(snp.data)\n",
    "\n",
    "snp.data[,\"rs1112\"] <- as.numeric(snp.data[,\"rs1112\"]) - 1\n",
    "summary(snp.data)\n",
    "\n",
    "result.all = glm (affection ~ rs1112, family=binomial(\"logit\"), \n",
    "                  data=snp.data)\n",
    "dev.all    = anova (result.all, test=\"Chi\")\n",
    "summary(result.all)   \n",
    "print(dev.all) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "010526be-2e38-4c6d-82a1-9c1950811eba",
   "metadata": {
    "kernel": "R",
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       " affection     trait        sex          age        rs1112  rs1117 \n",
       " 0:300     Min.   : 60.50   1:329   Min.   :18.00   2:297   2:396  \n",
       " 1:300     1st Qu.: 77.44   2:271   1st Qu.:38.00   3:251   3:190  \n",
       "           Median : 82.00           Median :55.00   4: 52   4: 14  \n",
       "           Mean   : 81.85           Mean   :55.49                  \n",
       "           3rd Qu.: 86.09           3rd Qu.:74.00                  \n",
       "           Max.   :101.49           Max.   :90.00                  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "glm(formula = affection ~ sex + rs1112, family = binomial(\"logit\"), \n",
       "    data = snp.data)\n",
       "\n",
       "Coefficients:\n",
       "            Estimate Std. Error z value Pr(>|z|)    \n",
       "(Intercept) -0.08386    0.13730  -0.611    0.541    \n",
       "sex2        -0.81412    0.17253  -4.719 2.37e-06 ***\n",
       "rs1112       0.77139    0.13840   5.574 2.49e-08 ***\n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "(Dispersion parameter for binomial family taken to be 1)\n",
       "\n",
       "    Null deviance: 831.78  on 599  degrees of freedom\n",
       "Residual deviance: 774.98  on 597  degrees of freedom\n",
       "AIC: 780.98\n",
       "\n",
       "Number of Fisher Scoring iterations: 4\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "glm(formula = affection ~ age + rs1112, family = binomial(\"logit\"), \n",
       "    data = snp.data)\n",
       "\n",
       "Coefficients:\n",
       "             Estimate Std. Error z value Pr(>|z|)    \n",
       "(Intercept) -0.520422   0.250956  -2.074   0.0381 *  \n",
       "age          0.001322   0.004020   0.329   0.7423    \n",
       "rs1112       0.765189   0.135624   5.642 1.68e-08 ***\n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "(Dispersion parameter for binomial family taken to be 1)\n",
       "\n",
       "    Null deviance: 831.78  on 599  degrees of freedom\n",
       "Residual deviance: 797.64  on 597  degrees of freedom\n",
       "AIC: 803.64\n",
       "\n",
       "Number of Fisher Scoring iterations: 4\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "glm(formula = affection ~ sex + age + rs1112, family = binomial(\"logit\"), \n",
       "    data = snp.data)\n",
       "\n",
       "Coefficients:\n",
       "             Estimate Std. Error z value Pr(>|z|)    \n",
       "(Intercept) -0.198133   0.263732  -0.751    0.452    \n",
       "sex2        -0.817603   0.172736  -4.733 2.21e-06 ***\n",
       "age          0.002084   0.004105   0.508    0.612    \n",
       "rs1112       0.771546   0.138411   5.574 2.48e-08 ***\n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "(Dispersion parameter for binomial family taken to be 1)\n",
       "\n",
       "    Null deviance: 831.78  on 599  degrees of freedom\n",
       "Residual deviance: 774.72  on 596  degrees of freedom\n",
       "AIC: 782.72\n",
       "\n",
       "Number of Fisher Scoring iterations: 4\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "glm(formula = affection ~ rs1117 + rs1112, family = binomial(\"logit\"), \n",
       "    data = snp.data)\n",
       "\n",
       "Coefficients:\n",
       "            Estimate Std. Error z value Pr(>|z|)    \n",
       "(Intercept)  -0.4523     0.1144  -3.955 7.66e-05 ***\n",
       "rs1117        0.2853     0.2297   1.242  0.21431    \n",
       "rs1112        0.5999     0.1883   3.186  0.00144 ** \n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "(Dispersion parameter for binomial family taken to be 1)\n",
       "\n",
       "    Null deviance: 831.78  on 599  degrees of freedom\n",
       "Residual deviance: 796.21  on 597  degrees of freedom\n",
       "AIC: 802.21\n",
       "\n",
       "Number of Fisher Scoring iterations: 4\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A anova: 3 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Df</th><th scope=col>Deviance</th><th scope=col>Resid. Df</th><th scope=col>Resid. Dev</th><th scope=col>Pr(&gt;Chi)</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>NULL</th><td>NA</td><td>      NA</td><td>599</td><td>831.7766</td><td>          NA</td></tr>\n",
       "\t<tr><th scope=row>rs1117</th><td> 1</td><td>25.06376</td><td>598</td><td>806.7129</td><td>5.546551e-07</td></tr>\n",
       "\t<tr><th scope=row>rs1112</th><td> 1</td><td>10.50081</td><td>597</td><td>796.2120</td><td>1.193220e-03</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A anova: 3 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & Df & Deviance & Resid. Df & Resid. Dev & Pr(>Chi)\\\\\n",
       "  & <int> & <dbl> & <int> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tNULL & NA &       NA & 599 & 831.7766 &           NA\\\\\n",
       "\trs1117 &  1 & 25.06376 & 598 & 806.7129 & 5.546551e-07\\\\\n",
       "\trs1112 &  1 & 10.50081 & 597 & 796.2120 & 1.193220e-03\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A anova: 3 × 5\n",
       "\n",
       "| <!--/--> | Df &lt;int&gt; | Deviance &lt;dbl&gt; | Resid. Df &lt;int&gt; | Resid. Dev &lt;dbl&gt; | Pr(&gt;Chi) &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| NULL | NA |       NA | 599 | 831.7766 |           NA |\n",
       "| rs1117 |  1 | 25.06376 | 598 | 806.7129 | 5.546551e-07 |\n",
       "| rs1112 |  1 | 10.50081 | 597 | 796.2120 | 1.193220e-03 |\n",
       "\n"
      ],
      "text/plain": [
       "       Df Deviance Resid. Df Resid. Dev Pr(>Chi)    \n",
       "NULL   NA       NA 599       831.7766             NA\n",
       "rs1117  1 25.06376 598       806.7129   5.546551e-07\n",
       "rs1112  1 10.50081 597       796.2120   1.193220e-03"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "glm(formula = affection ~ rs1112 + rs1117, family = binomial(\"logit\"), \n",
       "    data = snp.data)\n",
       "\n",
       "Coefficients:\n",
       "            Estimate Std. Error z value Pr(>|z|)    \n",
       "(Intercept)  -0.4523     0.1144  -3.955 7.66e-05 ***\n",
       "rs1112        0.5999     0.1883   3.186  0.00144 ** \n",
       "rs1117        0.2853     0.2297   1.242  0.21431    \n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "(Dispersion parameter for binomial family taken to be 1)\n",
       "\n",
       "    Null deviance: 831.78  on 599  degrees of freedom\n",
       "Residual deviance: 796.21  on 597  degrees of freedom\n",
       "AIC: 802.21\n",
       "\n",
       "Number of Fisher Scoring iterations: 4\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A anova: 3 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Df</th><th scope=col>Deviance</th><th scope=col>Resid. Df</th><th scope=col>Resid. Dev</th><th scope=col>Pr(&gt;Chi)</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>NULL</th><td>NA</td><td>       NA</td><td>599</td><td>831.7766</td><td>          NA</td></tr>\n",
       "\t<tr><th scope=row>rs1112</th><td> 1</td><td>34.026174</td><td>598</td><td>797.7504</td><td>5.437568e-09</td></tr>\n",
       "\t<tr><th scope=row>rs1117</th><td> 1</td><td> 1.538401</td><td>597</td><td>796.2120</td><td>2.148560e-01</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A anova: 3 × 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & Df & Deviance & Resid. Df & Resid. Dev & Pr(>Chi)\\\\\n",
       "  & <int> & <dbl> & <int> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tNULL & NA &        NA & 599 & 831.7766 &           NA\\\\\n",
       "\trs1112 &  1 & 34.026174 & 598 & 797.7504 & 5.437568e-09\\\\\n",
       "\trs1117 &  1 &  1.538401 & 597 & 796.2120 & 2.148560e-01\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A anova: 3 × 5\n",
       "\n",
       "| <!--/--> | Df &lt;int&gt; | Deviance &lt;dbl&gt; | Resid. Df &lt;int&gt; | Resid. Dev &lt;dbl&gt; | Pr(&gt;Chi) &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| NULL | NA |        NA | 599 | 831.7766 |           NA |\n",
       "| rs1112 |  1 | 34.026174 | 598 | 797.7504 | 5.437568e-09 |\n",
       "| rs1117 |  1 |  1.538401 | 597 | 796.2120 | 2.148560e-01 |\n",
       "\n"
      ],
      "text/plain": [
       "       Df Deviance  Resid. Df Resid. Dev Pr(>Chi)    \n",
       "NULL   NA        NA 599       831.7766             NA\n",
       "rs1112  1 34.026174 598       797.7504   5.437568e-09\n",
       "rs1117  1  1.538401 597       796.2120   2.148560e-01"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# II. Adjustment for the effects of covariates and of other SNPs\n",
    "\n",
    "snp.data = dbp[,c(\"affection\", \"trait\",\"sex\", \"age\", \"rs1112\", \"rs1117\")]\n",
    "summary(snp.data)\n",
    "\n",
    "snp.data[,\"rs1112\"] <- as.numeric(snp.data[,\"rs1112\"]) - 1\n",
    "snp.data[,\"rs1117\"] <- as.numeric(snp.data[,\"rs1117\"]) - 1\n",
    "\n",
    "result.adj = glm (affection ~ sex + rs1112      , family=binomial(\"logit\"), \n",
    "                  data=snp.data)\n",
    "summary(result.adj)\n",
    "\n",
    "result.adj = glm (affection ~ age + rs1112      , family=binomial(\"logit\"), \n",
    "                  data=snp.data)\n",
    "summary(result.adj)\n",
    "\n",
    "result.adj = glm (affection ~ sex + age + rs1112, family=binomial(\"logit\"), \n",
    "                  data=snp.data)\n",
    "summary(result.adj)\n",
    "\n",
    "result.adj = glm (affection ~ rs1117 + rs1112, family=binomial(\"logit\"), \n",
    "                  data=snp.data)\n",
    "summary(result.adj)\n",
    "anova (result.adj, test=\"Chi\")\n",
    "\n",
    "result.adj = glm (affection ~ rs1112 + rs1117, family=binomial(\"logit\"), \n",
    "                  data=snp.data)\n",
    "summary(result.adj)\n",
    "anova (result.adj, test=\"Chi\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "5dab210f-6a2b-4907-b891-53c05d9dbdfd",
   "metadata": {
    "kernel": "R",
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "lm(formula = trait ~ rs1112, data = snp.data)\n",
       "\n",
       "Residuals:\n",
       "     Min       1Q   Median       3Q      Max \n",
       "-22.5556  -3.9106   0.2194   4.0144  15.4809 \n",
       "\n",
       "Coefficients:\n",
       "            Estimate Std. Error t value Pr(>|t|)    \n",
       "(Intercept)  80.1021     0.3301 242.680  < 2e-16 ***\n",
       "rs1112        2.9535     0.3774   7.826 2.29e-14 ***\n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "Residual standard error: 5.954 on 598 degrees of freedom\n",
       "Multiple R-squared:  0.09291,\tAdjusted R-squared:  0.09139 \n",
       "F-statistic: 61.25 on 1 and 598 DF,  p-value: 2.292e-14\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "lm(formula = trait ~ sex + rs1112, data = snp.data)\n",
       "\n",
       "Residuals:\n",
       "     Min       1Q   Median       3Q      Max \n",
       "-20.9404  -3.6272   0.2234   3.7815  16.3480 \n",
       "\n",
       "Coefficients:\n",
       "            Estimate Std. Error t value Pr(>|t|)    \n",
       "(Intercept)  81.4542     0.3904 208.654  < 2e-16 ***\n",
       "sex2         -2.8823     0.4748  -6.071 2.27e-09 ***\n",
       "rs1112        2.8685     0.3668   7.820 2.41e-14 ***\n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "Residual standard error: 5.784 on 597 degrees of freedom\n",
       "Multiple R-squared:  0.1456,\tAdjusted R-squared:  0.1428 \n",
       "F-statistic: 50.89 on 2 and 597 DF,  p-value: < 2.2e-16\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# III. Analysis of quantitative instead of dichotomized trait\n",
    "\n",
    "result.adj = lm (trait ~ rs1112      , data=snp.data)\n",
    "summary(result.adj)\n",
    "\n",
    "result.adj = lm (trait ~ sex + rs1112, data=snp.data)\n",
    "summary(result.adj)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "4102ab5e-c9f6-4b03-92b6-b245cbfc285b",
   "metadata": {
    "kernel": "R",
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "glm(formula = affection ~ sex * rs1112, family = binomial(\"logit\"), \n",
       "    data = snp.data)\n",
       "\n",
       "Coefficients:\n",
       "            Estimate Std. Error z value Pr(>|z|)    \n",
       "(Intercept) -0.09415    0.15371  -0.613 0.540174    \n",
       "sex2        -0.79026    0.23515  -3.361 0.000777 ***\n",
       "rs1112       0.79049    0.18896   4.183 2.87e-05 ***\n",
       "sex2:rs1112 -0.04141    0.27771  -0.149 0.881472    \n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "(Dispersion parameter for binomial family taken to be 1)\n",
       "\n",
       "    Null deviance: 831.78  on 599  degrees of freedom\n",
       "Residual deviance: 774.96  on 596  degrees of freedom\n",
       "AIC: 782.96\n",
       "\n",
       "Number of Fisher Scoring iterations: 4\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "glm(formula = affection ~ age * rs1112, family = binomial(\"logit\"), \n",
       "    data = snp.data)\n",
       "\n",
       "Coefficients:\n",
       "             Estimate Std. Error z value Pr(>|z|)   \n",
       "(Intercept) -0.764365   0.328207  -2.329  0.01986 * \n",
       "age          0.005719   0.005508   1.038  0.29909   \n",
       "rs1112       1.193715   0.393377   3.035  0.00241 **\n",
       "age:rs1112  -0.007716   0.006585  -1.172  0.24130   \n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "(Dispersion parameter for binomial family taken to be 1)\n",
       "\n",
       "    Null deviance: 831.78  on 599  degrees of freedom\n",
       "Residual deviance: 796.26  on 596  degrees of freedom\n",
       "AIC: 804.26\n",
       "\n",
       "Number of Fisher Scoring iterations: 4\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Call:\n",
       "glm(formula = affection ~ rs1112 * rs1117, family = binomial(\"logit\"), \n",
       "    data = snp.data)\n",
       "\n",
       "Coefficients:\n",
       "              Estimate Std. Error z value Pr(>|z|)    \n",
       "(Intercept)   -0.45855    0.11749  -3.903  9.5e-05 ***\n",
       "rs1112         0.61285    0.19612   3.125  0.00178 ** \n",
       "rs1117         0.37232    0.43522   0.855  0.39228    \n",
       "rs1112:rs1117 -0.07464    0.31590  -0.236  0.81323    \n",
       "---\n",
       "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
       "\n",
       "(Dispersion parameter for binomial family taken to be 1)\n",
       "\n",
       "    Null deviance: 831.78  on 599  degrees of freedom\n",
       "Residual deviance: 796.16  on 596  degrees of freedom\n",
       "AIC: 804.16\n",
       "\n",
       "Number of Fisher Scoring iterations: 4\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# IV. Gene-environment (GxE) and gene-gene (GxG) interaction\n",
    "\n",
    "result.inter = glm (affection ~ sex * rs1112, family=binomial(\"logit\"), \n",
    "                    data=snp.data)\n",
    "summary(result.inter)\n",
    "\n",
    "result.inter = glm (affection ~ age * rs1112, family=binomial(\"logit\"), \n",
    "                    data=snp.data)\n",
    "summary(result.inter)\n",
    "\n",
    "result.inter = glm (affection ~ rs1112 * rs1117, family=binomial(\"logit\"), \n",
    "                    data=snp.data)\n",
    "summary(result.inter)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "65029901-7822-4f6a-aef9-8629b0c21e09",
   "metadata": {
    "kernel": "R"
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f31ab497-2221-4cbf-9ed4-f189deef5940",
   "metadata": {
    "kernel": "R"
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b36740d6-5b57-40e0-8491-45384737ffb6",
   "metadata": {
    "kernel": "R"
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "SoS",
   "language": "sos",
   "name": "sos"
  },
  "language_info": {
   "codemirror_mode": "sos",
   "file_extension": ".sos",
   "mimetype": "text/x-sos",
   "name": "sos",
   "nbconvert_exporter": "sos_notebook.converter.SoS_Exporter",
   "pygments_lexer": "sos"
  },
  "sos": {
   "kernels": [
    [
     "Bash",
     "calysto_bash",
     "Bash",
     "#E6EEFF",
     "shell"
    ],
    [
     "R",
     "ir",
     "R",
     "#DCDCDA",
     "r"
    ]
   ],
   "version": ""
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
